<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-06-16" />

<title>characterization Spleen data</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSLO_BRC_CXCL13</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">characterization Spleen data</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">16 June 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSLO_BRC_CXCL13/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210519code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210519)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210519code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210519)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree1f27cea5638ec11795ed815300ef71869e6d3b0btargetblank1f27ceaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/1f27cea5638ec11795ed815300ef71869e6d3b0b" target="_blank">1f27cea</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree1f27cea5638ec11795ed815300ef71869e6d3b0btargetblank1f27ceaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/1f27cea5638ec11795ed815300ef71869e6d3b0b" target="_blank">1f27cea</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/destiny/
    Ignored:    data/processedData/

Unstaged changes:
    Modified:   analysis/CharacterizePP_CXCL13seurat.Rmd
    Modified:   analysis/CharacterizeSpleen_CXCL13seurat.Rmd
    Modified:   analysis/IntegrateAcrossSLOs.Rmd
    Modified:   analysis/SLOspecific_signature.Rmd
    Modified:   analysis/visFolBRC_merged.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/CharacterizeSpleen_CXCL13seurat.Rmd</code>) and HTML (<code>docs/CharacterizeSpleen_CXCL13seurat.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/analysis/CharacterizeSpleen_CXCL13seurat.Rmd" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
<td>
update colors and labels plus marker genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/CharacterizeSpleen_CXCL13seurat.html" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
<td>
update colors and labels plus marker genes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/analysis/CharacterizeSpleen_CXCL13seurat.Rmd" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
<td>
characterize all SLO
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/CharacterizeSpleen_CXCL13seurat.html" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
<td>
characterize all SLO
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(SingleCellExperiment)
  library(RColorBrewer)
  library(viridis)
  library(ggsci)
  library(scater)
  library(scran)
  library(pheatmap)
  library(CellMixS)
  library(muscat)
  library(edgeR)
  library(rcartocolor)
  library(harmony)
})</code></pre>
</div>
<div id="set-dir-and-read-input-data" class="section level2">
<h2>set dir and read input data</h2>
<pre class="r"><code>basedir &lt;- here()
sce &lt;- readRDS(paste0(basedir, &quot;/data/allSamplesMerged_plusDR_sce.rds&quot;))
sce &lt;- sce[,sce$SLO ==&quot;Spleen&quot;]</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color Vectors</h2>
<pre class="r"><code>colorSLO &lt;- c(viridis(3)[c(1,2)], brewer.pal(n=9, name=&quot;YlOrRd&quot;)[c(4)])
colorCond &lt;- brewer.pal(n=3, name=&quot;Paired&quot;)[c(1,2)]
colorSLOCond &lt;- c(&quot;#440154FF&quot;,&quot;#807DBA&quot;, &quot;#FEB24C&quot;, &quot;#F16913&quot;, &quot;#21908CFF&quot;)
colorBatch &lt;- pal_igv()(n=length(unique(sce$batch)))

# set colorPal names
names(colorSLO) &lt;- c(&quot;LN&quot;, &quot;PP&quot;, &quot;Spleen&quot;)
names(colorCond) &lt;- c(&quot;naive&quot;, &quot;immunized&quot;)
names(colorBatch) &lt;- unique(sce$batch)
names(colorSLOCond) &lt;- c(&quot;LN_immunized&quot;,&quot;LN_naive&quot;, &quot;Spleen_naive&quot;,
                         &quot;Spleen_immunized&quot;, &quot;PP_immunized&quot;)


colPal2 &lt;- c(rcartocolor::carto_pal(name=&quot;Safe&quot;), pal_futurama()(8))

## Spleen specific colPal
## final color palette
colPal3 &lt;- c(&quot;#87c5af&quot;, &quot;#94033b&quot;, &quot;#fc8f3b&quot;, &quot;#2580fe&quot;, &quot;#514e72&quot;, &quot;#007961&quot;,
            &quot;#f4d403&quot;, &quot;#38b5fc&quot;, &quot;#eb6382&quot;, &quot;#8fa38c&quot;, &quot;#4673b4&quot;, &quot;#b94403&quot;,
            &quot;#b7d165&quot;, &quot;#9b5662&quot;, &quot;#666b73&quot;, &quot;#075c62&quot;)
names(colPal3) &lt;- c(&quot;LZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;DZFDC&quot;,&quot;PRC2&quot;,&quot;RPF&quot;, &quot;muralCells&quot;, 
                   &quot;PRC1&quot;, &quot;BCRC&quot;, &quot;medRC1&quot;,&quot;IFRC&quot;, &quot;Cxcl9TBRC&quot;, &quot;capsular&quot;,
                   &quot;LPFRC&quot;, &quot;subepithelial&quot;, &quot;aFDC&quot;)


## final color palette
colPal &lt;- c(&quot;#87c5af&quot;, &quot;#94033b&quot;, &quot;#fc8f3b&quot;, &quot;#2580fe&quot;, &quot;#514e72&quot;, &quot;#007961&quot;,
            &quot;#f4d403&quot;, &quot;#38b5fc&quot;, &quot;#eb6382&quot;, &quot;#8fa38c&quot;, &quot;#4673b4&quot;, &quot;#b94403&quot;,
            &quot;#b7d165&quot;, &quot;#9b5662&quot;, &quot;#666b73&quot;)
names(colPal) &lt;- c(&quot;LZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;DZFDC&quot;,&quot;PRC2&quot;,&quot;RPF&quot;, &quot;muralCells&quot;, 
                   &quot;PRC1&quot;, &quot;BCRC&quot;, &quot;medRC&quot;,&quot;IFRC&quot;, &quot;Cxcl9TBRC&quot;, &quot;capsular&quot;,
                   &quot;LPFRC&quot;, &quot;subepithelial&quot;)</code></pre>
</div>
<div id="run-seurat-pipeline" class="section level2">
<h2>run seurat pipeline</h2>
<pre class="r"><code>seurat &lt;- as.Seurat(sce)
dim(seurat)</code></pre>
<pre><code>[1] 29568 15590</code></pre>
<pre class="r"><code>seurat &lt;- subset(seurat, subset = ENSMUSG00000023078.Cxcl13 &gt;0)
seurat &lt;- rerunSeurat3(seurat)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10712
Number of edges: 370096

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9477
Number of communities: 12
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10712
Number of edges: 370096

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9067
Number of communities: 16
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10712
Number of edges: 370096

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8876
Number of communities: 18
Elapsed time: 1 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 10712
Number of edges: 370096

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.9282
Number of communities: 14
Elapsed time: 1 seconds</code></pre>
<pre class="r"><code>dim(seurat)</code></pre>
<pre><code>[1] 29568 10712</code></pre>
</div>
<div id="visualize-data" class="section level2 tabset">
<h2 class="tabset">visualize data</h2>
<div id="clustering" class="section level3">
<h3>clustering</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal2)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-1">
Past versions of clustering-1.png
</button>
</p>
<div id="fig-clustering-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/clustering-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="batch" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorBatch, group.by = &quot;batch&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Batch-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Batch-1">
Past versions of Batch-1.png
</button>
</p>
<div id="fig-Batch-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Batch-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Batch-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cond" class="section level3">
<h3>Cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorCond, group.by = &quot;cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Side-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Side-1">
Past versions of Side-1.png
</button>
</p>
<div id="fig-Side-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Side-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Side-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="vis-qc" class="section level2 tabset">
<h2 class="tabset">vis QC</h2>
<div id="total-features" class="section level3">
<h3>total features</h3>
<pre class="r"><code>FeaturePlot(seurat, reduction = &quot;umap&quot;,  features = &quot;total_features_by_counts&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20QC%20total%20features-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-QC-total-features-1">
Past versions of “vis QC total features-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-QC-total-features-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis QC total features-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="total-counts" class="section level3">
<h3>total counts</h3>
<pre class="r"><code>FeaturePlot(seurat, reduction = &quot;umap&quot;,  features = &quot;total_counts&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20QC%20total%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-QC-total-counts-1">
Past versions of “vis QC total counts-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-QC-total-counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis QC total counts-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="percentage-mt-counts" class="section level3">
<h3>percentage mt counts</h3>
<pre class="r"><code>FeaturePlot(seurat, reduction = &quot;umap&quot;,  features = &quot;pct_counts_Mt&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20QC%20percentage%20mt%20counts-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-QC-percentage-mt-counts-1">
Past versions of “vis QC percentage mt counts-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-QC-percentage-mt-counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis QC percentage mt counts-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="find-marker" class="section level2">
<h2>Find marker</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$originalexp_snn_res.0.25
seurat_markers_all &lt;- FindAllMarkers(object = seurat,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)

write.table(seurat_markers_all,
            file=paste0(basedir, &quot;/data/Spleen_marker_SeuratClust.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)


saveRDS(seurat, file=paste0(basedir, &quot;/data/Spleen_seurat.rds&quot;))

### res 0.25
## cluster 0,5,10: red pulb fibroblasts (Eng, Ly6a, Cxcl12)
## cluster 11: bridging TBRC/MRC (Ccl21a, Ccl19, Tnfsf11)
## cluster 2, 12: mural cells (Cnn1, Rgs5, Mcam, Pdgfrb)
## cluster 5: adventitial cells (Wt1, Il33, Cd34)
## cluster 3: FDC (Cr2, Fcamr, Fcgr2b, Sox9)
## cluster 7: intPRC (Madcam1. Il33, Cxcl13)
## cluster 9: PRC (Ly6a, Tlx1, Wt1)
## cluster 8: capsular fibroblasts (Lum, Cd34)
## cluster 1: TBRC (Ccl19, Ccl21a) 
## cluster 6: aFDC (Cr2, Fcamr, Fcgr2b, Sox9)


### res 0.4:
## cluster 0,3, 12: red pulb fibroblasts (Eng, Ly6a, Cxcl12)
## cluster 5: adventitial cells (Wt1, Il33, Cd34)
## cluster 2: TRC/TBRC (Ccl21a, Ccl19, Grem1)
## cluster 1,7,12: TNC1 (mural cells) (Mcam, Cnn1, Cspg4)
## cluster 4, 9: FDC (Cr2, Mfge8, Fcgr2b)
## cluster : MRC (Madcam1. Il33, Cxcl13)
## cluster 6: capsular fibroblasts (Lum, Cd34)
## cluster : PRCs/mural cells (?) 
## cluster : (Srgn, Bst1, Crym, Mfge8, Ccl21a) </code></pre>
</div>
<div id="expression-of-xist" class="section level2 tabset">
<h2 class="tabset">expression of Xist</h2>
<div id="all" class="section level3">
<h3>all</h3>
<pre class="r"><code>FeaturePlot(seurat, reduction = &quot;umap&quot;,  features = &quot;ENSMUSG00000086503.Xist&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist%20expression%20all-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Xist-expression-all-1">
Past versions of “Xist expression all-1.png”
</button>
</p>
<div id="fig-no-spaces-Xist-expression-all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist expression all-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="batch-x" class="section level3">
<h3>batch X</h3>
<pre class="r"><code>seuratS &lt;- subset(seurat, batch==&quot;BX&quot;)
FeaturePlot(seuratS, reduction = &quot;umap&quot;,  features = &quot;ENSMUSG00000086503.Xist&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist%20expression%20BX-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Xist-expression-BX-1">
Past versions of “Xist expression BX-1.png”
</button>
</p>
<div id="fig-no-spaces-Xist-expression-BX-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist expression BX-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="batch-y" class="section level3">
<h3>batch Y</h3>
<pre class="r"><code>seuratS &lt;- subset(seurat, batch==&quot;BY&quot;)
FeaturePlot(seuratS, reduction = &quot;umap&quot;,  features = &quot;ENSMUSG00000086503.Xist&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist%20expression%20BY-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Xist-expression-BY-1">
Past versions of “Xist expression BY-1.png”
</button>
</p>
<div id="fig-no-spaces-Xist-expression-BY-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist expression BY-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="batch-z" class="section level3">
<h3>batch Z</h3>
<pre class="r"><code>seuratS &lt;- subset(seurat, batch==&quot;BZ&quot;)
FeaturePlot(seuratS, reduction = &quot;umap&quot;,  features = &quot;ENSMUSG00000086503.Xist&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist%20expression%20BZ-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Xist-expression-BZ-1">
Past versions of “Xist expression BZ-1.png”
</button>
</p>
<div id="fig-no-spaces-Xist-expression-BZ-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist expression BZ-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="batch-w" class="section level3">
<h3>batch W</h3>
<pre class="r"><code>seuratS &lt;- subset(seurat, batch==&quot;BW&quot;)
FeaturePlot(seuratS, reduction = &quot;umap&quot;,  features = &quot;ENSMUSG00000086503.Xist&quot;,
            cols = c(&quot;lightgrey&quot;, &quot;darkred&quot;)) +
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist%20expression%20BW-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Xist-expression-BW-1">
Past versions of “Xist expression BW-1.png”
</button>
</p>
<div id="fig-no-spaces-Xist-expression-BW-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/Xist expression BW-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>remove(seuratS)</code></pre>
</div>
</div>
<div id="re-embedding-fdcsmrcstbrcs" class="section level2">
<h2>re-embedding FDCs/MRCs/TBRCs</h2>
<pre class="r"><code>seuratSub &lt;- subset(seurat, idents=c(&quot;3&quot;, &quot;1&quot;))

seuratSub &lt;- rerunSeurat3(seuratSub, res = c(0.4, 0.8, 0.25, 0.6))</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3322
Number of edges: 116653

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8222
Number of communities: 5
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3322
Number of edges: 116653

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7515
Number of communities: 11
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3322
Number of edges: 116653

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8706
Number of communities: 4
Elapsed time: 0 seconds
Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 3322
Number of edges: 116653

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.7834
Number of communities: 7
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>seurat_markers_sub &lt;- FindAllMarkers(object = seuratSub,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)


## res 0.6
## cluster 1 + 5 + 4 = TBRC (Ccl21a, Ccl19, Fmod, Prelp, Cxcl13 gradient)
## cluster 2 = LZFDC (Cr2, Fcamr, Fcer2a)
## cluster 3 = DZFDC
## cluster 0 = MRC (Tnfsf11, Madcam1)
## cluster 6 = mural</code></pre>
</div>
<div id="assign-labels" class="section level2">
<h2>assign labels</h2>
<pre class="r"><code>## cluster 0,9: red pulb fibroblasts (Eng, Ly6a, Cxcl12, Kazald1)
  ## cluster : bridging TBRC/MRC (Ccl21a, Ccl19, Tnfsf11)
## cluster 2, 5, 8, 11: mural cells/pericytes (Cnn1, Rgs5, Mcam, Pdgfrb, Myh11)
## cluster 5: half Thy1 TRC (Cst3, Gsn, Mfap4, Thy1, Dpt, Ccl11, Ccl19, Ccl21a)
## cluster 11: &quot;pericytes2&quot; (Lyz1, Mcam, Cpxm2, Nog) 
## cluster 4: adventitial cells/mesothelial (Wt1, Krt19, Lgals7, Cd34, no Ly6c1 or Pi16)
## cluster 3: FDC (Cr2, Fcamr, Fcgr2b, Sox9)
  ## cluster 7: intPRC (Madcam1. Il33, Cxcl13)
  ## cluster 9: PRC (Ly6a, Tlx1, Wt1)
## cluster 7: capsular fibroblasts (Lum, Cd34) (also Pi16 and Ly6c1!)
## cluster 1: TBRC/MRC (Ccl19, Ccl21a) 
## cluster 6: aFDC (Cr2, Fcamr, Fcgr2b, Sox9, Atp5e)
## cluster 10: BCRC(?) (Bst1, Clic5, Ccl21a, Kazald1) (markers from both RPRC + TBRCs)


## label cells
seurat$clusterLabel &lt;- &quot;muralCells&quot;
seurat$clusterLabel[which(seurat$originalexp_snn_res.0.25 %in%
                            c(&quot;0&quot;, &quot;9&quot;))] &lt;- &quot;RPF&quot;
seurat$clusterLabel[which(seurat$originalexp_snn_res.0.25 == &quot;10&quot;)] &lt;- &quot;BCRC&quot;
seurat$clusterLabel[which(seurat$originalexp_snn_res.0.25 == &quot;4&quot;)] &lt;- &quot;capsular&quot;
seurat$clusterLabel[which(seurat$originalexp_snn_res.0.25 == &quot;7&quot;)] &lt;- &quot;PRC2&quot;
seurat$clusterLabel[which(seurat$originalexp_snn_res.0.25 == &quot;6&quot;)] &lt;- &quot;aFDC&quot;

### label from re-embedding
fdc &lt;- colnames(seuratSub)[which(seuratSub$originalexp_snn_res.0.8 %in%
                                   c(&quot;2&quot;, &quot;3&quot;, &quot;7&quot;, &quot;8&quot;, &quot;10&quot;))]
mrc &lt;- colnames(seuratSub)[which(seuratSub$originalexp_snn_res.0.8 == &quot;5&quot;)]
tbrc &lt;- colnames(seuratSub)[which(seuratSub$originalexp_snn_res.0.8 %in%
                                    c(&quot;0&quot;, &quot;1&quot;, &quot;6&quot;, &quot;9&quot;))]
dzfdc &lt;- colnames(seuratSub)[which(seuratSub$originalexp_snn_res.0.8 == &quot;4&quot;)]

seurat$clusterLabel[which(colnames(seurat) %in% fdc)] &lt;- &quot;LZFDC&quot;
seurat$clusterLabel[which(colnames(seurat) %in% mrc)] &lt;- &quot;MRC&quot;
seurat$clusterLabel[which(colnames(seurat) %in% tbrc)] &lt;- &quot;TBRC&quot;
seurat$clusterLabel[which(colnames(seurat) %in% dzfdc)] &lt;- &quot;DZFDC&quot;


Idents(seurat) &lt;- seurat$clusterLabel</code></pre>
<p>Note 1: BCRC express both, markers from RPRC and TBRCs (e.g. Bst1, Ccl19 (WP marker)<br />
as well as Clic5, Kazald1 (RP marker))<br />
Note 2: mural cells comprise multiple cluster that are assigned as pericytes in<br />
the Scott Mueller paper and intPRCs plus TNC1 in HW paper<br />
Note 3: Some cells of one mural cell cluster express marker similar to the Thy1 TRC<br />
cluster from the Scott Mueller study<br />
Note 4: The adventitial cell cluster includes mesothelial cells (Krt19, Lgals7) and<br />
shows no expression of Ly6c1 and Pi16<br />
Note 5: capsular fibroblasts contain “universal Fibroblast” cluster (Cd34, Dpt, Pi16 and Ly6c1)</p>
</div>
<div id="vis-cluster-label" class="section level2">
<h2>vis cluster label</h2>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal3)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20cl%20lab-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cl-lab-1">
Past versions of “vis cl lab-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-cl-lab-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis cl lab-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis cl lab-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal3, shuffle=T,pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20cl%20lab-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cl-lab-2">
Past versions of “vis cl lab-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-cl-lab-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis cl lab-2.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="vis-cluster-characterization" class="section level2 tabset">
<h2 class="tabset">vis cluster characterization</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  selGenes &lt;- selGenes$gene
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;% filter(geneID %in% selGenes)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
<pre class="r"><code>## cluster 0 = TBRC (Ccl21a, Ccl19, Fmod)
## cluster 1 = MRC (Tnfsf11, Madcam1)
## cluster 2 = FDC (Cr2, Fcamr, Fcer2a)
## cluster 3 = FDC (Cox4i1, Atp5e)
## cluster 4 = bTBRC (Ccl21a, Ccl19, Wt1 ,Cyp4b1, Bst1)


selMarker &lt;- list(FDC = c(&quot;Xist&quot;,&quot;Sox9&quot;, &quot;Fcer2a&quot;, &quot;Cr2&quot;, &quot;Fcamr&quot;),
                  aFDC = c(&quot;Cox4i1&quot;,&quot;Atp5e&quot;),
                  MRC = c(&quot;Tnfsf11&quot;, &quot;Hamp2&quot;, &quot;Madcam1&quot;, &quot;Cxcl13&quot;),
                  TBRC = c(&quot;Bst1&quot;, &quot;Glycam1&quot;, &quot;Fmod&quot;,&quot;Ccl21a&quot;,&quot;Ccl19&quot;,&quot;Ifitm1&quot;),
                  DZFDC = c(&quot;Pdlim3&quot;, &quot;Crym&quot;, &quot;Emp2&quot;, &quot;Ecm1&quot;, &quot;Trf&quot;, &quot;Il6&quot;,
                            &quot;Cxcl1&quot;, &quot;Cxcl16&quot;, &quot;Ptgs2&quot;, &quot;Gstp1&quot;, &quot;Cd151&quot;, 
                            &quot;Clec14a&quot;, &quot;Plat&quot;, &quot;Ptn&quot;, &quot;Mmp3&quot;),
                  PRC = c(&quot;Pdgfrb&quot;, &quot;Tlx1&quot;, &quot;Eng&quot;, &quot;Cxcl12&quot;, &quot;Ly6a&quot;, &quot;Wt1&quot;),
                  mural = c(&quot;Rgs5&quot;, &quot;Cnn1&quot;, &quot;Acta2&quot;, &quot;Mcam&quot;),
                  adventitial = c(&quot;Cd34&quot;,&quot;Igfbp6&quot;, &quot;Lum&quot;)
                  )


selMarker &lt;- list(FDC = c(&quot;Fcgr2b&quot;, &quot;Cr2&quot;,&quot;Sox9&quot;),
                  aFDC = c(&quot;Cox4i1&quot;,&quot;Atp5e&quot;),
                  MRC = c(&quot;Cxcl13&quot;, &quot;Madcam1&quot;, &quot;Tnfsf11&quot;),                  
                  DZFDC = c(&quot;Pdlim3&quot;, &quot;Crym&quot;, &quot;Cxcl12&quot;),
                  TBRC = c(&quot;Ccl21a&quot;,&quot;Ccl19&quot;, &quot;Fmod&quot;),
                  RPF = c(&quot;Pdgfrb&quot;, &quot;Eng&quot;, &quot;Ly6a&quot;),
                  mural = c(&quot;Cnn1&quot;, &quot;Cspg4&quot;, &quot;Mcam&quot;),
                  capsular = c(&quot;Krt19&quot;, &quot;Upk3b&quot;),
                  adventitial = c(&quot;Cd34&quot;,&quot;Pi16&quot;)
                  )</code></pre>
<div id="avg-heatmap" class="section level3">
<h3>avg heatmap</h3>
<pre class="r"><code>selGenes &lt;- data.frame(gene=unlist(selMarker)) %&gt;%
  rownames_to_column(var=&quot;grp&quot;) %&gt;% mutate(Grp=gsub(&quot;.{1}$&quot;, &quot;&quot;, grp))
grpCnt &lt;- selGenes %&gt;% group_by(Grp) %&gt;% summarise(cnt=n())
gapR &lt;- data.frame(Grp=unique(selGenes$Grp)) %&gt;% 
  left_join(.,grpCnt, by=&quot;Grp&quot;) %&gt;% mutate(cumSum=cumsum(cnt)) 
ordVec &lt;- levels(seurat)[c(1, 9, 6, 4, 3, 10, 2, 8, 5,7)]

pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenes,
                  colVecIdent = colPal3, 
                  ordVec=ordVec,
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=F,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/avg%20heatmap%20cl%20marker-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avg-heatmap-cl-marker-1">
Past versions of “avg heatmap cl marker-1.png”
</button>
</p>
<div id="fig-no-spaces-avg-heatmap-cl-marker-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/avg heatmap cl marker-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/avg heatmap cl marker-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>##save seurat</p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterLabel
saveRDS(seurat, file=paste0(basedir, &quot;/data/Spleen_seurat.rds&quot;))</code></pre>
</div>
</div>
<div id="vis-cluster-label-wo-afdcs" class="section level2">
<h2>vis cluster label wo aFDCs</h2>
<pre class="r"><code>seurat &lt;- subset(seurat, clusterLabel != &quot;aFDC&quot;)
seurat &lt;- ScaleData(object = seurat, verbose = FALSE,
                        features = rownames(seurat))
seurat &lt;- RunPCA(object = seurat, npcs = 20, verbose = FALSE)
seurat &lt;- RunTSNE(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)
seurat &lt;- RunUMAP(object = seurat, reduction = &quot;pca&quot;, dims = 1:20)

Idents(seurat) &lt;- seurat$clusterLabel

DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, shuffle=T)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20cl%20lab%20merged-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cl-lab-merged-1">
Past versions of “vis cl lab merged-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-cl-lab-merged-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis cl lab merged-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, shuffle=T,pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis%20cl%20lab%20merged-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-cl-lab-merged-2">
Past versions of “vis cl lab merged-2.png”
</button>
</p>
<div id="fig-no-spaces-vis-cl-lab-merged-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/vis cl lab merged-2.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="avg-heatmap-1" class="section level3">
<h3>avg heatmap</h3>
<pre class="r"><code>selMarker &lt;- list(FDC = c(&quot;Fcgr2b&quot;, &quot;Cr2&quot;,&quot;Sox9&quot;),
#                  aFDC = c(&quot;Cox4i1&quot;,&quot;Atp5e&quot;),
                  DZFDC = c(&quot;Pdlim3&quot;, &quot;Crym&quot;, &quot;Cxcl12&quot;),
                  MRC = c(&quot;Cxcl13&quot;, &quot;Madcam1&quot;, &quot;Tnfsf11&quot;),                  
                  TBRC = c(&quot;Ccl21a&quot;,&quot;Ccl19&quot;, &quot;Fmod&quot;),
                  RPF = c(&quot;Pdgfrb&quot;, &quot;Eng&quot;, &quot;Ly6a&quot;),
                  mural = c(&quot;Cnn1&quot;, &quot;Cspg4&quot;, &quot;Mcam&quot;),
                  capsular = c(&quot;Krt19&quot;, &quot;Upk3b&quot;),
                  adventitial = c(&quot;Cd34&quot;,&quot;Pi16&quot;)
                  )

selGenes &lt;- data.frame(gene=unlist(selMarker)) %&gt;%
  rownames_to_column(var=&quot;grp&quot;) %&gt;% mutate(Grp=gsub(&quot;.{1}$&quot;, &quot;&quot;, grp))
grpCnt &lt;- selGenes %&gt;% group_by(Grp) %&gt;% summarise(cnt=n())
gapR &lt;- data.frame(Grp=unique(selGenes$Grp)) %&gt;% 
  left_join(.,grpCnt, by=&quot;Grp&quot;) %&gt;% mutate(cumSum=cumsum(cnt)) 
ordVec &lt;- levels(seurat)[c(5,3,8,2,9,1,7,4,6)]

pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecR=gapR$cumSum, gapVecC=NULL,cc=F,
                  cr=F, condCol=F)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/avg%20heatmap%20cl%20marker%20merged-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-avg-heatmap-cl-marker-merged-1">
Past versions of “avg heatmap cl marker merged-1.png”
</button>
</p>
<div id="fig-no-spaces-avg-heatmap-cl-marker-merged-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/avg heatmap cl marker merged-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="compare-to-all-eyfp-data" class="section level2">
<h2>compare to all eyfp data</h2>
<pre class="r"><code>seuratEYFP &lt;- readRDS(paste0(&quot;/Users/mechthildluetge/Projects/scRNAseq/&quot;,
                             &quot;CrossSLO_BCR/data/Spleen_seurat.rds&quot;))
seuratEYFPdat &lt;- data.frame(labelOld=seuratEYFP$clusterLabel) %&gt;%
  rownames_to_column(var=&quot;cellname&quot;)

seuratdat &lt;- data.frame(labelNew=seurat$clusterLabel) %&gt;%
  rownames_to_column(var=&quot;cellname&quot;) %&gt;%
  left_join(., seuratEYFPdat, by=&quot;cellname&quot;)

table(seuratdat$labelNew, seuratdat$labelOld)</code></pre>
<pre><code>            
             adventitial aFDC capsular DZFDC LZFDC  MRC muralCells  PRC redPulpFibroblast TBRC
  BCRC                 0    2        0     0     0    0          0  124                 1    1
  capsular          1000    0        1     0     0    1          1    0                 0    1
  DZFDC                0    0        0    13     0  363          0    0                 0    0
  LZFDC                0    0        0   515   632    8          0    1                 0   12
  MRC                  0    0        0     0     0  330          0    3                 1   32
  muralCells           2    0        1     0     0    0       1970  439                12  191
  PRC2                 1    0      309     0     0    0          0    0                 0    0
  RPF                  0    1        0     0     0    0          3    3              2972    3
  TBRC                 1    0        0     0     1   16          0   11                 4 1379</code></pre>
<pre class="r"><code>seurat$labelOld &lt;- seuratdat$labelOld
table(seurat$labelOld)</code></pre>
<pre><code>
      adventitial              aFDC          capsular             DZFDC             LZFDC 
             1004                 3               311               528               633 
              MRC        muralCells               PRC redPulpFibroblast              TBRC 
              718              1974               581              2990              1619 </code></pre>
<pre class="r"><code>table(seurat$clusterLabel)</code></pre>
<pre><code>
      BCRC   capsular      DZFDC      LZFDC        MRC muralCells       PRC2        RPF       TBRC 
       128       1004        376       1168        366       2615        310       2982       1412 </code></pre>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, group.by=&quot;labelOld&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/CharacterizeSpleen_CXCL13seurat.Rmd/comp%20all%20eyfp-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-comp-all-eyfp-1">
Past versions of “comp all eyfp-1.png”
</button>
</p>
<div id="fig-no-spaces-comp-all-eyfp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/comp all eyfp-1.png" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/afcdedc9a4aa5d06f26fd369c0a974adfc7971ba/docs/figure/CharacterizeSpleen_CXCL13seurat.Rmd/comp all eyfp-1.png" target="_blank">afcdedc</a>
</td>
<td>
mluetge
</td>
<td>
2021-05-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>##save seurat wo aFDC</p>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterLabel
saveRDS(seurat, file=paste0(basedir, &quot;/data/Spleen_wo_aFDC_seurat.rds&quot;))</code></pre>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] harmony_0.1.0               Rcpp_1.0.7                  rcartocolor_2.0.0          
 [4] edgeR_3.34.0                limma_3.48.1                muscat_1.6.0               
 [7] CellMixS_1.8.0              kSamples_1.2-9              SuppDists_1.1-9.5          
[10] pheatmap_1.0.12             scran_1.20.1                scater_1.20.1              
[13] scuttle_1.2.0               ggsci_2.9                   viridis_0.6.1              
[16] viridisLite_0.4.0           RColorBrewer_1.1-2          SingleCellExperiment_1.14.1
[19] SummarizedExperiment_1.22.0 Biobase_2.52.0              GenomicRanges_1.44.0       
[22] GenomeInfoDb_1.28.1         IRanges_2.26.0              S4Vectors_0.30.0           
[25] BiocGenerics_0.38.0         MatrixGenerics_1.4.0        matrixStats_0.60.0         
[28] runSeurat3_0.1.0            here_1.0.1                  magrittr_2.0.1             
[31] SeuratObject_4.0.2          Seurat_4.0.3                forcats_0.5.1              
[34] stringr_1.4.0               dplyr_1.0.7                 purrr_0.3.4                
[37] readr_2.0.0                 tidyr_1.1.3                 tibble_3.1.3               
[40] ggplot2_3.3.5               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] estimability_1.3          scattermore_0.7           coda_0.19-4              
  [4] bit64_4.0.5               knitr_1.33                multcomp_1.4-17          
  [7] irlba_2.3.3               DelayedArray_0.18.0       data.table_1.14.0        
 [10] rpart_4.1-15              KEGGREST_1.32.0           RCurl_1.98-1.3           
 [13] doParallel_1.0.16         generics_0.1.0            ScaledMatrix_1.0.0       
 [16] TH.data_1.0-10            cowplot_1.1.1             RSQLite_2.2.7            
 [19] RANN_2.6.1                future_1.21.0             bit_4.0.4                
 [22] tzdb_0.1.2                spatstat.data_2.1-0       xml2_1.3.2               
 [25] lubridate_1.7.10          httpuv_1.6.1              assertthat_0.2.1         
 [28] xfun_0.24                 hms_1.1.0                 jquerylib_0.1.4          
 [31] evaluate_0.14             promises_1.2.0.1          progress_1.2.2           
 [34] fansi_0.5.0               caTools_1.18.2            dbplyr_2.1.1             
 [37] readxl_1.3.1              geneplotter_1.70.0        igraph_1.2.6             
 [40] DBI_1.1.1                 htmlwidgets_1.5.3         spatstat.geom_2.2-2      
 [43] ellipsis_0.3.2            RSpectra_0.16-0           backports_1.2.1          
 [46] annotate_1.70.0           deldir_0.2-10             sparseMatrixStats_1.4.0  
 [49] vctrs_0.3.8               Cairo_1.5-12.2            ROCR_1.0-11              
 [52] abind_1.4-5               cachem_1.0.5              withr_2.4.2              
 [55] emmeans_1.6.2-1           sctransform_0.3.2         prettyunits_1.1.1        
 [58] goftest_1.2-2             cluster_2.1.2             lazyeval_0.2.2           
 [61] crayon_1.4.1              genefilter_1.74.0         labeling_0.4.2           
 [64] pkgconfig_2.0.3           nlme_3.1-152              vipor_0.4.5              
 [67] blme_1.0-5                rlang_0.4.11              globals_0.14.0           
 [70] lifecycle_1.0.0           miniUI_0.1.1.1            sandwich_3.0-1           
 [73] modelr_0.1.8              rsvd_1.0.5                cellranger_1.1.0         
 [76] rprojroot_2.0.2           polyclip_1.10-0           lmtest_0.9-38            
 [79] Matrix_1.3-4              boot_1.3-28               zoo_1.8-9                
 [82] reprex_2.0.0              beeswarm_0.4.0            whisker_0.4              
 [85] ggridges_0.5.3            GlobalOptions_0.1.2       png_0.1-7                
 [88] rjson_0.2.20              bitops_1.0-7              KernSmooth_2.23-20       
 [91] Biostrings_2.60.1         blob_1.2.2                DelayedMatrixStats_1.14.0
 [94] workflowr_1.6.2           shape_1.4.6               parallelly_1.27.0        
 [97] beachmat_2.8.0            scales_1.1.1              memoise_2.0.0            
[100] plyr_1.8.6                ica_1.0-2                 gplots_3.1.1             
[103] zlibbioc_1.38.0           compiler_4.1.0            dqrng_0.3.0              
[106] lme4_1.1-27.1             clue_0.3-59               DESeq2_1.32.0            
[109] fitdistrplus_1.1-5        cli_3.0.1                 XVector_0.32.0           
[112] lmerTest_3.1-3            listenv_0.8.0             TMB_1.7.20               
[115] patchwork_1.1.1           pbapply_1.4-3             MASS_7.3-54              
[118] mgcv_1.8-36               tidyselect_1.1.1          stringi_1.7.3            
[121] highr_0.9                 yaml_2.2.1                BiocSingular_1.8.1       
[124] locfit_1.5-9.4            ggrepel_0.9.1             grid_4.1.0               
[127] sass_0.4.0                tools_4.1.0               future.apply_1.7.0       
[130] circlize_0.4.13           rstudioapi_0.13           bluster_1.2.1            
[133] foreach_1.5.1             git2r_0.28.0              metapod_1.0.0            
[136] gridExtra_2.3             farver_2.1.0              Rtsne_0.15               
[139] digest_0.6.27             shiny_1.6.0               broom_0.7.8              
[142] later_1.2.0               RcppAnnoy_0.0.18          httr_1.4.2               
[145] AnnotationDbi_1.54.1      ComplexHeatmap_2.8.0      colorspace_2.0-2         
[148] rvest_1.0.1               XML_3.99-0.6              fs_1.5.0                 
[151] tensor_1.5                reticulate_1.20           splines_4.1.0            
[154] uwot_0.1.10               statmod_1.4.36            spatstat.utils_2.2-0     
[157] plotly_4.9.4.1            xtable_1.8-4              nloptr_1.2.2.2           
[160] jsonlite_1.7.2            R6_2.5.0                  pillar_1.6.1             
[163] htmltools_0.5.1.1         mime_0.11                 minqa_1.2.4              
[166] glue_1.4.2                fastmap_1.1.0             BiocParallel_1.26.1      
[169] BiocNeighbors_1.10.0      codetools_0.2-18          mvtnorm_1.1-2            
[172] utf8_1.2.2                lattice_0.20-44           bslib_0.2.5.1            
[175] spatstat.sparse_2.0-0     pbkrtest_0.5.1            numDeriv_2016.8-1.1      
[178] ggbeeswarm_0.6.0          colorRamps_2.3            leiden_0.3.8             
[181] gtools_3.9.2              glmmTMB_1.1.2             survival_3.2-11          
[184] rmarkdown_2.9             munsell_0.5.0             GetoptLong_1.0.5         
[187] GenomeInfoDbData_1.2.6    iterators_1.0.13          variancePartition_1.22.0 
[190] haven_2.4.1               reshape2_1.4.4            gtable_0.3.0             
[193] spatstat.core_2.3-0      </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Jul 27 17:46:42 2021&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] harmony_0.1.0               Rcpp_1.0.7                  rcartocolor_2.0.0          
 [4] edgeR_3.34.0                limma_3.48.1                muscat_1.6.0               
 [7] CellMixS_1.8.0              kSamples_1.2-9              SuppDists_1.1-9.5          
[10] pheatmap_1.0.12             scran_1.20.1                scater_1.20.1              
[13] scuttle_1.2.0               ggsci_2.9                   viridis_0.6.1              
[16] viridisLite_0.4.0           RColorBrewer_1.1-2          SingleCellExperiment_1.14.1
[19] SummarizedExperiment_1.22.0 Biobase_2.52.0              GenomicRanges_1.44.0       
[22] GenomeInfoDb_1.28.1         IRanges_2.26.0              S4Vectors_0.30.0           
[25] BiocGenerics_0.38.0         MatrixGenerics_1.4.0        matrixStats_0.60.0         
[28] runSeurat3_0.1.0            here_1.0.1                  magrittr_2.0.1             
[31] SeuratObject_4.0.2          Seurat_4.0.3                forcats_0.5.1              
[34] stringr_1.4.0               dplyr_1.0.7                 purrr_0.3.4                
[37] readr_2.0.0                 tidyr_1.1.3                 tibble_3.1.3               
[40] ggplot2_3.3.5               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] estimability_1.3          scattermore_0.7           coda_0.19-4              
  [4] bit64_4.0.5               knitr_1.33                multcomp_1.4-17          
  [7] irlba_2.3.3               DelayedArray_0.18.0       data.table_1.14.0        
 [10] rpart_4.1-15              KEGGREST_1.32.0           RCurl_1.98-1.3           
 [13] doParallel_1.0.16         generics_0.1.0            ScaledMatrix_1.0.0       
 [16] TH.data_1.0-10            cowplot_1.1.1             RSQLite_2.2.7            
 [19] RANN_2.6.1                future_1.21.0             bit_4.0.4                
 [22] tzdb_0.1.2                spatstat.data_2.1-0       xml2_1.3.2               
 [25] lubridate_1.7.10          httpuv_1.6.1              assertthat_0.2.1         
 [28] xfun_0.24                 hms_1.1.0                 jquerylib_0.1.4          
 [31] evaluate_0.14             promises_1.2.0.1          progress_1.2.2           
 [34] fansi_0.5.0               caTools_1.18.2            dbplyr_2.1.1             
 [37] readxl_1.3.1              geneplotter_1.70.0        igraph_1.2.6             
 [40] DBI_1.1.1                 htmlwidgets_1.5.3         spatstat.geom_2.2-2      
 [43] ellipsis_0.3.2            RSpectra_0.16-0           backports_1.2.1          
 [46] annotate_1.70.0           deldir_0.2-10             sparseMatrixStats_1.4.0  
 [49] vctrs_0.3.8               Cairo_1.5-12.2            ROCR_1.0-11              
 [52] abind_1.4-5               cachem_1.0.5              withr_2.4.2              
 [55] emmeans_1.6.2-1           sctransform_0.3.2         prettyunits_1.1.1        
 [58] goftest_1.2-2             cluster_2.1.2             lazyeval_0.2.2           
 [61] crayon_1.4.1              genefilter_1.74.0         labeling_0.4.2           
 [64] pkgconfig_2.0.3           nlme_3.1-152              vipor_0.4.5              
 [67] blme_1.0-5                rlang_0.4.11              globals_0.14.0           
 [70] lifecycle_1.0.0           miniUI_0.1.1.1            sandwich_3.0-1           
 [73] modelr_0.1.8              rsvd_1.0.5                cellranger_1.1.0         
 [76] rprojroot_2.0.2           polyclip_1.10-0           lmtest_0.9-38            
 [79] Matrix_1.3-4              boot_1.3-28               zoo_1.8-9                
 [82] reprex_2.0.0              beeswarm_0.4.0            whisker_0.4              
 [85] ggridges_0.5.3            GlobalOptions_0.1.2       png_0.1-7                
 [88] rjson_0.2.20              bitops_1.0-7              KernSmooth_2.23-20       
 [91] Biostrings_2.60.1         blob_1.2.2                DelayedMatrixStats_1.14.0
 [94] workflowr_1.6.2           shape_1.4.6               parallelly_1.27.0        
 [97] beachmat_2.8.0            scales_1.1.1              memoise_2.0.0            
[100] plyr_1.8.6                ica_1.0-2                 gplots_3.1.1             
[103] zlibbioc_1.38.0           compiler_4.1.0            dqrng_0.3.0              
[106] lme4_1.1-27.1             clue_0.3-59               DESeq2_1.32.0            
[109] fitdistrplus_1.1-5        cli_3.0.1                 XVector_0.32.0           
[112] lmerTest_3.1-3            listenv_0.8.0             TMB_1.7.20               
[115] patchwork_1.1.1           pbapply_1.4-3             MASS_7.3-54              
[118] mgcv_1.8-36               tidyselect_1.1.1          stringi_1.7.3            
[121] highr_0.9                 yaml_2.2.1                BiocSingular_1.8.1       
[124] locfit_1.5-9.4            ggrepel_0.9.1             grid_4.1.0               
[127] sass_0.4.0                tools_4.1.0               future.apply_1.7.0       
[130] circlize_0.4.13           rstudioapi_0.13           bluster_1.2.1            
[133] foreach_1.5.1             git2r_0.28.0              metapod_1.0.0            
[136] gridExtra_2.3             farver_2.1.0              Rtsne_0.15               
[139] digest_0.6.27             shiny_1.6.0               broom_0.7.8              
[142] later_1.2.0               RcppAnnoy_0.0.18          httr_1.4.2               
[145] AnnotationDbi_1.54.1      ComplexHeatmap_2.8.0      colorspace_2.0-2         
[148] rvest_1.0.1               XML_3.99-0.6              fs_1.5.0                 
[151] tensor_1.5                reticulate_1.20           splines_4.1.0            
[154] uwot_0.1.10               statmod_1.4.36            spatstat.utils_2.2-0     
[157] plotly_4.9.4.1            xtable_1.8-4              nloptr_1.2.2.2           
[160] jsonlite_1.7.2            R6_2.5.0                  pillar_1.6.1             
[163] htmltools_0.5.1.1         mime_0.11                 minqa_1.2.4              
[166] glue_1.4.2                fastmap_1.1.0             BiocParallel_1.26.1      
[169] BiocNeighbors_1.10.0      codetools_0.2-18          mvtnorm_1.1-2            
[172] utf8_1.2.2                lattice_0.20-44           bslib_0.2.5.1            
[175] spatstat.sparse_2.0-0     pbkrtest_0.5.1            numDeriv_2016.8-1.1      
[178] ggbeeswarm_0.6.0          colorRamps_2.3            leiden_0.3.8             
[181] gtools_3.9.2              glmmTMB_1.1.2             survival_3.2-11          
[184] rmarkdown_2.9             munsell_0.5.0             GetoptLong_1.0.5         
[187] GenomeInfoDbData_1.2.6    iterators_1.0.13          variancePartition_1.22.0 
[190] haven_2.4.1               reshape2_1.4.4            gtable_0.3.0             
[193] spatstat.core_2.3-0      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
