<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>create input files for cellphonedb</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSLO_BRC_CXCL13</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">create input files for cellphonedb</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-23
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSLO_BRC_CXCL13/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210519code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210519)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210519code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210519)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree525b9f61d4976a8e83ac84bd63f17780d5cd2405targetblank525b9f6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/525b9f61d4976a8e83ac84bd63f17780d5cd2405" target="_blank">525b9f6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree525b9f61d4976a8e83ac84bd63f17780d5cd2405targetblank525b9f6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/525b9f61d4976a8e83ac84bd63f17780d5cd2405" target="_blank">525b9f6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/cellphonedb/
    Ignored:    data/destiny/
    Ignored:    data/lympho/
    Ignored:    data/processedData/
    Ignored:    data/processedDataLymphocytes/
    Ignored:    figure/.DS_Store
    Ignored:    figure/plotSubsetSpecificSignatures.Rmd/.DS_Store

Untracked files:
    Untracked:  analysis/CharacterizeLymphoSubgrps.Rmd
    Untracked:  analysis/SubsetSpecificGeneSignatures2.Rmd
    Untracked:  analysis/plotSubsetSpecificSignatures2.Rmd
    Untracked:  analysis/visSelIntHeatmap.Rmd
    Untracked:  figure/plotSubsetSpecificSignatures2.Rmd/

Unstaged changes:
    Modified:   analysis/SubsetSpecificGeneSignatures.Rmd
    Modified:   analysis/createCPDBinput_immOnly.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Costimulatory and inhibitory molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Costimulatory and inhibitory molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC   -1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC   -1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Metabolite transporter_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Metabolite transporter_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Stemness_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Stemness_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factor_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factor_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2 -1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2 -1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Costimulatory and inhibitory molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Costimulatory and inhibitory molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Metabolite transporter_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Metabolite transporter_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Stemness_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Stemness_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_TBRC-1.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/createCPDBinput_immOnly.Rmd</code>) and HTML (<code>docs/createCPDBinput_immOnly.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/525b9f61d4976a8e83ac84bd63f17780d5cd2405/analysis/createCPDBinput_immOnly.Rmd" target="_blank">525b9f6</a>
</td>
<td>
mluetge
</td>
<td>
2021-11-04
</td>
<td>
cellphone imm only
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/94961678506115e0ed4a809d24ad48c7f426416f/analysis/createCPDBinput_immOnly.Rmd" target="_blank">9496167</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-22
</td>
<td>
add DCs to int analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/94961678506115e0ed4a809d24ad48c7f426416f/docs/createCPDBinput_immOnly.html" target="_blank">9496167</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-22
</td>
<td>
add DCs to int analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<pre class="r"><code>suppressPackageStartupMessages({
  library(scater)
  library(ensembldb)
  library(dplyr)
  library(reshape2)
  library(Matrix)
  library(org.Mm.eg.db)
  library(purrr)
  library(scran)
  library(Seurat)
  library(mclust)
  library(tidyverse)
  library(biomaRt)
  library(here)
})</code></pre>
<div id="load-input-data" class="section level2">
<h2>load input data</h2>
<pre class="r"><code>basedir &lt;- here()
seuratLY &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/lympho/allSamples_mergedLab_seurat.rds&quot;))
seuratBRC &lt;- readRDS(file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))

## Each SLO individually
selLab &lt;- c(&quot;naiveBcells&quot;, &quot;LZBcells&quot;, &quot;DZshmBcells&quot;, &quot;proDZ&quot;,
                   &quot;plasmaCells&quot;, 
                   &quot;Tfh&quot;, &quot;DCs&quot;,
                   &quot;mdMacrophages&quot;, &quot;trMacrophages&quot;, &quot;infMacrophages&quot;)
seuratLY &lt;- subset(seuratLY, clusterLabel %in% selLab)
GCBs &lt;- c(&quot;LZBcells&quot;, &quot;DZshmBcells&quot;, &quot;proDZ&quot;)
Myel &lt;- c(&quot;mdMacrophages&quot;, &quot;trMacrophages&quot;, &quot;infMacrophages&quot;)
seuratLY$clusterLabel2 &lt;- seuratLY$clusterLabel
seuratLY$clusterLabel2[which(seuratLY$clusterLabel %in% GCBs)] &lt;- &quot;GCBcells&quot;
seuratLY$clusterLabel2[which(seuratLY$clusterLabel %in% Myel)] &lt;- &quot;Myeloids&quot;

seuratLY$label_plus_SLO &lt;- paste0(seuratLY$clusterLabel2, &quot;_&quot;, seuratLY$SLO)
seuratLY$clusterLabel &lt;- seuratLY$clusterLabel2

### only imm BRCs
seuratBRC &lt;- subset(seuratBRC, cond == &quot;immunized&quot;)

## plot input data
Idents(seuratLY) &lt;- seuratLY$clusterLabel2
DimPlot(seuratLY, reduction = &quot;umap&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/createCPDBinput_immOnly.Rmd/input%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-input-data-1">
Past versions of “input data-1.png”
</button>
</p>
<div id="fig-no-spaces-input-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/94961678506115e0ed4a809d24ad48c7f426416f/docs/figure/createCPDBinput_immOnly.Rmd/input data-1.png" target="_blank">9496167</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratBRC) &lt;- seuratBRC$clusterLabel
DimPlot(seuratBRC, reduction = &quot;umap&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/createCPDBinput_immOnly.Rmd/input%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-input-data-2">
Past versions of “input data-2.png”
</button>
</p>
<div id="fig-no-spaces-input-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/94961678506115e0ed4a809d24ad48c7f426416f/docs/figure/createCPDBinput_immOnly.Rmd/input data-2.png" target="_blank">9496167</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="filter-cells-and-assign-cluster" class="section level1">
<h1>filter cells and assign cluster</h1>
<pre class="r"><code>seuratBRC$group &lt;- &quot;BRC&quot;
seuratLY$group &lt;- &quot;LYMPHO&quot;

## only cluster XYZ
## seuratLY &lt;- subset(seuratLY, idents=&quot;XYZ&quot;)

## subsample seuratLY
Idents(seuratLY) &lt;- seuratLY$label_plus_SLO
seuratLY &lt;- subset(seuratLY, downsample = 200) ## uncomment to subsample in R script

## subsample seuratBRC
Idents(seuratBRC) &lt;- seuratBRC$SLO_plus_clusterlabel
seuratBRC &lt;- subset(seuratBRC, downsample = 200) ## uncomment to subsample in R script


Idents(seuratLY) &lt;- seuratLY$clusterLabel
Idents(seuratBRC) &lt;- seuratBRC$clusterLabel
  
## merge
### fix assay and gene names
sceBRC &lt;- as.SingleCellExperiment(seuratBRC)
sceLY &lt;- as.SingleCellExperiment(seuratLY)

genesBRC &lt;- data.frame(geneB=rownames(sceBRC)) %&gt;% 
  mutate(ensID=gsub(&quot;\\..*&quot;,&quot;&quot;,geneB))
genesLY &lt;- data.frame(geneL=rownames(sceLY)) %&gt;% 
  mutate(ensID=gsub(&quot;\\..*&quot;,&quot;&quot;,geneL))
genesBoth &lt;- full_join(genesBRC, genesLY, by=&quot;ensID&quot;) %&gt;% 
  mutate(gene=ifelse(is.na(geneL), geneB, geneL)) %&gt;% 
  dplyr::select(ensID, gene)
genesBRC &lt;- genesBRC %&gt;% left_join(., genesBoth, by=&quot;ensID&quot;)
genesLY &lt;- genesLY %&gt;% left_join(., genesBoth, by=&quot;ensID&quot;)

rownames(sceBRC) &lt;- genesBRC$gene
rownames(sceLY) &lt;- genesLY$gene

seuratBRC &lt;- as.Seurat(sceBRC)
seuratLY &lt;- as.Seurat(sceLY)

seuratBRC@assays$RNA &lt;- seuratBRC@assays$originalexp
seuratLY@assays$originalexp &lt;- seuratLY@assays$RNA

seuratMerge &lt;- merge(seuratBRC, seuratLY)
seuratMerge &lt;- ScaleData(seuratMerge, assay = &quot;RNA&quot;)
seuratMerge &lt;- ScaleData(seuratMerge, assay = &quot;originalexp&quot;)

remove(seuratBRC)
remove(seuratLY)

sceMerge &lt;- as.SingleCellExperiment(seuratMerge)
(table(sceMerge@colData$group))</code></pre>
<pre><code>
   BRC LYMPHO 
  1883   3423 </code></pre>
<pre class="r"><code>## assign celltype
colData(sceMerge)$clusterFinal &lt;- paste0(colData(sceMerge)$group, &quot;_&quot;, 
                                         colData(sceMerge)$clusterLabel)
(table(sceMerge@colData$clusterFinal))</code></pre>
<pre><code>
         BRC_DZFDC          BRC_LZFDC            BRC_MRC           BRC_PRC2           BRC_TBRC 
               263                494                202                324                600 
        LYMPHO_DCs    LYMPHO_GCBcells    LYMPHO_Myeloids LYMPHO_naiveBcells LYMPHO_plasmaCells 
               455                600                577                600                591 
        LYMPHO_Tfh 
               600 </code></pre>
</div>
<div id="create-count-matrix-for-cellphonedb" class="section level1">
<h1>create count matrix for cellphonedb</h1>
<pre class="r"><code>## set human ensIDs as rownames
ensIDs &lt;- gsub(&quot;\\..*&quot;,&quot;&quot;,rownames(sceMerge))

## map to human orthologs
mart2 &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
mart1 = useMart(&quot;ensembl&quot;, dataset=&quot;mmusculus_gene_ensembl&quot;) 

# human / mouse
geneIDS_hum &lt;- getLDS(attributes=c(&quot;ensembl_gene_id&quot;),
           filters=&quot;ensembl_gene_id&quot;, values=ensIDs, mart=mart1,
           attributesL=c(&quot;ensembl_gene_id&quot;), martL=mart2)

geneIDsMatch &lt;- as.data.frame(ensIDs) %&gt;% 
  left_join(., geneIDS_hum, by=c(&quot;ensIDs&quot;=&quot;Gene.stable.ID&quot;)) 
geneIDsMatch$Gene.stable.ID.1[
  which(is.na(geneIDsMatch$Gene.stable.ID.1))] &lt;- &quot;unknown&quot;
geneIDsMatch &lt;- geneIDsMatch[!duplicated(geneIDsMatch$ensIDs),]
rowData(sceMerge)$humGene &lt;- geneIDsMatch$Gene.stable.ID.1
dim(sceMerge)</code></pre>
<pre><code>[1] 31474  5306</code></pre>
<pre class="r"><code>sceMerge &lt;- sceMerge[which(rowData(sceMerge)$humGene != &quot;unknown&quot;),] 
dim(sceMerge)</code></pre>
<pre><code>[1] 16695  5306</code></pre>
<pre class="r"><code>## filter for SLO and write cnt mat plus metadat
lapply(unique(sceMerge$SLO), function(slo){
  sceSLO &lt;- sceMerge[,which(sceMerge@colData$SLO == slo)]
  cellPhoneSLO &lt;- cbind(rowData(sceSLO)$humGene, as.matrix(logcounts(sceSLO)))
  colnames(cellPhoneSLO)[1] &lt;- &quot;gene&quot;
  write.table(cellPhoneSLO, 
              file = paste0(basedir,
                            &quot;/data/cellphonedb/folBRCimm_LYMGrpsub_&quot;,slo,
                            &quot;_countMatrix.txt&quot;),
              row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)

  metaDatSLO &lt;- as.data.frame(cbind(colnames(sceSLO),
                                    colData(sceSLO)$clusterFinal))
  colnames(metaDatSLO) &lt;- c(&quot;Cell&quot;, &quot;cell_type&quot;)
  write.table(metaDatSLO, 
              file = paste0(basedir,
                            &quot;/data/cellphonedb/folBRCimm_LYMGrpsub_&quot;,slo,
                            &quot;_metaData.txt&quot;),
              row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)


})</code></pre>
<pre><code>[[1]]
NULL

[[2]]
NULL

[[3]]
NULL</code></pre>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1                  biomaRt_2.48.3              forcats_0.5.1              
 [4] stringr_1.4.0               readr_2.0.2                 tidyr_1.1.4                
 [7] tibble_3.1.6                tidyverse_1.3.1             mclust_5.4.8               
[10] SeuratObject_4.0.2          Seurat_4.0.5                scran_1.20.1               
[13] purrr_0.3.4                 org.Mm.eg.db_3.13.0         Matrix_1.3-4               
[16] reshape2_1.4.4              dplyr_1.0.7                 ensembldb_2.16.4           
[19] AnnotationFilter_1.16.0     GenomicFeatures_1.44.2      AnnotationDbi_1.54.1       
[22] scater_1.20.1               ggplot2_3.3.5               scuttle_1.2.1              
[25] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0 Biobase_2.52.0             
[28] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4         IRanges_2.26.0             
[31] S4Vectors_0.30.2            BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[34] matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            rtracklayer_1.52.1        scattermore_0.7          
  [4] bit64_4.0.5               knitr_1.36                irlba_2.3.3              
  [7] DelayedArray_0.18.0       data.table_1.14.2         rpart_4.1-15             
 [10] KEGGREST_1.32.0           RCurl_1.98-1.5            generics_0.1.1           
 [13] ScaledMatrix_1.0.0        cowplot_1.1.1             RSQLite_2.2.8            
 [16] RANN_2.6.1                future_1.23.0             bit_4.0.4                
 [19] tzdb_0.2.0                spatstat.data_2.1-0       xml2_1.3.2               
 [22] lubridate_1.8.0           httpuv_1.6.3              assertthat_0.2.1         
 [25] viridis_0.6.2             xfun_0.28                 hms_1.1.1                
 [28] jquerylib_0.1.4           evaluate_0.14             promises_1.2.0.1         
 [31] fansi_0.5.0               restfulr_0.0.13           progress_1.2.2           
 [34] dbplyr_2.1.1              readxl_1.3.1              igraph_1.2.8             
 [37] DBI_1.1.1                 htmlwidgets_1.5.4         spatstat.geom_2.3-0      
 [40] ellipsis_0.3.2            backports_1.3.0           deldir_1.0-6             
 [43] sparseMatrixStats_1.4.2   vctrs_0.3.8               ROCR_1.0-11              
 [46] abind_1.4-5               cachem_1.0.6              withr_2.4.2              
 [49] sctransform_0.3.2         GenomicAlignments_1.28.0  prettyunits_1.1.1        
 [52] goftest_1.2-3             cluster_2.1.2             lazyeval_0.2.2           
 [55] crayon_1.4.2              labeling_0.4.2            edgeR_3.34.1             
 [58] pkgconfig_2.0.3           nlme_3.1-153              vipor_0.4.5              
 [61] ProtGenerics_1.24.0       rlang_0.4.12              globals_0.14.0           
 [64] lifecycle_1.0.1           miniUI_0.1.1.1            filelock_1.0.2           
 [67] BiocFileCache_2.0.0       modelr_0.1.8              rsvd_1.0.5               
 [70] cellranger_1.1.0          rprojroot_2.0.2           polyclip_1.10-0          
 [73] lmtest_0.9-39             zoo_1.8-9                 reprex_2.0.1             
 [76] beeswarm_0.4.0            whisker_0.4               ggridges_0.5.3           
 [79] png_0.1-7                 viridisLite_0.4.0         rjson_0.2.20             
 [82] bitops_1.0-7              KernSmooth_2.23-20        Biostrings_2.60.2        
 [85] blob_1.2.2                DelayedMatrixStats_1.14.3 workflowr_1.6.2          
 [88] parallelly_1.28.1         beachmat_2.8.1            scales_1.1.1             
 [91] memoise_2.0.0             magrittr_2.0.1            plyr_1.8.6               
 [94] ica_1.0-2                 zlibbioc_1.38.0           compiler_4.1.0           
 [97] dqrng_0.3.0               BiocIO_1.2.0              RColorBrewer_1.1-2       
[100] fitdistrplus_1.1-6        cli_3.1.0                 Rsamtools_2.8.0          
[103] XVector_0.32.0            listenv_0.8.0             patchwork_1.1.1          
[106] pbapply_1.5-0             MASS_7.3-54               mgcv_1.8-38              
[109] tidyselect_1.1.1          stringi_1.7.5             highr_0.9                
[112] yaml_2.2.1                BiocSingular_1.8.1        locfit_1.5-9.4           
[115] ggrepel_0.9.1             grid_4.1.0                sass_0.4.0               
[118] tools_4.1.0               future.apply_1.8.1        rstudioapi_0.13          
[121] bluster_1.2.1             git2r_0.28.0              metapod_1.0.0            
[124] gridExtra_2.3             farver_2.1.0              Rtsne_0.15               
[127] digest_0.6.28             shiny_1.7.1               Rcpp_1.0.7               
[130] broom_0.7.10              later_1.3.0               RcppAnnoy_0.0.19         
[133] httr_1.4.2                colorspace_2.0-2          rvest_1.0.2              
[136] XML_3.99-0.8              fs_1.5.0                  tensor_1.5               
[139] reticulate_1.22           splines_4.1.0             uwot_0.1.10              
[142] statmod_1.4.36            spatstat.utils_2.2-0      plotly_4.10.0            
[145] xtable_1.8-4              jsonlite_1.7.2            R6_2.5.1                 
[148] pillar_1.6.4              htmltools_0.5.2           mime_0.12                
[151] glue_1.5.0                fastmap_1.1.0             BiocParallel_1.26.2      
[154] BiocNeighbors_1.10.0      codetools_0.2-18          utf8_1.2.2               
[157] lattice_0.20-45           bslib_0.3.1               spatstat.sparse_2.0-0    
[160] curl_4.3.2                ggbeeswarm_0.6.0          leiden_0.3.9             
[163] survival_3.2-13           limma_3.48.3              rmarkdown_2.11           
[166] munsell_0.5.0             GenomeInfoDbData_1.2.6    haven_2.4.3              
[169] gtable_0.3.0              spatstat.core_2.3-1      </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Nov 23 09:54:41 2021&quot;</code></pre>
<p>cd /Users/mechthildluetge/Projects/scRNAseq/ILC_Hungwei/data/cellphonedb eval “<span class="math inline">\((pyenv init -)&quot; eval &quot;\)</span>(pyenv virtualenv-init -)”</p>
</div>
<div id="run-cellphonedb" class="section level2">
<h2>run cellphonedb</h2>
<div id="build-database" class="section level3">
<h3>build database</h3>
<p>cellphonedb database generate –user-interactions custom_interaction_file.csv</p>
<p>##run statistical analysis ## subsampling in R script.. cellphonedb method statistical_analysis /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_LN_metaData.txt /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_LN_countMatrix.txt –output-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outLNimm/ –database /Users/mechthildluetge/Projects/scRNAseq/ILC_Hungwei/data/cellphonedb/out/cellphonedb_user_2021-10-05-09_17.db –debug-seed 3 –iterations 100</p>
<p>cellphonedb method statistical_analysis /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_PP_metaData.txt /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_PP_countMatrix.txt –output-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outPPimm/ –database /Users/mechthildluetge/Projects/scRNAseq/ILC_Hungwei/data/cellphonedb/out/cellphonedb_user_2021-10-05-09_17.db –debug-seed 3 –iterations 100</p>
<p>cellphonedb method statistical_analysis /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_Spleen_metaData.txt /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_Spleen_countMatrix.txt –output-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outSPimm/ –database /Users/mechthildluetge/Projects/scRNAseq/ILC_Hungwei/data/cellphonedb/out/cellphonedb_user_2021-10-05-09_17.db –debug-seed 3 –iterations 100</p>
<p>Count network interactions: cellphonedb plot heatmap_plot /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_LN_metaData.txt –pvalues-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outLNimm/pvalues.txt –output-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outLNimm cellphonedb plot heatmap_plot /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_PP_metaData.txt –pvalues-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outPPimm/pvalues.txt –output-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outPPimm cellphonedb plot heatmap_plot /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/folBRCimm_LYMsub_Spleen_metaData.txt –pvalues-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outSPimm/pvalues.txt –output-path /Users/mechthildluetge/Projects/scRNAseq/CrossSLO_BRC_CXCL13/data/cellphonedb/outSPimm</p>
<p>We use CellPhoneDB as tool to predict cellular communication between SILT FRCs and ILC populations in our dataset. In brief, pairs of known interacting ligands and receptors are derived from public databases and cell types are analysed for enriched receptor–ligand interactions based on expression of a receptor by one cell type and a ligand by another cell type.</p>
<p>deactivate</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1                  biomaRt_2.48.3              forcats_0.5.1              
 [4] stringr_1.4.0               readr_2.0.2                 tidyr_1.1.4                
 [7] tibble_3.1.6                tidyverse_1.3.1             mclust_5.4.8               
[10] SeuratObject_4.0.2          Seurat_4.0.5                scran_1.20.1               
[13] purrr_0.3.4                 org.Mm.eg.db_3.13.0         Matrix_1.3-4               
[16] reshape2_1.4.4              dplyr_1.0.7                 ensembldb_2.16.4           
[19] AnnotationFilter_1.16.0     GenomicFeatures_1.44.2      AnnotationDbi_1.54.1       
[22] scater_1.20.1               ggplot2_3.3.5               scuttle_1.2.1              
[25] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0 Biobase_2.52.0             
[28] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4         IRanges_2.26.0             
[31] S4Vectors_0.30.2            BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[34] matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3            rtracklayer_1.52.1        scattermore_0.7          
  [4] bit64_4.0.5               knitr_1.36                irlba_2.3.3              
  [7] DelayedArray_0.18.0       data.table_1.14.2         rpart_4.1-15             
 [10] KEGGREST_1.32.0           RCurl_1.98-1.5            generics_0.1.1           
 [13] ScaledMatrix_1.0.0        cowplot_1.1.1             RSQLite_2.2.8            
 [16] RANN_2.6.1                future_1.23.0             bit_4.0.4                
 [19] tzdb_0.2.0                spatstat.data_2.1-0       xml2_1.3.2               
 [22] lubridate_1.8.0           httpuv_1.6.3              assertthat_0.2.1         
 [25] viridis_0.6.2             xfun_0.28                 hms_1.1.1                
 [28] jquerylib_0.1.4           evaluate_0.14             promises_1.2.0.1         
 [31] fansi_0.5.0               restfulr_0.0.13           progress_1.2.2           
 [34] dbplyr_2.1.1              readxl_1.3.1              igraph_1.2.8             
 [37] DBI_1.1.1                 htmlwidgets_1.5.4         spatstat.geom_2.3-0      
 [40] ellipsis_0.3.2            backports_1.3.0           deldir_1.0-6             
 [43] sparseMatrixStats_1.4.2   vctrs_0.3.8               ROCR_1.0-11              
 [46] abind_1.4-5               cachem_1.0.6              withr_2.4.2              
 [49] sctransform_0.3.2         GenomicAlignments_1.28.0  prettyunits_1.1.1        
 [52] goftest_1.2-3             cluster_2.1.2             lazyeval_0.2.2           
 [55] crayon_1.4.2              labeling_0.4.2            edgeR_3.34.1             
 [58] pkgconfig_2.0.3           nlme_3.1-153              vipor_0.4.5              
 [61] ProtGenerics_1.24.0       rlang_0.4.12              globals_0.14.0           
 [64] lifecycle_1.0.1           miniUI_0.1.1.1            filelock_1.0.2           
 [67] BiocFileCache_2.0.0       modelr_0.1.8              rsvd_1.0.5               
 [70] cellranger_1.1.0          rprojroot_2.0.2           polyclip_1.10-0          
 [73] lmtest_0.9-39             zoo_1.8-9                 reprex_2.0.1             
 [76] beeswarm_0.4.0            whisker_0.4               ggridges_0.5.3           
 [79] png_0.1-7                 viridisLite_0.4.0         rjson_0.2.20             
 [82] bitops_1.0-7              KernSmooth_2.23-20        Biostrings_2.60.2        
 [85] blob_1.2.2                DelayedMatrixStats_1.14.3 workflowr_1.6.2          
 [88] parallelly_1.28.1         beachmat_2.8.1            scales_1.1.1             
 [91] memoise_2.0.0             magrittr_2.0.1            plyr_1.8.6               
 [94] ica_1.0-2                 zlibbioc_1.38.0           compiler_4.1.0           
 [97] dqrng_0.3.0               BiocIO_1.2.0              RColorBrewer_1.1-2       
[100] fitdistrplus_1.1-6        cli_3.1.0                 Rsamtools_2.8.0          
[103] XVector_0.32.0            listenv_0.8.0             patchwork_1.1.1          
[106] pbapply_1.5-0             MASS_7.3-54               mgcv_1.8-38              
[109] tidyselect_1.1.1          stringi_1.7.5             highr_0.9                
[112] yaml_2.2.1                BiocSingular_1.8.1        locfit_1.5-9.4           
[115] ggrepel_0.9.1             grid_4.1.0                sass_0.4.0               
[118] tools_4.1.0               future.apply_1.8.1        rstudioapi_0.13          
[121] bluster_1.2.1             git2r_0.28.0              metapod_1.0.0            
[124] gridExtra_2.3             farver_2.1.0              Rtsne_0.15               
[127] digest_0.6.28             shiny_1.7.1               Rcpp_1.0.7               
[130] broom_0.7.10              later_1.3.0               RcppAnnoy_0.0.19         
[133] httr_1.4.2                colorspace_2.0-2          rvest_1.0.2              
[136] XML_3.99-0.8              fs_1.5.0                  tensor_1.5               
[139] reticulate_1.22           splines_4.1.0             uwot_0.1.10              
[142] statmod_1.4.36            spatstat.utils_2.2-0      plotly_4.10.0            
[145] xtable_1.8-4              jsonlite_1.7.2            R6_2.5.1                 
[148] pillar_1.6.4              htmltools_0.5.2           mime_0.12                
[151] glue_1.5.0                fastmap_1.1.0             BiocParallel_1.26.2      
[154] BiocNeighbors_1.10.0      codetools_0.2-18          utf8_1.2.2               
[157] lattice_0.20-45           bslib_0.3.1               spatstat.sparse_2.0-0    
[160] curl_4.3.2                ggbeeswarm_0.6.0          leiden_0.3.9             
[163] survival_3.2-13           limma_3.48.3              rmarkdown_2.11           
[166] munsell_0.5.0             GenomeInfoDbData_1.2.6    haven_2.4.3              
[169] gtable_0.3.0              spatstat.core_2.3-1      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
