<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-05-14" />

<title>organ specific signatures - BRCs only</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSLO_BRC_CXCL13</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">organ specific signatures - BRCs only</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">14 May 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSLO_BRC_CXCL13/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210519code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210519)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210519code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210519)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree525b9f61d4976a8e83ac84bd63f17780d5cd2405targetblank525b9f6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/525b9f61d4976a8e83ac84bd63f17780d5cd2405" target="_blank">525b9f6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree525b9f61d4976a8e83ac84bd63f17780d5cd2405targetblank525b9f6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/525b9f61d4976a8e83ac84bd63f17780d5cd2405" target="_blank">525b9f6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/cellphonedb/
    Ignored:    data/destiny/
    Ignored:    data/lympho/
    Ignored:    data/processedData/
    Ignored:    data/processedDataLymphocytes/
    Ignored:    figure/.DS_Store
    Ignored:    figure/plotSubsetSpecificSignatures.Rmd/.DS_Store

Untracked files:
    Untracked:  analysis/CharacterizeLymphoSubgrps.Rmd
    Untracked:  analysis/SubsetSpecificGeneSignatures2.Rmd
    Untracked:  analysis/SubsetSpecificGeneSignaturesSignaling.Rmd
    Untracked:  analysis/analyseCPDBoutImmGrp.Rmd
    Untracked:  analysis/plotSubsetSpecificSignatures2.Rmd
    Untracked:  analysis/plotSubsetSpecificSignaturesSignaling.Rmd
    Untracked:  analysis/visSelIntHeatmap.Rmd
    Untracked:  figure/plotSubsetSpecificSignatures2.Rmd/

Unstaged changes:
    Modified:   analysis/SubsetSpecificGeneSignatures.Rmd
    Modified:   analysis/analyseCPDBoutImm.Rmd
    Modified:   analysis/createCPDBinput_immOnly.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Antigen capture and presentation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Costimulatory and inhibitory molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Costimulatory and inhibitory molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC   -1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC   -1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Cytokines and chemokines_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Antigen capture and presentation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Costimulatory and inhibitory molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Cytokines and chemokines_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Metabolite transporter_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Stemness_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM Tnf superfamily factors_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM adhesion molecules_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM growth and survival factors_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2 cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2 cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC cut 2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr DM network formation_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Metabolite transporter_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Metabolite transporter_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Stemness_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Stemness_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr Tnf superfamily factors_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr adhesion molecules_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factor_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factor_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr growth and survival factors_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2 -1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2 -1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig expr network formation_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Antigen capture and presentation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Costimulatory and inhibitory molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Costimulatory and inhibitory molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Cytokines and chemokines_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Metabolite transporter_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Metabolite transporter_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Stemness_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Stemness_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm Tnf superfamily factors_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm adhesion molecules_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm growth and survival factors_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_PRC2-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_PRC2-1.png
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignatures.Rmd/sig hm network formation_TBRC-1.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(SingleCellExperiment)
  library(RColorBrewer)
  library(viridis)
  library(ggsci)
  library(scater)
  library(scran)
  library(pheatmap)
  library(biomaRt)
  library(fgsea)
  library(grid)
  library(gridExtra)
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  library(enrichplot)
  library(msigdbr)
  library(muscat)
})</code></pre>
</div>
<div id="plotting-funct" class="section level2">
<h2>plotting funct</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  ## format gene names (depends on how gene list is inputed)
  selGenes &lt;- selGenes %&gt;% 
      mutate(geneID = (str_split(gene, &#39;\\.&#39;, simplify = T)[,2]))
  #selGenes &lt;- selGenes$labelNam
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;%
    filter(geneID %in% selGenes$geneID)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes$geneID,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="define-plotting-function-dm" class="section level2">
<h2>define plotting function DM</h2>
<pre class="r"><code>## adapted from CellMixS
visGroup_adapt &lt;- function (sce,group,dim_red = &quot;TSNE&quot;,col_group=pal_nejm()(8)) 
{
    if (!is(sce, &quot;SingleCellExperiment&quot;)) {
        stop(&quot;Error:&#39;sce&#39; must be a &#39;SingleCellExperiment&#39; object.&quot;)
    }
    if (!group %in% names(colData(sce))) {
        stop(&quot;Error: &#39;group&#39; variable must be in &#39;colData(sce)&#39;&quot;)
    }
    cell_names &lt;- colnames(sce)
    if (!dim_red %in% &quot;TSNE&quot;) {
        if (!dim_red %in% reducedDimNames(sce)) {
            stop(&quot;Please provide a dim_red method listed in reducedDims of sce&quot;)
        }
        red_dim &lt;- as.data.frame(reducedDim(sce, dim_red))
    }
    else {
        if (!&quot;TSNE&quot; %in% reducedDimNames(sce)) {
            if (&quot;logcounts&quot; %in% names(assays(sce))) {
                sce &lt;- runTSNE(sce)
            }
            else {
                sce &lt;- runTSNE(sce, exprs_values = &quot;counts&quot;)
            }
        }
        red_dim &lt;- as.data.frame(reducedDim(sce, &quot;TSNE&quot;))
    }
    colnames(red_dim) &lt;- c(&quot;red_dim1&quot;, &quot;red_dim2&quot;)
    df &lt;- data.frame(sample_id = cell_names, group_var = colData(sce)[, 
        group], red_Dim1 = red_dim$red_dim1, red_Dim2 = red_dim$red_dim2)
    t &lt;- ggplot(df, aes_string(x = &quot;red_Dim1&quot;, y = &quot;red_Dim2&quot;)) + 
        xlab(paste0(dim_red, &quot;_1&quot;)) + ylab(paste0(dim_red, &quot;_2&quot;)) + 
        theme_void() + theme(aspect.ratio = 1,
                             panel.grid.minor = element_blank(), 
        panel.grid.major = element_line(color = &quot;grey&quot;, size = 0.3))
    t_group &lt;- t + geom_point(size = 3, alpha = 0.5,
                              aes_string(color = &quot;group_var&quot;)) + 
        guides(color = guide_legend(override.aes = list(size = 1), 
            title = group)) + ggtitle(group)
    if (is.numeric(df$group_var)) {
        t_group &lt;- t_group + scale_color_viridis(option = &quot;C&quot;)
    }
    else {
        t_group &lt;- t_group + scale_color_manual(values = col_group)
    }
    t_group
}</code></pre>
</div>
<div id="set-dir-and-read-input-data" class="section level2">
<h2>set dir and read input data</h2>
<pre class="r"><code>basedir &lt;- here()

## read seurat objects with BRCs
seurat &lt;- readRDS(file=paste0(basedir, 
                              &quot;/data/folBRC_allSLO_subSampled_seurat.rds&quot;))
## seurat &lt;-rerunSeurat3(seurat)
## saveRDS(seurat, file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color Vectors</h2>
<pre class="r"><code>colPal &lt;- c(&quot;#87c5af&quot;, &quot;#94033b&quot;, &quot;#fc8f3b&quot;, &quot;#2580fe&quot;, &quot;#514e72&quot;)
names(colPal) &lt;- c(&quot;LZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;DZFDC&quot;,&quot;PRC2&quot;)


colorSLO &lt;- c(viridis(3)[c(1,2)], brewer.pal(n=9, name=&quot;YlOrRd&quot;)[c(4)])
colorCond &lt;- brewer.pal(n=3, name=&quot;Paired&quot;)[c(1,2)]
colorSLOCond &lt;- c(&quot;#440154FF&quot;,&quot;#807DBA&quot;, &quot;#FEB24C&quot;, &quot;#F16913&quot;, &quot;#21908CFF&quot;)
colorBatch &lt;- pal_igv()(n=length(unique(seurat$batch)))
colCl &lt;- c(rcartocolor::carto_pal(name=&quot;Safe&quot;),pal_aaas()(8))

names(colorSLO) &lt;- c(&quot;LN&quot;, &quot;PP&quot;, &quot;Spleen&quot;)
names(colorCond) &lt;- c(&quot;naive&quot;, &quot;immunized&quot;)
names(colorBatch) &lt;- unique(seurat$batch)
names(colorSLOCond) &lt;- c(&quot;LN_immunized&quot;,&quot;LN_naive&quot;, &quot;Spleen_naive&quot;,
                         &quot;Spleen_immunized&quot;, &quot;PP_immunized&quot;)</code></pre>
</div>
<div id="visualize-data" class="section level2 tabset">
<h2 class="tabset">visualize data</h2>
<div id="clustering" class="section level3">
<h3>clustering</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, group.by = &quot;clusterLabel&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, group.by = &quot;clusterLabel&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/clustering-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="slo" class="section level3">
<h3>SLO</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLO, group.by = &quot;SLO&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/SLO-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLO, group.by = &quot;SLO&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/SLO-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="batch" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorBatch, group.by = &quot;batch&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/Batch-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cond" class="section level3">
<h3>Cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorCond, group.by = &quot;cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/Cond-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="slo-plus-cond" class="section level3">
<h3>SLO plus cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLOCond,group.by=&quot;SLO_plus_cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/SLO%20plus%20cond-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,cols=colorSLOCond,group.by = &quot;SLO_plus_cond&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/SLO%20plus%20cond-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="vis-subset-specific-gene-signatures" class="section level2">
<h2>vis subset specific gene signatures</h2>
<div id="load-signatures" class="section level3">
<h3>load signatures</h3>
<pre class="r"><code>signDat &lt;- read_delim(file = paste0(basedir,
                    &quot;/data/GSEA/SubsetSpecificSelectedGenesSignaling.txt&quot;),
                    delim = &quot;\t&quot;) %&gt;%
  mutate(signSubset=paste0(signature, &quot;_&quot;, subset))
genes &lt;- data.frame(geneID=rownames(seurat)) %&gt;% 
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, geneID))
signDat &lt;- signDat %&gt;% left_join(.,genes, by=&quot;gene&quot;)</code></pre>
</div>
<div id="project-sign-umap-cut-3" class="section level3 tabset">
<h3 class="tabset">project sign umap cut 3</h3>
<pre class="r"><code>allSign &lt;- unique(signDat$signSubset)
sce &lt;- as.SingleCellExperiment(seurat)

cutOff &lt;- 3
pal = colorRampPalette(c(&quot;#053061&quot;, &quot;#2166ac&quot;, &quot;#f7f7f7&quot;,
                         &quot;#f4a582&quot;, &quot;#b2183c&quot;, &quot;#85122d&quot;))(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

template_gs &lt;- c(
    &quot;#### {{sig}} cut3\n&quot;,
    &quot;```{r sig expr cut3 {{sig}}, fig.height=6, fig.width=8}\n&quot;,
    &quot;sign &lt;- as.character(&#39;{{sig}}&#39;)&quot;,
    &quot;signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)&quot;,
    &quot;sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]&quot;,
    &quot;cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)&quot;,
    &quot;sceSub$sign &lt;- cntMat&quot;,
    &quot;sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff&quot;,
    &quot;p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)&quot;,
    &quot;p\n&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_gs &lt;- lapply(allSign, 
  function(sig) knitr::knit_expand(text = template_gs)
)</code></pre>
<div id="regulation-of-lipid-biosynthetic-process_lzfdc-cut3" class="section level4">
<h4>regulation of lipid biosynthetic process_LZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of lipid biosynthetic process_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of lipid biosynthetic process_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="i-kappab-kinasenf-kappab-signaling_lzfdc-cut3" class="section level4">
<h4>I-kappaB kinase/NF-kappaB signaling_LZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;I-kappaB kinase/NF-kappaB signaling_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 I-kappaB kinase/NF-kappaB signaling_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-transforming-growth-factor-beta_lzfdc-cut3" class="section level4">
<h4>response to transforming growth factor beta_LZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to transforming growth factor beta_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 response to transforming growth factor beta_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="fatty-acid-metabolic-process_dzfdc-cut3" class="section level4">
<h4>fatty acid metabolic process_DZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;fatty acid metabolic process_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 fatty acid metabolic process_DZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="gliogenesis_dzfdc-cut3" class="section level4">
<h4>gliogenesis_DZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;gliogenesis_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 gliogenesis_DZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cellular-response-to-fibroblast-growth-factor-stimulus_mrc-cut3" class="section level4">
<h4>cellular response to fibroblast growth factor stimulus_MRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;cellular response to fibroblast growth factor stimulus_MRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 cellular response to fibroblast growth factor stimulus_MRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_tbrc-cut3" class="section level4">
<h4>regulation of angiogenesis_TBRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of angiogenesis_TBRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-interferon-alpha_tbrc-cut3" class="section level4">
<h4>response to interferon-alpha_TBRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to interferon-alpha_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 response to interferon-alpha_TBRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="organ-growth_prc-cut3" class="section level4">
<h4>organ growth_PRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;organ growth_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 organ growth_PRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_prc-cut3" class="section level4">
<h4>regulation of angiogenesis_PRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;UMAP&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of angiogenesis_PRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="project-sign-dm-cut-3" class="section level3 tabset">
<h3 class="tabset">project sign DM cut 3</h3>
<pre class="r"><code>seurat &lt;- readRDS(file=paste0(basedir, 
                              &quot;/data/folBRC_allSLO_seurat.rds&quot;))

sce &lt;- as.SingleCellExperiment(seurat)
sce &lt;- runDiffusionMap(sce)

p &lt;- visGroup_adapt(sce, &#39;SLO&#39;, dim_red = &#39;DiffusionMap&#39;) +
    scale_colour_manual(values = colorSLO) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/project%20sign%20DM%20cut3-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- visGroup_adapt(sce, &#39;clusterLabel&#39;, dim_red = &#39;DiffusionMap&#39;) +
    scale_colour_manual(values = colPal) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/project%20sign%20DM%20cut3-2.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>template_gs &lt;- c(
    &quot;#### {{sig}} cut3\n&quot;,
    &quot;```{r sig expr DM cut3 {{sig}}, fig.height=6, fig.width=8}\n&quot;,
    &quot;sign &lt;- as.character(&#39;{{sig}}&#39;)&quot;,
    &quot;signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)&quot;,
    &quot;sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]&quot;,
    &quot;cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)&quot;,
    &quot;sceSub$sign &lt;- cntMat&quot;,
    &quot;sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff&quot;,
    &quot;p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)&quot;,
    &quot;p\n&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_gs &lt;- lapply(allSign, 
  function(sig) knitr::knit_expand(text = template_gs)
)</code></pre>
<div id="regulation-of-lipid-biosynthetic-process_lzfdc-cut3-1" class="section level4">
<h4>regulation of lipid biosynthetic process_LZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of lipid biosynthetic process_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of lipid biosynthetic process_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="i-kappab-kinasenf-kappab-signaling_lzfdc-cut3-1" class="section level4">
<h4>I-kappaB kinase/NF-kappaB signaling_LZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;I-kappaB kinase/NF-kappaB signaling_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 I-kappaB kinase/NF-kappaB signaling_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-transforming-growth-factor-beta_lzfdc-cut3-1" class="section level4">
<h4>response to transforming growth factor beta_LZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to transforming growth factor beta_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 response to transforming growth factor beta_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="fatty-acid-metabolic-process_dzfdc-cut3-1" class="section level4">
<h4>fatty acid metabolic process_DZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;fatty acid metabolic process_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 fatty acid metabolic process_DZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="gliogenesis_dzfdc-cut3-1" class="section level4">
<h4>gliogenesis_DZFDC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;gliogenesis_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 gliogenesis_DZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cellular-response-to-fibroblast-growth-factor-stimulus_mrc-cut3-1" class="section level4">
<h4>cellular response to fibroblast growth factor stimulus_MRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;cellular response to fibroblast growth factor stimulus_MRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 cellular response to fibroblast growth factor stimulus_MRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_tbrc-cut3-1" class="section level4">
<h4>regulation of angiogenesis_TBRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of angiogenesis_TBRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-interferon-alpha_tbrc-cut3-1" class="section level4">
<h4>response to interferon-alpha_TBRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to interferon-alpha_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 response to interferon-alpha_TBRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="organ-growth_prc-cut3-1" class="section level4">
<h4>organ growth_PRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;organ growth_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 organ growth_PRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_prc-cut3-1" class="section level4">
<h4>regulation of angiogenesis_PRC cut3</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of angiogenesis_PRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="project-sign-dm-cut-2" class="section level3 tabset">
<h3 class="tabset">project sign DM cut 2</h3>
<pre class="r"><code>cutOff &lt;- 2
pal = colorRampPalette(c(&quot;#053061&quot;, &quot;#2166ac&quot;,&quot;#f7f7f7&quot;,
                         &quot;#f4a582&quot;, &quot;#b2183c&quot;, &quot;#85122d&quot;))(100)
sc &lt;- scale_colour_gradientn(colours = pal, limits=c(0, cutOff))

template_gs &lt;- c(
    &quot;#### {{sig}} cut2\n&quot;,
    &quot;```{r sig expr DM cut2 {{sig}} cut 2, fig.height=6, fig.width=8}\n&quot;,
    &quot;sign &lt;- as.character(&#39;{{sig}}&#39;)&quot;,
    &quot;signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)&quot;,
    &quot;sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]&quot;,
    &quot;cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)&quot;,
    &quot;sceSub$sign &lt;- cntMat&quot;,
    &quot;sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff&quot;,
    &quot;p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)&quot;,
    &quot;p\n&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_gs &lt;- lapply(allSign, 
  function(sig) knitr::knit_expand(text = template_gs)
)</code></pre>
<div id="regulation-of-lipid-biosynthetic-process_lzfdc-cut2" class="section level4">
<h4>regulation of lipid biosynthetic process_LZFDC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of lipid biosynthetic process_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of lipid biosynthetic process_LZFDC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="i-kappab-kinasenf-kappab-signaling_lzfdc-cut2" class="section level4">
<h4>I-kappaB kinase/NF-kappaB signaling_LZFDC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;I-kappaB kinase/NF-kappaB signaling_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 I-kappaB kinase/NF-kappaB signaling_LZFDC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-transforming-growth-factor-beta_lzfdc-cut2" class="section level4">
<h4>response to transforming growth factor beta_LZFDC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to transforming growth factor beta_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 response to transforming growth factor beta_LZFDC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="fatty-acid-metabolic-process_dzfdc-cut2" class="section level4">
<h4>fatty acid metabolic process_DZFDC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;fatty acid metabolic process_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 fatty acid metabolic process_DZFDC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="gliogenesis_dzfdc-cut2" class="section level4">
<h4>gliogenesis_DZFDC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;gliogenesis_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 gliogenesis_DZFDC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cellular-response-to-fibroblast-growth-factor-stimulus_mrc-cut2" class="section level4">
<h4>cellular response to fibroblast growth factor stimulus_MRC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;cellular response to fibroblast growth factor stimulus_MRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 cellular response to fibroblast growth factor stimulus_MRC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_tbrc-cut2" class="section level4">
<h4>regulation of angiogenesis_TBRC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of angiogenesis_TBRC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-interferon-alpha_tbrc-cut2" class="section level4">
<h4>response to interferon-alpha_TBRC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to interferon-alpha_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 response to interferon-alpha_TBRC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="organ-growth_prc-cut2" class="section level4">
<h4>organ growth_PRC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;organ growth_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 organ growth_PRC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_prc-cut2" class="section level4">
<h4>regulation of angiogenesis_PRC cut2</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
sceSub &lt;- sce[which(rownames(sce) %in% signGenes$geneID),]
cntMat &lt;- rowSums(t(as.matrix(
    sceSub@assays@data$logcounts)))/nrow(signGenes)
sceSub$sign &lt;- cntMat
sceSub$sign[which(sceSub$sign &gt; cutOff)] &lt;- cutOff
p &lt;- visGroup_adapt(sceSub, &#39;sign&#39;, dim_red = &#39;DiffusionMap&#39;) +
    sc +
    guides(colour = guide_colourbar(title = &#39;&#39;)) +
    ggtitle(paste0(sign, &#39; signature&#39;)) +
    theme_classic() + 
    theme(axis.text = element_blank(),
          axis.ticks = element_blank()) +
    labs(x=&#39;Dimension 1&#39;, y=&#39;Dimension 2&#39;)
p</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of angiogenesis_PRC cut 2-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plot-heatmap-signatures" class="section level3 tabset">
<h3 class="tabset">plot heatmap signatures</h3>
<pre class="r"><code>seurat$label_plus_SLO &lt;- paste0(seurat$clusterLabel, &quot;_&quot;, seurat$SLO)
Idents(seurat) &lt;- seurat$label_plus_SLO
ordVec &lt;- levels(seurat)[c(4,13,8,2,12,10,5,15,9,1,11,7,3,14,6)]
gapC &lt;- c(3,6,9,12)

template_hm &lt;- c(
    &quot;#### {{sig}}\n&quot;,
    &quot;```{r sig hm {{sig}}, fig.height=8, fig.width=8}\n&quot;,
    &quot;sign &lt;- as.character(&#39;{{sig}}&#39;)&quot;,
    &quot;signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)&quot;,
    &quot;signGenes$gene &lt;- signGenes$geneID&quot;,
    &quot;pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_hm &lt;- lapply(allSign, 
  function(sig) knitr::knit_expand(text = template_hm)
)</code></pre>
<div id="regulation-of-lipid-biosynthetic-process_lzfdc" class="section level4">
<h4>regulation of lipid biosynthetic process_LZFDC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of lipid biosynthetic process_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of lipid biosynthetic process_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="i-kappab-kinasenf-kappab-signaling_lzfdc" class="section level4">
<h4>I-kappaB kinase/NF-kappaB signaling_LZFDC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;I-kappaB kinase/NF-kappaB signaling_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm I-kappaB kinase/NF-kappaB signaling_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-transforming-growth-factor-beta_lzfdc" class="section level4">
<h4>response to transforming growth factor beta_LZFDC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to transforming growth factor beta_LZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm response to transforming growth factor beta_LZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="fatty-acid-metabolic-process_dzfdc" class="section level4">
<h4>fatty acid metabolic process_DZFDC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;fatty acid metabolic process_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm fatty acid metabolic process_DZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="gliogenesis_dzfdc" class="section level4">
<h4>gliogenesis_DZFDC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;gliogenesis_DZFDC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm gliogenesis_DZFDC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cellular-response-to-fibroblast-growth-factor-stimulus_mrc" class="section level4">
<h4>cellular response to fibroblast growth factor stimulus_MRC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;cellular response to fibroblast growth factor stimulus_MRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm cellular response to fibroblast growth factor stimulus_MRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_tbrc" class="section level4">
<h4>regulation of angiogenesis_TBRC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of angiogenesis_TBRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="response-to-interferon-alpha_tbrc" class="section level4">
<h4>response to interferon-alpha_TBRC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;response to interferon-alpha_TBRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm response to interferon-alpha_TBRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="organ-growth_prc" class="section level4">
<h4>organ growth_PRC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;organ growth_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm organ growth_PRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="regulation-of-angiogenesis_prc" class="section level4">
<h4>regulation of angiogenesis_PRC</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;regulation of angiogenesis_PRC&#39;)
signGenes &lt;- signDat %&gt;% dplyr::filter(signSubset == sign)
signGenes$gene &lt;- signGenes$geneID
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = signGenes,
                  colVecIdent = colPal, 
                  ordVec=ordVec,
                  gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of angiogenesis_PRC-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] muscat_1.6.0                msigdbr_7.4.1               enrichplot_1.12.3          
 [4] DOSE_3.18.3                 org.Mm.eg.db_3.13.0         AnnotationDbi_1.54.1       
 [7] clusterProfiler_4.0.5       gridExtra_2.3               fgsea_1.18.0               
[10] biomaRt_2.48.3              pheatmap_1.0.12             scran_1.20.1               
[13] scater_1.20.1               scuttle_1.2.1               ggsci_2.9                  
[16] viridis_0.6.2               viridisLite_0.4.0           RColorBrewer_1.1-2         
[19] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0 Biobase_2.52.0             
[22] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4         IRanges_2.26.0             
[25] S4Vectors_0.30.2            BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[28] matrixStats_0.61.0          runSeurat3_0.1.0            here_1.0.1                 
[31] magrittr_2.0.1              SeuratObject_4.0.2          Seurat_4.0.5               
[34] forcats_0.5.1               stringr_1.4.0               dplyr_1.0.7                
[37] purrr_0.3.4                 readr_2.0.2                 tidyr_1.1.4                
[40] tibble_3.1.6                ggplot2_3.3.5               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] vcd_1.4-9                 rsvd_1.0.5                ica_1.0-2                
  [4] class_7.3-19              laeken_0.5.2              foreach_1.5.1            
  [7] lmtest_0.9-39             rprojroot_2.0.2           crayon_1.4.2             
 [10] spatstat.core_2.3-1       MASS_7.3-54               nlme_3.1-153             
 [13] backports_1.3.0           reprex_2.0.1              GOSemSim_2.18.1          
 [16] rlang_0.4.12              XVector_0.32.0            ROCR_1.0-11              
 [19] readxl_1.3.1              irlba_2.3.3               nloptr_1.2.2.3           
 [22] limma_3.48.3              smoother_1.1              filelock_1.0.2           
 [25] BiocParallel_1.26.2       rjson_0.2.20              bit64_4.0.5              
 [28] glue_1.5.0                sctransform_0.3.2         pbkrtest_0.5.1           
 [31] vipor_0.4.5               spatstat.sparse_2.0-0     spatstat.geom_2.3-0      
 [34] haven_2.4.3               tidyselect_1.1.1          fitdistrplus_1.1-6       
 [37] variancePartition_1.22.0  XML_3.99-0.8              zoo_1.8-9                
 [40] RcppHNSW_0.3.0            xtable_1.8-4              evaluate_0.14            
 [43] cli_3.1.0                 zlibbioc_1.38.0           rstudioapi_0.13          
 [46] miniUI_0.1.1.1            sp_1.4-5                  bslib_0.3.1              
 [49] rpart_4.1-15              fastmatch_1.1-3           treeio_1.16.2            
 [52] RcppEigen_0.3.3.9.1       shiny_1.7.1               BiocSingular_1.8.1       
 [55] xfun_0.28                 clue_0.3-60               cluster_2.1.2            
 [58] caTools_1.18.2            tidygraph_1.2.0           pcaMethods_1.84.0        
 [61] KEGGREST_1.32.0           ggrepel_0.9.1             ape_5.5                  
 [64] listenv_0.8.0             Biostrings_2.60.2         png_0.1-7                
 [67] future_1.23.0             withr_2.4.2               bitops_1.0-7             
 [70] ggforce_0.3.3             ranger_0.13.1             plyr_1.8.6               
 [73] cellranger_1.1.0          e1071_1.7-9               coda_0.19-4              
 [76] dqrng_0.3.0               pillar_1.6.4              gplots_3.1.1             
 [79] GlobalOptions_0.1.2       cachem_1.0.6              multcomp_1.4-17          
 [82] fs_1.5.0                  TTR_0.24.2                scatterplot3d_0.3-41     
 [85] GetoptLong_1.0.5          xts_0.12.1                DelayedMatrixStats_1.14.3
 [88] vctrs_0.3.8               ellipsis_0.3.2            generics_0.1.1           
 [91] tools_4.1.0               beeswarm_0.4.0            munsell_0.5.0            
 [94] tweenr_1.0.2              proxy_0.4-26              emmeans_1.7.0            
 [97] DelayedArray_0.18.0       fastmap_1.1.0             compiler_4.1.0           
[100] abind_1.4-5               httpuv_1.6.3              rcartocolor_2.0.0        
[103] plotly_4.10.0             GenomeInfoDbData_1.2.6    glmmTMB_1.1.2.3          
[106] workflowr_1.6.2           edgeR_3.34.1              lattice_0.20-45          
[109] deldir_1.0-6              utf8_1.2.2                later_1.3.0              
[112] BiocFileCache_2.0.0       jsonlite_1.7.2            ggplot.multistats_1.0.0  
[115] scales_1.1.1              ScaledMatrix_1.0.0        carData_3.0-4            
[118] estimability_1.3          tidytree_0.3.5            pbapply_1.5-0            
[121] sparseMatrixStats_1.4.2   genefilter_1.74.1         lazyeval_0.2.2           
[124] promises_1.2.0.1          car_3.0-12                doParallel_1.0.16        
[127] goftest_1.2-3             spatstat.utils_2.2-0      reticulate_1.22          
[130] sandwich_3.0-1            rmarkdown_2.11            cowplot_1.1.1            
[133] blme_1.0-5                statmod_1.4.36            Rtsne_0.15               
[136] downloader_0.4            uwot_0.1.10               igraph_1.2.8             
[139] survival_3.2-13           numDeriv_2016.8-1.1       yaml_2.2.1               
[142] htmltools_0.5.2           memoise_2.0.0             locfit_1.5-9.4           
[145] graphlayouts_0.7.1        destiny_3.7.1             digest_0.6.28            
[148] assertthat_0.2.1          mime_0.12                 rappdirs_0.3.3           
[151] RSQLite_2.2.8             yulab.utils_0.0.4         future.apply_1.8.1       
[154] data.table_1.14.2         blob_1.2.2                labeling_0.4.2           
[157] splines_4.1.0             Cairo_1.5-12.2            RCurl_1.98-1.5           
[160] broom_0.7.10              hms_1.1.1                 modelr_0.1.8             
[163] colorspace_2.0-2          ggbeeswarm_0.6.0          shape_1.4.6              
[166] aplot_0.1.1               nnet_7.3-16               sass_0.4.0               
[169] Rcpp_1.0.7                RANN_2.6.1                mvtnorm_1.1-3            
[172] circlize_0.4.13           fansi_0.5.0               tzdb_0.2.0               
[175] VIM_6.1.1                 parallelly_1.28.1         R6_2.5.1                 
[178] ggridges_0.5.3            lifecycle_1.0.1           bluster_1.2.1            
[181] curl_4.3.2                minqa_1.2.4               leiden_0.3.9             
[184] jquerylib_0.1.4           robustbase_0.93-9         DO.db_2.9                
[187] Matrix_1.3-4              qvalue_2.24.0             TH.data_1.1-0            
[190] RcppAnnoy_0.0.19          iterators_1.0.13          TMB_1.7.22               
[193] htmlwidgets_1.5.4         beachmat_2.8.1            polyclip_1.10-0          
[196] shadowtext_0.0.9          gridGraphics_0.5-1        rvest_1.0.2              
[199] ComplexHeatmap_2.8.0      mgcv_1.8-38               globals_0.14.0           
[202] patchwork_1.1.1           codetools_0.2-18          lubridate_1.8.0          
[205] GO.db_3.13.0              metapod_1.0.0             gtools_3.9.2             
[208] prettyunits_1.1.1         dbplyr_2.1.1              RSpectra_0.16-0          
[211] gtable_0.3.0              DBI_1.1.1                 git2r_0.28.0             
[214] highr_0.9                 ggfun_0.0.4               tensor_1.5               
[217] httr_1.4.2                KernSmooth_2.23-20        vroom_1.5.5              
[220] stringi_1.7.5             progress_1.2.2            reshape2_1.4.4           
[223] farver_2.1.0              ggthemes_4.2.4            annotate_1.70.0          
[226] hexbin_1.28.2             ggtree_3.0.4              xml2_1.3.2               
[229] colorRamps_2.3            boot_1.3-28               BiocNeighbors_1.10.0     
[232] lme4_1.1-27.1             geneplotter_1.70.0        ggplotify_0.1.0          
[235] scattermore_0.7           DEoptimR_1.0-9            DESeq2_1.32.0            
[238] bit_4.0.4                 scatterpie_0.1.7          spatstat.data_2.1-0      
[241] ggraph_2.0.5              pkgconfig_2.0.3           babelgene_21.4           
[244] lmerTest_3.1-3            knn.covertree_1.0         knitr_1.36               </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Thu Nov 25 13:25:02 2021&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] muscat_1.6.0                msigdbr_7.4.1               enrichplot_1.12.3          
 [4] DOSE_3.18.3                 org.Mm.eg.db_3.13.0         AnnotationDbi_1.54.1       
 [7] clusterProfiler_4.0.5       gridExtra_2.3               fgsea_1.18.0               
[10] biomaRt_2.48.3              pheatmap_1.0.12             scran_1.20.1               
[13] scater_1.20.1               scuttle_1.2.1               ggsci_2.9                  
[16] viridis_0.6.2               viridisLite_0.4.0           RColorBrewer_1.1-2         
[19] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0 Biobase_2.52.0             
[22] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4         IRanges_2.26.0             
[25] S4Vectors_0.30.2            BiocGenerics_0.38.0         MatrixGenerics_1.4.3       
[28] matrixStats_0.61.0          runSeurat3_0.1.0            here_1.0.1                 
[31] magrittr_2.0.1              SeuratObject_4.0.2          Seurat_4.0.5               
[34] forcats_0.5.1               stringr_1.4.0               dplyr_1.0.7                
[37] purrr_0.3.4                 readr_2.0.2                 tidyr_1.1.4                
[40] tibble_3.1.6                ggplot2_3.3.5               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] vcd_1.4-9                 rsvd_1.0.5                ica_1.0-2                
  [4] class_7.3-19              laeken_0.5.2              foreach_1.5.1            
  [7] lmtest_0.9-39             rprojroot_2.0.2           crayon_1.4.2             
 [10] spatstat.core_2.3-1       MASS_7.3-54               nlme_3.1-153             
 [13] backports_1.3.0           reprex_2.0.1              GOSemSim_2.18.1          
 [16] rlang_0.4.12              XVector_0.32.0            ROCR_1.0-11              
 [19] readxl_1.3.1              irlba_2.3.3               nloptr_1.2.2.3           
 [22] limma_3.48.3              smoother_1.1              filelock_1.0.2           
 [25] BiocParallel_1.26.2       rjson_0.2.20              bit64_4.0.5              
 [28] glue_1.5.0                sctransform_0.3.2         pbkrtest_0.5.1           
 [31] vipor_0.4.5               spatstat.sparse_2.0-0     spatstat.geom_2.3-0      
 [34] haven_2.4.3               tidyselect_1.1.1          fitdistrplus_1.1-6       
 [37] variancePartition_1.22.0  XML_3.99-0.8              zoo_1.8-9                
 [40] RcppHNSW_0.3.0            xtable_1.8-4              evaluate_0.14            
 [43] cli_3.1.0                 zlibbioc_1.38.0           rstudioapi_0.13          
 [46] miniUI_0.1.1.1            sp_1.4-5                  bslib_0.3.1              
 [49] rpart_4.1-15              fastmatch_1.1-3           treeio_1.16.2            
 [52] RcppEigen_0.3.3.9.1       shiny_1.7.1               BiocSingular_1.8.1       
 [55] xfun_0.28                 clue_0.3-60               cluster_2.1.2            
 [58] caTools_1.18.2            tidygraph_1.2.0           pcaMethods_1.84.0        
 [61] KEGGREST_1.32.0           ggrepel_0.9.1             ape_5.5                  
 [64] listenv_0.8.0             Biostrings_2.60.2         png_0.1-7                
 [67] future_1.23.0             withr_2.4.2               bitops_1.0-7             
 [70] ggforce_0.3.3             ranger_0.13.1             plyr_1.8.6               
 [73] cellranger_1.1.0          e1071_1.7-9               coda_0.19-4              
 [76] dqrng_0.3.0               pillar_1.6.4              gplots_3.1.1             
 [79] GlobalOptions_0.1.2       cachem_1.0.6              multcomp_1.4-17          
 [82] fs_1.5.0                  TTR_0.24.2                scatterplot3d_0.3-41     
 [85] GetoptLong_1.0.5          xts_0.12.1                DelayedMatrixStats_1.14.3
 [88] vctrs_0.3.8               ellipsis_0.3.2            generics_0.1.1           
 [91] tools_4.1.0               beeswarm_0.4.0            munsell_0.5.0            
 [94] tweenr_1.0.2              proxy_0.4-26              emmeans_1.7.0            
 [97] DelayedArray_0.18.0       fastmap_1.1.0             compiler_4.1.0           
[100] abind_1.4-5               httpuv_1.6.3              rcartocolor_2.0.0        
[103] plotly_4.10.0             GenomeInfoDbData_1.2.6    glmmTMB_1.1.2.3          
[106] workflowr_1.6.2           edgeR_3.34.1              lattice_0.20-45          
[109] deldir_1.0-6              utf8_1.2.2                later_1.3.0              
[112] BiocFileCache_2.0.0       jsonlite_1.7.2            ggplot.multistats_1.0.0  
[115] scales_1.1.1              ScaledMatrix_1.0.0        carData_3.0-4            
[118] estimability_1.3          tidytree_0.3.5            pbapply_1.5-0            
[121] sparseMatrixStats_1.4.2   genefilter_1.74.1         lazyeval_0.2.2           
[124] promises_1.2.0.1          car_3.0-12                doParallel_1.0.16        
[127] goftest_1.2-3             spatstat.utils_2.2-0      reticulate_1.22          
[130] sandwich_3.0-1            rmarkdown_2.11            cowplot_1.1.1            
[133] blme_1.0-5                statmod_1.4.36            Rtsne_0.15               
[136] downloader_0.4            uwot_0.1.10               igraph_1.2.8             
[139] survival_3.2-13           numDeriv_2016.8-1.1       yaml_2.2.1               
[142] htmltools_0.5.2           memoise_2.0.0             locfit_1.5-9.4           
[145] graphlayouts_0.7.1        destiny_3.7.1             digest_0.6.28            
[148] assertthat_0.2.1          mime_0.12                 rappdirs_0.3.3           
[151] RSQLite_2.2.8             yulab.utils_0.0.4         future.apply_1.8.1       
[154] data.table_1.14.2         blob_1.2.2                labeling_0.4.2           
[157] splines_4.1.0             Cairo_1.5-12.2            RCurl_1.98-1.5           
[160] broom_0.7.10              hms_1.1.1                 modelr_0.1.8             
[163] colorspace_2.0-2          ggbeeswarm_0.6.0          shape_1.4.6              
[166] aplot_0.1.1               nnet_7.3-16               sass_0.4.0               
[169] Rcpp_1.0.7                RANN_2.6.1                mvtnorm_1.1-3            
[172] circlize_0.4.13           fansi_0.5.0               tzdb_0.2.0               
[175] VIM_6.1.1                 parallelly_1.28.1         R6_2.5.1                 
[178] ggridges_0.5.3            lifecycle_1.0.1           bluster_1.2.1            
[181] curl_4.3.2                minqa_1.2.4               leiden_0.3.9             
[184] jquerylib_0.1.4           robustbase_0.93-9         DO.db_2.9                
[187] Matrix_1.3-4              qvalue_2.24.0             TH.data_1.1-0            
[190] RcppAnnoy_0.0.19          iterators_1.0.13          TMB_1.7.22               
[193] htmlwidgets_1.5.4         beachmat_2.8.1            polyclip_1.10-0          
[196] shadowtext_0.0.9          gridGraphics_0.5-1        rvest_1.0.2              
[199] ComplexHeatmap_2.8.0      mgcv_1.8-38               globals_0.14.0           
[202] patchwork_1.1.1           codetools_0.2-18          lubridate_1.8.0          
[205] GO.db_3.13.0              metapod_1.0.0             gtools_3.9.2             
[208] prettyunits_1.1.1         dbplyr_2.1.1              RSpectra_0.16-0          
[211] gtable_0.3.0              DBI_1.1.1                 git2r_0.28.0             
[214] highr_0.9                 ggfun_0.0.4               tensor_1.5               
[217] httr_1.4.2                KernSmooth_2.23-20        vroom_1.5.5              
[220] stringi_1.7.5             progress_1.2.2            reshape2_1.4.4           
[223] farver_2.1.0              ggthemes_4.2.4            annotate_1.70.0          
[226] hexbin_1.28.2             ggtree_3.0.4              xml2_1.3.2               
[229] colorRamps_2.3            boot_1.3-28               BiocNeighbors_1.10.0     
[232] lme4_1.1-27.1             geneplotter_1.70.0        ggplotify_0.1.0          
[235] scattermore_0.7           DEoptimR_1.0-9            DESeq2_1.32.0            
[238] bit_4.0.4                 scatterpie_0.1.7          spatstat.data_2.1-0      
[241] ggraph_2.0.5              pkgconfig_2.0.3           babelgene_21.4           
[244] lmerTest_3.1-3            knn.covertree_1.0         knitr_1.36               </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
