<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-05-14" />

<title>organ specific signatures - BRCs only</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSLO_BRC_CXCL13</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">organ specific signatures - BRCs only</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">14 May 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-12-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSLO_BRC_CXCL13/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210519code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210519)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210519code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210519)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13treed202c26a3d47b21ce7a146a8a175c19cd9d5c2e8targetblankd202c26a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8" target="_blank">d202c26</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13treed202c26a3d47b21ce7a146a8a175c19cd9d5c2e8targetblankd202c26a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8" target="_blank">d202c26</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/GSEA/
    Ignored:    data/cellphonedb/
    Ignored:    data/destiny/
    Ignored:    data/lympho/
    Ignored:    data/processedData/
    Ignored:    data/processedDataLymphocytes/
    Ignored:    figure/.DS_Store
    Ignored:    figure/plotSubsetSpecificSignatures.Rmd/
    Ignored:    figure/plotSubsetSpecificSignatures2.Rmd/.DS_Store

Untracked files:
    Untracked:  analysis/plotSubsetSLOSignatures.Rmd
    Untracked:  figure/plotSubsetSLOSignatures.Rmd/
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 I-kappaB kinase/
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 cellular response to fibroblast growth factor stimulus_MRC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 cellular response to fibroblast growth factor stimulus_MRC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 chondrocyte differentiation_LZFDC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 chondrocyte differentiation_LZFDC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 fatty acid metabolic process_DZFDC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 fatty acid metabolic process_DZFDC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 gliogenesis_DZFDC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 gliogenesis_DZFDC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 organ growth_PRC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 organ growth_PRC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 regulation of angiogenesis_PRC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 regulation of angiogenesis_PRC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 regulation of angiogenesis_TBRC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 regulation of angiogenesis_TBRC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 regulation of lipid biosynthetic process_LZFDC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 regulation of lipid biosynthetic process_LZFDC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 response to interferon-alpha_TBRC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 response to interferon-alpha_TBRC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 response to transforming growth factor beta_LZFDC cut 1-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut1 response to transforming growth factor beta_LZFDC cut 1-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 I-kappaB kinase/
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 cellular response to fibroblast growth factor stimulus_MRC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 cellular response to fibroblast growth factor stimulus_MRC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 chondrocyte differentiation_LZFDC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 chondrocyte differentiation_LZFDC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 fatty acid metabolic process_DZFDC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 fatty acid metabolic process_DZFDC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 gliogenesis_DZFDC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 gliogenesis_DZFDC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 organ growth_PRC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 organ growth_PRC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 regulation of angiogenesis_PRC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 regulation of angiogenesis_PRC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 regulation of angiogenesis_TBRC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 regulation of angiogenesis_TBRC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 regulation of lipid biosynthetic process_LZFDC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 regulation of lipid biosynthetic process_LZFDC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 response to interferon-alpha_TBRC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 response to interferon-alpha_TBRC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 response to transforming growth factor beta_LZFDC cut 15-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut15 response to transforming growth factor beta_LZFDC cut 15-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 chondrocyte differentiation_LZFDC cut 2-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 chondrocyte differentiation_LZFDC cut 2-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 I-kappaB kinase/
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 cellular response to fibroblast growth factor stimulus_MRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 cellular response to fibroblast growth factor stimulus_MRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 chondrocyte differentiation_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 chondrocyte differentiation_LZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 fatty acid metabolic process_DZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 fatty acid metabolic process_DZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 gliogenesis_DZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 gliogenesis_DZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 organ growth_PRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 organ growth_PRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 regulation of angiogenesis_PRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 regulation of angiogenesis_PRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 regulation of angiogenesis_TBRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 regulation of angiogenesis_TBRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 regulation of lipid biosynthetic process_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 regulation of lipid biosynthetic process_LZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 response to interferon-alpha_TBRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 response to interferon-alpha_TBRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 response to transforming growth factor beta_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut25 response to transforming growth factor beta_LZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 I-kappaB kinase/
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 cellular response to fibroblast growth factor stimulus_MRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 cellular response to fibroblast growth factor stimulus_MRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 chondrocyte differentiation_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 chondrocyte differentiation_LZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 fatty acid metabolic process_DZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 fatty acid metabolic process_DZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 gliogenesis_DZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 gliogenesis_DZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 organ growth_PRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 organ growth_PRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 regulation of angiogenesis_PRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 regulation of angiogenesis_PRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 regulation of angiogenesis_TBRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 regulation of angiogenesis_TBRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 regulation of lipid biosynthetic process_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 regulation of lipid biosynthetic process_LZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 response to interferon-alpha_TBRC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 response to interferon-alpha_TBRC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 response to transforming growth factor beta_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut25 response to transforming growth factor beta_LZFDC-1.png
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm chondrocyte differentiation_LZFDC-1.pdf
    Untracked:  figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm chondrocyte differentiation_LZFDC-1.png
    Untracked:  figure/visSelIntHeatmapSignaling.Rmd/

Unstaged changes:
    Modified:   analysis/SLOspecific_signature.Rmd
    Modified:   analysis/SubsetSpecificGeneSignaturesSignaling.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/plotSubsetSpecificSignaturesSignaling.Rmd
    Modified:   analysis/visSelIntHeatmapSignaling.Rmd
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 I-kappaB kinase/NF-kappaB signaling_LZFDC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 cellular response to fibroblast growth factor stimulus_MRC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 cellular response to fibroblast growth factor stimulus_MRC cut 2-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 fatty acid metabolic process_DZFDC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 gliogenesis_DZFDC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 gliogenesis_DZFDC cut 2-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 organ growth_PRC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of angiogenesis_PRC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of angiogenesis_TBRC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of angiogenesis_TBRC cut 2-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 regulation of lipid biosynthetic process_LZFDC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 response to interferon-alpha_TBRC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 response to interferon-alpha_TBRC cut 2-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 response to transforming growth factor beta_LZFDC cut 2-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut2 response to transforming growth factor beta_LZFDC cut 2-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 I-kappaB kinase/NF-kappaB signaling_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 I-kappaB kinase/NF-kappaB signaling_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 cellular response to fibroblast growth factor stimulus_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 cellular response to fibroblast growth factor stimulus_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 fatty acid metabolic process_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 fatty acid metabolic process_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 gliogenesis_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 gliogenesis_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 organ growth_PRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 organ growth_PRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of angiogenesis_PRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of angiogenesis_PRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of angiogenesis_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of angiogenesis_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of lipid biosynthetic process_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 regulation of lipid biosynthetic process_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 response to interferon-alpha_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 response to interferon-alpha_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 response to transforming growth factor beta_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr DM cut3 response to transforming growth factor beta_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 I-kappaB kinase/NF-kappaB signaling_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 I-kappaB kinase/NF-kappaB signaling_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 cellular response to fibroblast growth factor stimulus_MRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 cellular response to fibroblast growth factor stimulus_MRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 fatty acid metabolic process_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 fatty acid metabolic process_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 gliogenesis_DZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 gliogenesis_DZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 organ growth_PRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 organ growth_PRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of angiogenesis_PRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of angiogenesis_PRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of angiogenesis_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of angiogenesis_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of lipid biosynthetic process_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 regulation of lipid biosynthetic process_LZFDC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 response to interferon-alpha_TBRC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 response to interferon-alpha_TBRC-1.png
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 response to transforming growth factor beta_LZFDC-1.pdf
    Deleted:    figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig expr cut3 response to transforming growth factor beta_LZFDC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm I-kappaB kinase/NF-kappaB signaling_LZFDC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm I-kappaB kinase/NF-kappaB signaling_LZFDC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm cellular response to fibroblast growth factor stimulus_MRC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm cellular response to fibroblast growth factor stimulus_MRC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm fatty acid metabolic process_DZFDC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm fatty acid metabolic process_DZFDC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm gliogenesis_DZFDC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm gliogenesis_DZFDC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm organ growth_PRC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm organ growth_PRC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of angiogenesis_PRC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of angiogenesis_PRC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of angiogenesis_TBRC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of angiogenesis_TBRC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of lipid biosynthetic process_LZFDC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm regulation of lipid biosynthetic process_LZFDC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm response to interferon-alpha_TBRC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm response to interferon-alpha_TBRC-1.png
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm response to transforming growth factor beta_LZFDC-1.pdf
    Modified:   figure/plotSubsetSpecificSignaturesSignaling.Rmd/sig hm response to transforming growth factor beta_LZFDC-1.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SLOspecific_signature.Rmd</code>) and HTML (<code>docs/SLOspecific_signature.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/analysis/SLOspecific_signature.Rmd" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
<td>
add CPDB analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/SLOspecific_signature.html" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
<td>
add CPDB analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/analysis/SLOspecific_signature.Rmd" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
<td>
update BRC specific genes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/SLOspecific_signature.html" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
<td>
update BRC specific genes
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/1f27cea5638ec11795ed815300ef71869e6d3b0b/analysis/SLOspecific_signature.Rmd" target="_blank">1f27cea</a>
</td>
<td>
mluetge
</td>
<td>
2021-07-02
</td>
<td>
update colors and labels plus marker genes
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(SingleCellExperiment)
  library(RColorBrewer)
  library(viridis)
  library(ggsci)
  library(scater)
  library(scran)
  library(pheatmap)
  library(biomaRt)
  library(fgsea)
  library(grid)
  library(gridExtra)
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  library(enrichplot)
  library(msigdbr)
  library(muscat)
  library(ggpubr)
})</code></pre>
</div>
<div id="plotting-funct" class="section level2">
<h2>plotting funct</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  ## format gene names (depends on how gene list is inputed)
  selGenes &lt;- selGenes %&gt;% 
      mutate(geneID = (str_split(gene, &#39;\\.&#39;, simplify = T)[,2]))
  #selGenes &lt;- selGenes$labelNam
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;%
    filter(geneID %in% selGenes$geneID)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes$geneID,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="set-dir-and-read-input-data" class="section level2">
<h2>set dir and read input data</h2>
<pre class="r"><code>basedir &lt;- here()

## read seurat objects with BRCs
seurat &lt;- readRDS(file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))
## seurat &lt;-rerunSeurat3(seurat)
## saveRDS(seurat, file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color Vectors</h2>
<pre class="r"><code>colPal &lt;- c(&quot;#87c5af&quot;, &quot;#94033b&quot;, &quot;#fc8f3b&quot;, &quot;#2580fe&quot;, &quot;#514e72&quot;)
names(colPal) &lt;- c(&quot;LZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;DZFDC&quot;,&quot;PRC2&quot;)


colorSLO &lt;- c(viridis(3)[c(1,2)], brewer.pal(n=9, name=&quot;YlOrRd&quot;)[c(4)])
colorCond &lt;- brewer.pal(n=3, name=&quot;Paired&quot;)[c(1,2)]
colorSLOCond &lt;- c(&quot;#440154FF&quot;,&quot;#807DBA&quot;, &quot;#FEB24C&quot;, &quot;#F16913&quot;, &quot;#21908CFF&quot;)
colorBatch &lt;- pal_igv()(n=length(unique(seurat$batch)))
colCl &lt;- c(rcartocolor::carto_pal(name=&quot;Safe&quot;),pal_aaas()(8))

names(colorSLO) &lt;- c(&quot;LN&quot;, &quot;PP&quot;, &quot;Spleen&quot;)
names(colorCond) &lt;- c(&quot;naive&quot;, &quot;immunized&quot;)
names(colorBatch) &lt;- unique(seurat$batch)
names(colorSLOCond) &lt;- c(&quot;LN_immunized&quot;,&quot;LN_naive&quot;, &quot;Spleen_naive&quot;,
                         &quot;Spleen_immunized&quot;, &quot;PP_immunized&quot;)</code></pre>
</div>
<div id="visualize-data" class="section level2 tabset">
<h2 class="tabset">visualize data</h2>
<div id="clustering" class="section level3">
<h3>clustering</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, group.by = &quot;clusterLabel&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-clustering-1">
Past versions of clustering-1.png
</button>
</p>
<div id="fig-clustering-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/clustering-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/clustering-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="slo" class="section level3">
<h3>SLO</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLO, group.by = &quot;SLO&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/SLO-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-SLO-1">
Past versions of SLO-1.png
</button>
</p>
<div id="fig-SLO-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/SLO-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/SLO-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLO, group.by = &quot;SLO&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/SLO-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-SLO-2">
Past versions of SLO-2.png
</button>
</p>
<div id="fig-SLO-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/SLO-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/SLO-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="batch" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorBatch, group.by = &quot;batch&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/Batch-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Batch-1">
Past versions of Batch-1.png
</button>
</p>
<div id="fig-Batch-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/Batch-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/Batch-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cond" class="section level3">
<h3>Cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorCond, group.by = &quot;cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/Cond-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Cond-1">
Past versions of Cond-1.png
</button>
</p>
<div id="fig-Cond-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/Cond-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/Cond-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="slo-plus-cond" class="section level3">
<h3>SLO plus cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLOCond,group.by=&quot;SLO_plus_cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/SLO%20plus%20cond-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-SLO-plus-cond-1">
Past versions of “SLO plus cond-1.png”
</button>
</p>
<div id="fig-no-spaces-SLO-plus-cond-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/SLO plus cond-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/SLO plus cond-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,cols=colorSLOCond,group.by = &quot;SLO_plus_cond&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/SLO%20plus%20cond-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-SLO-plus-cond-2">
Past versions of “SLO plus cond-2.png”
</button>
</p>
<div id="fig-no-spaces-SLO-plus-cond-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/SLO plus cond-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/SLO plus cond-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="slo-specific-brc-gene-signatures" class="section level2">
<h2>SLO specific BRC gene signatures</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$SLO
seurat_markers_all &lt;- FindAllMarkers(object = seurat,
                                     only.pos = TRUE, min.pct = 0.25,
                                     logfc.threshold = 0.25,
                                     test.use = &quot;wilcox&quot;)

write.table(seurat_markers_all,
            file=paste0(basedir, &quot;/data/allBRC_SLOsignature.txt&quot;),
            row.names = FALSE, col.names = TRUE, quote = FALSE, sep = &quot;\t&quot;)</code></pre>
<div id="vis-top-marker" class="section level3 tabset">
<h3 class="tabset">vis top marker</h3>
<div id="across-slo" class="section level4">
<h4>across SLO</h4>
<pre class="r"><code>selGenes &lt;- seurat_markers_all %&gt;% group_by(cluster) %&gt;% 
  top_n(-15, p_val_adj) %&gt;% 
  top_n(15, avg_log2FC)

Idents(seurat) &lt;- seurat$SLO_plus_clusterlabel

gapC &lt;- c((length(unique(seurat$SLO_plus_clusterlabel))/3),
          2*(length(unique(seurat$SLO_plus_clusterlabel))/3))
gapR &lt;- c(15,30)

pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenes,
                  colVecIdent = colorSLO, 
                  ordVec=levels(seurat),
                  gapVecR=gapR, gapVecC=gapC,cc=F,
                  cr=F, condCol=T, colVecCond = colPal)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/vis%20top%20marker%20SLO-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-top-marker-SLO-1">
Past versions of “vis top marker SLO-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-top-marker-SLO-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/vis top marker SLO-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/vis top marker SLO-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="slo-specific-gene-signatures-per-subset" class="section level2 tabset">
<h2 class="tabset">SLO specific gene signatures per subset</h2>
<pre class="r"><code>seurat$clusterGrp &lt;- seurat$clusterLabel

Idents(seurat) &lt;- seurat$SLO
cwDEgenes &lt;- lapply(unique(seurat$clusterGrp), function(grp){
  grpSub &lt;- unique(seurat$clusterGrp)[which(unique(seurat$clusterGrp)==grp)]
  seuratSub &lt;- subset(seurat, clusterGrp == grpSub)
  DEgenes &lt;-FindAllMarkers(seuratSub, only.pos=T, logfc.threshold = 0.25,
                           min.pct = 0.1)
  DEgenes &lt;- DEgenes %&gt;% filter(p_val_adj&lt;0.01)
})

names(cwDEgenes) &lt;- unique(seurat$clusterGrp)</code></pre>
</div>
<div id="slo-marker-overlap-between-grps" class="section level2">
<h2>SLO marker overlap between grps</h2>
<pre class="r"><code>grpVec &lt;- unique(seurat$clusterGrp)
orgVec &lt;- unique(seurat$SLO)

orgMarker &lt;- lapply(orgVec, 
  function(slo){
    TBRCgenes &lt;- cwDEgenes[[&quot;TBRC&quot;]] %&gt;% filter(cluster==slo)
    MRCgenes &lt;- cwDEgenes[[&quot;MRC&quot;]] %&gt;% filter(cluster==slo)
    LZFDCgenes &lt;- cwDEgenes[[&quot;LZFDC&quot;]] %&gt;% filter(cluster==slo)
    DZFDCgenes &lt;- cwDEgenes[[&quot;DZFDC&quot;]] %&gt;% filter(cluster==slo)
    PRC2genes &lt;- cwDEgenes[[&quot;PRC2&quot;]] %&gt;% filter(cluster==slo)
    all &lt;- intersect(intersect(intersect(intersect(TBRCgenes$gene, MRCgenes$gene),
                               LZFDCgenes$gene), DZFDCgenes$gene), PRC2genes$gene)
  }
)

names(orgMarker) &lt;- orgVec
## orgMarker = DE genes in all shared folicular BRC subsets
## (adj_pVal &lt; 0.01, min_pct = 0.1, log_FC &gt; 0.25)

orgMarker &lt;- lapply(orgVec, 
                    function(slo){
                      GrpGeneList &lt;- lapply(grpVec, function(grp){
                        genesGrp &lt;- cwDEgenes[[grp]] %&gt;% filter(cluster==slo)
                        })
                      SLOgenes &lt;- data.frame(do.call(&quot;rbind&quot;, GrpGeneList)) %&gt;% 
                        group_by(gene) %&gt;% summarise(cnt=n()) %&gt;%
                        filter(cnt&gt;=(length(grpVec)-1))
                      out &lt;- SLOgenes$gene
                      }
                    )

names(orgMarker) &lt;- orgVec</code></pre>
</div>
<div id="vis-all-marker-conserved-across-all-subsets" class="section level2 tabset">
<h2 class="tabset">vis all marker conserved across all subsets</h2>
<div id="ln" class="section level3">
<h3>LN</h3>
<pre class="r"><code>seurat$label_plus_organ_plus_cond &lt;- factor(paste0(seurat$SLO_plus_clusterlabel, &quot;_&quot;,
                                            seurat$cond))
Idents(seurat) &lt;- seurat$label_plus_organ_plus_cond
selGenes &lt;- data.frame(gene=orgMarker[[&quot;LN&quot;]])
allGenes &lt;- data.frame(gene=rownames(seurat)) %&gt;%
  mutate(geneNam=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
selGenesDat &lt;- selGenes %&gt;% left_join(., allGenes, by=c(&quot;gene&quot;))

seurat$label_plus_cond &lt;- paste0(seurat$clusterLabel, &quot;_&quot;, seurat$cond)
colSel &lt;- c(rep(colorCond[[&quot;naive&quot;]], 5), rep(colorCond[[&quot;immunized&quot;]], 5))
names(colSel) &lt;- unique(seurat$label_plus_cond)
pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesDat,
                  colVecIdent = colorSLO, 
                  ordVec=levels(seurat),
                  gapVecC=c(10,15),cc=F,
                  cr=T,condCol=T, colVecCond=colSel)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/vis%20all%20conserved%20marker%20LN-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-all-conserved-marker-LN-1">
Past versions of “vis all conserved marker LN-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-all-conserved-marker-LN-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/vis all conserved marker LN-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/vis all conserved marker LN-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="spleen" class="section level3">
<h3>Spleen</h3>
<pre class="r"><code>selGenes &lt;- data.frame(gene=orgMarker[[&quot;Spleen&quot;]])
allGenes &lt;- data.frame(gene=rownames(seurat)) %&gt;%
  mutate(geneNam=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
selGenesDat &lt;- selGenes %&gt;% left_join(., allGenes, by=c(&quot;gene&quot;)) %&gt;% 
  filter(nchar(geneNam)&gt;1)

pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesDat,
                  colVecIdent = colorSLO, 
                  ordVec=levels(seurat),
                  gapVecC=c(10,15),cc=F,
                  cr=T,condCol=T, colVecCond=colSel)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/vis%20all%20conserved%20marker%20spleen-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-all-conserved-marker-spleen-1">
Past versions of “vis all conserved marker spleen-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-all-conserved-marker-spleen-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/vis all conserved marker spleen-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/vis all conserved marker spleen-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pp" class="section level3">
<h3>PP</h3>
<pre class="r"><code>selGenes &lt;- data.frame(gene=orgMarker[[&quot;PP&quot;]])
allGenes &lt;- data.frame(gene=rownames(seurat)) %&gt;%
  mutate(geneNam=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))
selGenesDat &lt;- selGenes %&gt;% left_join(., allGenes, by=c(&quot;gene&quot;))

pOut &lt;- avgHeatmap(seurat = seurat, selGenes = selGenesDat,
                  colVecIdent = colorSLO, 
                  ordVec=levels(seurat),
                  gapVecC=c(10,15),cc=F,
                  cr=T,condCol=T, colVecCond=colSel)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/vis%20all%20conserved%20marker%20PP-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-all-conserved-marker-PP-1">
Past versions of “vis all conserved marker PP-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-all-conserved-marker-PP-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/vis all conserved marker PP-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/vis all conserved marker PP-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sc-heatmap-slo-specific-marker" class="section level3">
<h3>sc heatmap SLO specific marker</h3>
<pre class="r"><code>selFeatures &lt;- data.frame(ensID=c(orgMarker$LN, orgMarker$PP, 
                                  orgMarker$Spleen)) %&gt;% 
  mutate(label=gsub(&quot;^.*\\.&quot;, &quot;&quot;, ensID))
DoHeatmap(seurat, features = selFeatures$ensID, group.by = &quot;SLO&quot;, 
          group.colors = colorSLO, slot = &#39;scale.data&#39;, label = F,
          disp.min = -0.5, disp.max = 2) +
  scale_fill_continuous(type = &quot;viridis&quot;) +
  scale_y_discrete(breaks=selFeatures$ensID, labels=selFeatures$label)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/sc%20heatmap%20slo%20specific-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sc-heatmap-slo-specific-1">
Past versions of “sc heatmap slo specific-1.png”
</button>
</p>
<div id="fig-no-spaces-sc-heatmap-slo-specific-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/sc heatmap slo specific-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/sc heatmap slo specific-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DoHeatmap(seurat, features = selFeatures$ensID, group.by = &quot;SLO&quot;, 
          group.colors = colorSLO, slot = &#39;scale.data&#39;, label = F,
          disp.min = -1, disp.max = 1.5, draw.lines = F) +
  scale_fill_gradientn(colors=colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(30)) +
  scale_y_discrete(breaks=selFeatures$ensID, labels=selFeatures$label)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/sc%20heatmap%20slo%20specific-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-sc-heatmap-slo-specific-2">
Past versions of “sc heatmap slo specific-2.png”
</button>
</p>
<div id="fig-no-spaces-sc-heatmap-slo-specific-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/sc heatmap slo specific-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/sc heatmap slo specific-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="gsea-clusterprofiler" class="section level2 tabset">
<h2 class="tabset">GSEA clusterProfiler</h2>
<div id="ln-1" class="section level3">
<h3>LN</h3>
<pre class="r"><code>orgMarkerDat &lt;- data.frame(do.call(&quot;cbind&quot;, orgMarker)) %&gt;% 
  gather(., SLO, gene, LN:Spleen) %&gt;% mutate(EnsID = gsub(&quot;\\..*$&quot;, &quot;&quot;, gene))

write.table(orgMarkerDat,
            file = paste0(basedir, &quot;/data/GSEA/SLOspecificGenes.txt&quot;),
            sep=&quot;\t&quot;, quote = F, row.names = F, col.names = T)

orgMarkerDatLN &lt;- orgMarkerDat %&gt;% filter(SLO == &quot;LN&quot;)
egoLN &lt;- enrichGO(gene         = unique(orgMarkerDatLN$EnsID),
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
egoLN &lt;- setReadable(egoLN, OrgDb = org.Mm.eg.db)
dotplot(egoLN, showCategory=30)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-1">
Past versions of “GSEA clusterprofiler-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(egoLN, node_label=&quot;all&quot;) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-2">
Past versions of “GSEA clusterprofiler-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pp-1" class="section level3">
<h3>PP</h3>
<pre class="r"><code>orgMarkerDatPP &lt;- orgMarkerDat %&gt;% filter(SLO == &quot;PP&quot;)
egoPP &lt;- enrichGO(gene         = unique(orgMarkerDatPP$EnsID),
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.5,
                qvalueCutoff  = 0.5)
egoPP &lt;- setReadable(egoPP, OrgDb = org.Mm.eg.db)
dotplot(egoPP, showCategory=30)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20PP-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-PP-1">
Past versions of “GSEA clusterprofiler PP-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-PP-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler PP-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler PP-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(egoPP, node_label=&quot;all&quot;) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20PP-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-PP-2">
Past versions of “GSEA clusterprofiler PP-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-PP-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler PP-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler PP-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="spleen-1" class="section level3">
<h3>Spleen</h3>
<pre class="r"><code>orgMarkerDatSP &lt;- orgMarkerDat %&gt;% filter(SLO == &quot;Spleen&quot;)
egoSP &lt;- enrichGO(gene         = unique(orgMarkerDatSP$EnsID),
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
egoSP &lt;- setReadable(egoSP, OrgDb = org.Mm.eg.db)
dotplot(egoSP, showCategory=30)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-1">
Past versions of “GSEA clusterprofiler Spleen-1.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cnetplot(egoSP, node_label=&quot;all&quot;) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-2">
Past versions of “GSEA clusterprofiler Spleen-2.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>goplot(egoSP)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-3">
Past versions of “GSEA clusterprofiler Spleen-3.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-3.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>egoSP2 &lt;- pairwise_termsim(egoSP)
treeplot(egoSP2)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-4">
Past versions of “GSEA clusterprofiler Spleen-4.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-4.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>emapplot(egoSP2)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-5">
Past versions of “GSEA clusterprofiler Spleen-5.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-5.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>upsetplot(egoSP2)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-6.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-6">
Past versions of “GSEA clusterprofiler Spleen-6.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-6.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>egoSP2 &lt;- simplify(egoSP2, cutoff=0.7, by=&quot;p.adjust&quot;, select_fun=min)
emapplot(egoSP2)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-7.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-7">
Past versions of “GSEA clusterprofiler Spleen-7.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-7.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>treeplot(egoSP2)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-8.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-8">
Past versions of “GSEA clusterprofiler Spleen-8.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-8.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dotplot(egoSP, showCategory=30)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/GSEA%20clusterprofiler%20Spleen-9.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-GSEA-clusterprofiler-Spleen-9">
Past versions of “GSEA clusterprofiler Spleen-9.png”
</button>
</p>
<div id="fig-no-spaces-GSEA-clusterprofiler-Spleen-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/GSEA clusterprofiler Spleen-9.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>egoLNdat &lt;- egoLN@result %&gt;% mutate(SLO=&quot;LN&quot;)
egoPPdat &lt;- egoPP@result %&gt;% mutate(SLO=&quot;PP&quot;)
egoSPdat &lt;- egoSP@result %&gt;% mutate(SLO=&quot;Spleen&quot;)

egoResAll &lt;- rbind(egoLNdat, egoPPdat, egoSPdat) 

write.table(egoResAll, quote=F, row.names = T, col.names = T, sep= &quot;\t&quot;,
            file = paste0(basedir, &quot;/data/GSEA/SLOSpecificGO.txt&quot;))</code></pre>
</div>
<div id="vis-summary-all" class="section level3 tabset">
<h3 class="tabset">vis summary all</h3>
<div id="dotplot" class="section level4">
<h4>Dotplot</h4>
<pre class="r"><code>selLN &lt;- c(&quot;GO:0003002&quot;, &quot;GO:0007389&quot;, &quot;GO:0010464&quot;,&quot;GO:1901342&quot;,&quot;GO:0009611&quot;)
selPW_LN &lt;- egoLN@result %&gt;% filter(qvalue&lt;0.05) %&gt;% 
  filter(ID %in% selLN) %&gt;%
  mutate(IDfac= factor(ID, levels = selLN)) %&gt;% 
  arrange(., IDfac) %&gt;% 
  dplyr::select(ID, Description, GeneRatio, qvalue, Count) %&gt;% 
  mutate(SLO=&quot;LN&quot;) %&gt;% 
  arrange(., Count)

selSP &lt;- c(&quot;GO:0019882&quot;, &quot;GO:0048536&quot;, &quot;GO:1903522&quot;, &quot;GO:0045766&quot;,&quot;GO:0055072&quot;)
selPW_SP &lt;- egoSP@result %&gt;% filter(qvalue&lt;0.05) %&gt;% 
  filter(ID %in% selSP) %&gt;% 
  mutate(IDfac= factor(ID, levels = selSP)) %&gt;% 
  arrange(., IDfac) %&gt;% 
  dplyr::select(ID, Description, GeneRatio, qvalue, Count) %&gt;% 
  mutate(SLO=&quot;Spleen&quot;) %&gt;% 
  arrange(., Count)

selPP &lt;- c(&quot;GO:0071604&quot;, &quot;GO:0042573&quot;,&quot;GO:0006694&quot;,&quot;GO:0030510&quot;, &quot;GO:0048562&quot;)
selPW_PP &lt;- egoPP@result %&gt;% filter(qvalue&lt;1) %&gt;% 
  filter(ID %in% selPP) %&gt;% 
  mutate(IDfac= factor(ID, levels = selPP)) %&gt;% 
  arrange(., IDfac) %&gt;% 
  dplyr::select(ID, Description, GeneRatio, qvalue, Count) %&gt;% 
  mutate(SLO=&quot;PP&quot;) %&gt;% 
  arrange(., Count)


selPW_all &lt;- rbind(selPW_PP, selPW_SP, selPW_LN) %&gt;%
  mutate(nom=as.numeric(gsub(&quot;\\/.*&quot;, &quot;&quot;, GeneRatio))) %&gt;% 
  mutate(denom=as.numeric(gsub(&quot;.*\\/&quot;, &quot;&quot;, GeneRatio))) %&gt;% 
  mutate(geneRatio=nom/denom) %&gt;% 
  mutate(one_minPadjust=(1-qvalue))
 
level_order &lt;- unique(selPW_all$Description)
ggplot(selPW_all, aes(x=factor(Description, level= level_order), y=geneRatio)) + 
  geom_point(aes(col=SLO, size=Count, alpha=one_minPadjust)) + 
  scale_color_manual(values = colorSLO)+
  scale_alpha_continuous(range = c(0.5,1), name=&quot;qvalue&quot;,
                         labels=c(0.04, 0.03, 0.02, 0.01),
                         breaks = c(0.96, 0.97, 0.98, 0.99)) +
  scale_size_continuous(range = c(3,6), labels = c(2,4,6,8),
                        breaks = c(2,4,6,8)) +
  geom_segment(aes(x=Description, 
                   xend=Description, 
                   y=min(geneRatio), 
                   yend=max(geneRatio)), 
               linetype=&quot;dashed&quot;, 
               size=0.1) +   # Draw dashed lines
  coord_flip() +
  theme_minimal() +
  theme(panel.border = element_rect(fill = NA), 
        panel.grid = element_blank(), axis.title.y = element_blank())</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/vis%20summary-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-vis-summary-1">
Past versions of “vis summary-1.png”
</button>
</p>
<div id="fig-no-spaces-vis-summary-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/vis summary-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/vis summary-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="network" class="section level4">
<h4>Network</h4>
<pre class="r"><code>selGOLN &lt;- c(&quot;GO:0007389&quot;, &quot;GO:0003002&quot;, &quot;GO:0010464&quot;, &quot;GO:1901342&quot;)
selGOPP &lt;- c(&quot;GO:0042573&quot;,&quot;GO:0006066&quot;, &quot;GO:0003002&quot;, &quot;GO:0048562&quot;) 
selGOSP &lt;- c(&quot;GO:0060537&quot;, &quot;GO:1903522&quot;, &quot;GO:0055072&quot;, &quot;GO:0048536&quot;) 
selGOall &lt;- c(selGOLN, selGOSP, selGOPP)
  
egoLN@result &lt;- egoLN@result %&gt;% filter(ID %in% selGOLN) %&gt;% mutate(SLO = &quot;LN&quot;)
cnetplot(egoLN, node_label=&quot;all&quot;, 
        color_category=&quot;#440154FF&quot;, 
        color_gene=&quot;#a473b0&quot;,
        cex_label_gene = 0.5) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/network-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-network-1">
Past versions of network-1.png
</button>
</p>
<div id="fig-network-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/network-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>egoPP@result &lt;- egoPP@result %&gt;% filter(ID %in% selGOPP) %&gt;% mutate(SLO = &quot;PP&quot;)
cnetplot(egoPP, node_label=&quot;all&quot;, 
        color_category=&quot;#21908CFF&quot;, 
        color_gene=&quot;#7cbab8&quot;,
        cex_label_gene = 0.5) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/network-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-network-2">
Past versions of network-2.png
</button>
</p>
<div id="fig-network-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/network-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>egoSP@result &lt;- egoSP@result %&gt;% filter(ID %in% selGOSP) %&gt;%
  mutate(SLO = &quot;Spleen&quot;)
cnetplot(egoSP, node_label=&quot;all&quot;, 
        color_category=&quot;#FEB24C&quot;, 
        color_gene=&quot;#e5c9a2&quot;,
        cex_label_gene = 0.5) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/network-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-network-3">
Past versions of network-3.png
</button>
</p>
<div id="fig-network-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/network-3.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="barplot" class="section level4">
<h4>barplot</h4>
<pre class="r"><code>egoResAll &lt;- rbind(egoPP@result, egoSP@result, egoLN@result) %&gt;%
  mutate(qscore=-log(p.adjust, base=10)) %&gt;% 
  mutate(DescriptionGrp = paste0(SLO, &quot;_&quot;, Description)) %&gt;% 
  group_by(SLO) %&gt;% 
  arrange(., qscore, decreasing = FALSE, .by_group = TRUE) %&gt;% 
  arrange(match(SLO, c(&quot;PP&quot;, &quot;Spleen&quot;, &quot;LN&quot;)))
p &lt;- ggbarplot(egoResAll, x = &quot;DescriptionGrp&quot;, y = &quot;qscore&quot;,
          fill = &quot;SLO&quot;,               
          color = &quot;SLO&quot;,            
          palette = colorSLO,            
          #sort.val = &quot;asc&quot;,           
          sort.by.groups = TRUE      
          #x.text.angle = 90           
          ) + 
  rotate()
p</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/unnamed-chunk-1-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="compare-cluster" class="section level4">
<h4>compare cluster</h4>
<pre class="r"><code>orgMarkerList &lt;- lapply(orgVec, function(slo){
  selMarker &lt;- orgMarkerDat %&gt;% filter(SLO == slo)
  out &lt;- selMarker$EnsID
})
names(orgMarkerList) &lt;- orgVec

ck &lt;- compareCluster(geneCluster = orgMarkerList,
                     fun = enrichGO,
                     OrgDb = org.Mm.eg.db,
                     keyType = &#39;ENSEMBL&#39;,
                     ont = &quot;BP&quot;,
                     pvalueCutoff  = 0.1,
                     qvalueCutoff  = 0.1
)
ck &lt;- setReadable(ck, OrgDb = org.Mm.eg.db)
ck@compareClusterResult &lt;- ck@compareClusterResult %&gt;% filter(ID %in% selGOall)


cnetplot(ck) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/compare%20cluster-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-compare-cluster-1">
Past versions of “compare cluster-1.png”
</button>
</p>
<div id="fig-no-spaces-compare-cluster-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/compare cluster-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="organ-specific-genes-dotplot" class="section level2">
<h2>organ specific genes dotplot</h2>
<pre class="r"><code>selMarkerAll &lt;- read_tsv(paste0(basedir,&quot;/data/SLOspecificSelMarker.txt&quot;))


genes &lt;- data.frame(ensID=rownames(seurat)) %&gt;%
  mutate(gene=gsub(&quot;^.*\\.&quot;, &quot;&quot;, ensID))
selGenes &lt;- selMarkerAll %&gt;% 
  left_join(., genes, by=c(&quot;geneNam&quot;=&quot;gene&quot;))

Idents(seurat) &lt;- seurat$SLO_plus_clusterlabel
clusterDat &lt;- data.frame(ident=levels(seurat)) %&gt;%
  mutate(SLO=gsub(&quot;_.*$&quot;, &quot;&quot;, ident)) %&gt;% 
  mutate(namFull=paste0(ident, &quot;_&quot;, SLO)) %&gt;%
  mutate(label= gsub(&quot;^.*_&quot;, &quot;&quot;, ident))


levels(seurat) &lt;- levels(seurat)[c(4,2,5,1,3,8,10,9,7,6,15,13,12,11,14)]

DotPlot(seurat, features = selGenes$ensID, split.by = &quot;SLO&quot;,
        cols = colorSLO, col.max = 6) +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenes$ensID, labels=selGenes$geneNam) +
  scale_y_discrete(breaks=clusterDat$namFull, labels=clusterDat$ident) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) + theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/SLO%20sepcific%20genes-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-SLO-sepcific-genes-1">
Past versions of “SLO sepcific genes-1.png”
</button>
</p>
<div id="fig-no-spaces-SLO-sepcific-genes-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/SLO sepcific genes-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/SLO sepcific genes-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DotPlot(seurat, features = selGenes$ensID, split.by = &quot;SLO&quot;,
        cols = colorSLO, col.max = 6) +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  scale_x_discrete(breaks=selGenes$ensID, labels=selGenes$geneNam) +
  scale_y_discrete(breaks=clusterDat$namFull, labels=clusterDat$ident) +
  xlab(&quot;&quot;) + ylab(&quot;&quot;) </code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/SLO%20sepcific%20genes-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-SLO-sepcific-genes-2">
Past versions of “SLO sepcific genes-2.png”
</button>
</p>
<div id="fig-no-spaces-SLO-sepcific-genes-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/SLO sepcific genes-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/SLO sepcific genes-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="mds-plot" class="section level2">
<h2>MDS plot</h2>
<pre class="r"><code>sce &lt;- as.SingleCellExperiment(seurat)
sce &lt;- prepSCE(sce, 
    kid = &quot;SLO&quot;, 
    gid = &quot;cond&quot;,   
    sid = &quot;dataset&quot;,  
    drop = F)       

pb &lt;- aggregateData(sce,
    assay = &quot;counts&quot;, fun = &quot;sum&quot;,
    by = c(&quot;cluster_id&quot;, &quot;sample_id&quot;))

pb_mds &lt;- pbMDS(pb) +
  scale_color_manual(values = colorSLO)
pb_mds</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/mds%20plot-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mds-plot-1">
Past versions of “mds plot-1.png”
</button>
</p>
<div id="fig-no-spaces-mds-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/mds plot-1.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/mds plot-1.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pb_mds + theme_void()</code></pre>
<p><img src="figure/SLOspecific_signature.Rmd/mds%20plot-2.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-mds-plot-2">
Past versions of “mds plot-2.png”
</button>
</p>
<div id="fig-no-spaces-mds-plot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/e71d77eda19ead98ddc3d5490ca1bb6addc09ba2/docs/figure/SLOspecific_signature.Rmd/mds plot-2.png" target="_blank">e71d77e</a>
</td>
<td>
mluetge
</td>
<td>
2021-10-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/f45bed1da1d13948e8079688920fdf0a61fdae21/docs/figure/SLOspecific_signature.Rmd/mds plot-2.png" target="_blank">f45bed1</a>
</td>
<td>
mluetge
</td>
<td>
2021-08-03
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] ggpubr_0.4.0                muscat_1.6.0                msigdbr_7.4.1              
 [4] enrichplot_1.12.3           DOSE_3.18.3                 org.Mm.eg.db_3.13.0        
 [7] AnnotationDbi_1.54.1        clusterProfiler_4.0.5       gridExtra_2.3              
[10] fgsea_1.18.0                biomaRt_2.48.3              pheatmap_1.0.12            
[13] scran_1.20.1                scater_1.20.1               scuttle_1.2.1              
[16] ggsci_2.9                   viridis_0.6.2               viridisLite_0.4.0          
[19] RColorBrewer_1.1-2          SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
[22] Biobase_2.52.0              GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[25] IRanges_2.26.0              S4Vectors_0.30.2            BiocGenerics_0.38.0        
[28] MatrixGenerics_1.4.3        matrixStats_0.61.0          runSeurat3_0.1.0           
[31] here_1.0.1                  magrittr_2.0.1              SeuratObject_4.0.2         
[34] Seurat_4.0.5                forcats_0.5.1               stringr_1.4.0              
[37] dplyr_1.0.7                 purrr_0.3.4                 readr_2.0.2                
[40] tidyr_1.1.4                 tibble_3.1.6                ggplot2_3.3.5              
[43] tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5                ica_1.0-2                 foreach_1.5.1            
  [4] lmtest_0.9-39             rprojroot_2.0.2           crayon_1.4.2             
  [7] spatstat.core_2.3-1       MASS_7.3-54               nlme_3.1-153             
 [10] backports_1.3.0           reprex_2.0.1              GOSemSim_2.18.1          
 [13] rlang_0.4.12              XVector_0.32.0            ROCR_1.0-11              
 [16] readxl_1.3.1              irlba_2.3.3               nloptr_1.2.2.3           
 [19] limma_3.48.3              filelock_1.0.2            BiocParallel_1.26.2      
 [22] rjson_0.2.20              bit64_4.0.5               glue_1.5.0               
 [25] sctransform_0.3.2         pbkrtest_0.5.1            vipor_0.4.5              
 [28] spatstat.sparse_2.0-0     spatstat.geom_2.3-0       haven_2.4.3              
 [31] tidyselect_1.1.1          fitdistrplus_1.1-6        variancePartition_1.22.0 
 [34] XML_3.99-0.8              zoo_1.8-9                 xtable_1.8-4             
 [37] evaluate_0.14             cli_3.1.0                 zlibbioc_1.38.0          
 [40] rstudioapi_0.13           miniUI_0.1.1.1            whisker_0.4              
 [43] bslib_0.3.1               rpart_4.1-15              fastmatch_1.1-3          
 [46] treeio_1.16.2             shiny_1.7.1               BiocSingular_1.8.1       
 [49] xfun_0.28                 clue_0.3-60               cluster_2.1.2            
 [52] caTools_1.18.2            tidygraph_1.2.0           KEGGREST_1.32.0          
 [55] ggrepel_0.9.1             ape_5.5                   listenv_0.8.0            
 [58] Biostrings_2.60.2         png_0.1-7                 future_1.23.0            
 [61] withr_2.4.2               bitops_1.0-7              ggforce_0.3.3            
 [64] plyr_1.8.6                cellranger_1.1.0          coda_0.19-4              
 [67] dqrng_0.3.0               pillar_1.6.4              gplots_3.1.1             
 [70] GlobalOptions_0.1.2       cachem_1.0.6              multcomp_1.4-17          
 [73] fs_1.5.0                  GetoptLong_1.0.5          DelayedMatrixStats_1.14.3
 [76] vctrs_0.3.8               ellipsis_0.3.2            generics_0.1.1           
 [79] tools_4.1.0               beeswarm_0.4.0            munsell_0.5.0            
 [82] tweenr_1.0.2              emmeans_1.7.0             DelayedArray_0.18.0      
 [85] fastmap_1.1.0             compiler_4.1.0            abind_1.4-5              
 [88] httpuv_1.6.3              rcartocolor_2.0.0         plotly_4.10.0            
 [91] GenomeInfoDbData_1.2.6    glmmTMB_1.1.2.3           workflowr_1.6.2          
 [94] edgeR_3.34.1              ggnewscale_0.4.5          lattice_0.20-45          
 [97] deldir_1.0-6              utf8_1.2.2                later_1.3.0              
[100] BiocFileCache_2.0.0       jsonlite_1.7.2            scales_1.1.1             
[103] ScaledMatrix_1.0.0        carData_3.0-4             estimability_1.3         
[106] tidytree_0.3.5            pbapply_1.5-0             sparseMatrixStats_1.4.2  
[109] genefilter_1.74.1         lazyeval_0.2.2            promises_1.2.0.1         
[112] car_3.0-12                doParallel_1.0.16         goftest_1.2-3            
[115] spatstat.utils_2.2-0      reticulate_1.22           sandwich_3.0-1           
[118] rmarkdown_2.11            cowplot_1.1.1             blme_1.0-5               
[121] statmod_1.4.36            Rtsne_0.15                downloader_0.4           
[124] uwot_0.1.10               igraph_1.2.8              survival_3.2-13          
[127] numDeriv_2016.8-1.1       yaml_2.2.1                htmltools_0.5.2          
[130] memoise_2.0.0             locfit_1.5-9.4            graphlayouts_0.7.1       
[133] digest_0.6.28             assertthat_0.2.1          mime_0.12                
[136] rappdirs_0.3.3            RSQLite_2.2.8             yulab.utils_0.0.4        
[139] future.apply_1.8.1        data.table_1.14.2         blob_1.2.2               
[142] labeling_0.4.2            splines_4.1.0             Cairo_1.5-12.2           
[145] ggupset_0.3.0             RCurl_1.98-1.5            broom_0.7.10             
[148] hms_1.1.1                 modelr_0.1.8              colorspace_2.0-2         
[151] ggbeeswarm_0.6.0          shape_1.4.6               aplot_0.1.1              
[154] sass_0.4.0                Rcpp_1.0.7                RANN_2.6.1               
[157] mvtnorm_1.1-3             circlize_0.4.13           fansi_0.5.0              
[160] tzdb_0.2.0                parallelly_1.28.1         R6_2.5.1                 
[163] ggridges_0.5.3            lifecycle_1.0.1           bluster_1.2.1            
[166] ggsignif_0.6.3            curl_4.3.2                minqa_1.2.4              
[169] leiden_0.3.9              jquerylib_0.1.4           DO.db_2.9                
[172] Matrix_1.3-4              qvalue_2.24.0             TH.data_1.1-0            
[175] RcppAnnoy_0.0.19          iterators_1.0.13          TMB_1.7.22               
[178] htmlwidgets_1.5.4         beachmat_2.8.1            polyclip_1.10-0          
[181] shadowtext_0.0.9          gridGraphics_0.5-1        rvest_1.0.2              
[184] ComplexHeatmap_2.8.0      mgcv_1.8-38               globals_0.14.0           
[187] patchwork_1.1.1           codetools_0.2-18          lubridate_1.8.0          
[190] GO.db_3.13.0              metapod_1.0.0             gtools_3.9.2             
[193] prettyunits_1.1.1         dbplyr_2.1.1              gtable_0.3.0             
[196] DBI_1.1.1                 git2r_0.28.0              highr_0.9                
[199] ggfun_0.0.4               tensor_1.5                httr_1.4.2               
[202] KernSmooth_2.23-20        vroom_1.5.5               stringi_1.7.5            
[205] progress_1.2.2            reshape2_1.4.4            farver_2.1.0             
[208] annotate_1.70.0           ggtree_3.0.4              xml2_1.3.2               
[211] colorRamps_2.3            boot_1.3-28               BiocNeighbors_1.10.0     
[214] lme4_1.1-27.1             geneplotter_1.70.0        ggplotify_0.1.0          
[217] scattermore_0.7           DESeq2_1.32.0             bit_4.0.4                
[220] scatterpie_0.1.7          spatstat.data_2.1-0       ggraph_2.0.5             
[223] pkgconfig_2.0.3           babelgene_21.4            lmerTest_3.1-3           
[226] rstatix_0.7.0             knitr_1.36               </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Tue Dec  7 13:57:08 2021&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] ggpubr_0.4.0                muscat_1.6.0                msigdbr_7.4.1              
 [4] enrichplot_1.12.3           DOSE_3.18.3                 org.Mm.eg.db_3.13.0        
 [7] AnnotationDbi_1.54.1        clusterProfiler_4.0.5       gridExtra_2.3              
[10] fgsea_1.18.0                biomaRt_2.48.3              pheatmap_1.0.12            
[13] scran_1.20.1                scater_1.20.1               scuttle_1.2.1              
[16] ggsci_2.9                   viridis_0.6.2               viridisLite_0.4.0          
[19] RColorBrewer_1.1-2          SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
[22] Biobase_2.52.0              GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[25] IRanges_2.26.0              S4Vectors_0.30.2            BiocGenerics_0.38.0        
[28] MatrixGenerics_1.4.3        matrixStats_0.61.0          runSeurat3_0.1.0           
[31] here_1.0.1                  magrittr_2.0.1              SeuratObject_4.0.2         
[34] Seurat_4.0.5                forcats_0.5.1               stringr_1.4.0              
[37] dplyr_1.0.7                 purrr_0.3.4                 readr_2.0.2                
[40] tidyr_1.1.4                 tibble_3.1.6                ggplot2_3.3.5              
[43] tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5                ica_1.0-2                 foreach_1.5.1            
  [4] lmtest_0.9-39             rprojroot_2.0.2           crayon_1.4.2             
  [7] spatstat.core_2.3-1       MASS_7.3-54               nlme_3.1-153             
 [10] backports_1.3.0           reprex_2.0.1              GOSemSim_2.18.1          
 [13] rlang_0.4.12              XVector_0.32.0            ROCR_1.0-11              
 [16] readxl_1.3.1              irlba_2.3.3               nloptr_1.2.2.3           
 [19] limma_3.48.3              filelock_1.0.2            BiocParallel_1.26.2      
 [22] rjson_0.2.20              bit64_4.0.5               glue_1.5.0               
 [25] sctransform_0.3.2         pbkrtest_0.5.1            vipor_0.4.5              
 [28] spatstat.sparse_2.0-0     spatstat.geom_2.3-0       haven_2.4.3              
 [31] tidyselect_1.1.1          fitdistrplus_1.1-6        variancePartition_1.22.0 
 [34] XML_3.99-0.8              zoo_1.8-9                 xtable_1.8-4             
 [37] evaluate_0.14             cli_3.1.0                 zlibbioc_1.38.0          
 [40] rstudioapi_0.13           miniUI_0.1.1.1            whisker_0.4              
 [43] bslib_0.3.1               rpart_4.1-15              fastmatch_1.1-3          
 [46] treeio_1.16.2             shiny_1.7.1               BiocSingular_1.8.1       
 [49] xfun_0.28                 clue_0.3-60               cluster_2.1.2            
 [52] caTools_1.18.2            tidygraph_1.2.0           KEGGREST_1.32.0          
 [55] ggrepel_0.9.1             ape_5.5                   listenv_0.8.0            
 [58] Biostrings_2.60.2         png_0.1-7                 future_1.23.0            
 [61] withr_2.4.2               bitops_1.0-7              ggforce_0.3.3            
 [64] plyr_1.8.6                cellranger_1.1.0          coda_0.19-4              
 [67] dqrng_0.3.0               pillar_1.6.4              gplots_3.1.1             
 [70] GlobalOptions_0.1.2       cachem_1.0.6              multcomp_1.4-17          
 [73] fs_1.5.0                  GetoptLong_1.0.5          DelayedMatrixStats_1.14.3
 [76] vctrs_0.3.8               ellipsis_0.3.2            generics_0.1.1           
 [79] tools_4.1.0               beeswarm_0.4.0            munsell_0.5.0            
 [82] tweenr_1.0.2              emmeans_1.7.0             DelayedArray_0.18.0      
 [85] fastmap_1.1.0             compiler_4.1.0            abind_1.4-5              
 [88] httpuv_1.6.3              rcartocolor_2.0.0         plotly_4.10.0            
 [91] GenomeInfoDbData_1.2.6    glmmTMB_1.1.2.3           workflowr_1.6.2          
 [94] edgeR_3.34.1              ggnewscale_0.4.5          lattice_0.20-45          
 [97] deldir_1.0-6              utf8_1.2.2                later_1.3.0              
[100] BiocFileCache_2.0.0       jsonlite_1.7.2            scales_1.1.1             
[103] ScaledMatrix_1.0.0        carData_3.0-4             estimability_1.3         
[106] tidytree_0.3.5            pbapply_1.5-0             sparseMatrixStats_1.4.2  
[109] genefilter_1.74.1         lazyeval_0.2.2            promises_1.2.0.1         
[112] car_3.0-12                doParallel_1.0.16         goftest_1.2-3            
[115] spatstat.utils_2.2-0      reticulate_1.22           sandwich_3.0-1           
[118] rmarkdown_2.11            cowplot_1.1.1             blme_1.0-5               
[121] statmod_1.4.36            Rtsne_0.15                downloader_0.4           
[124] uwot_0.1.10               igraph_1.2.8              survival_3.2-13          
[127] numDeriv_2016.8-1.1       yaml_2.2.1                htmltools_0.5.2          
[130] memoise_2.0.0             locfit_1.5-9.4            graphlayouts_0.7.1       
[133] digest_0.6.28             assertthat_0.2.1          mime_0.12                
[136] rappdirs_0.3.3            RSQLite_2.2.8             yulab.utils_0.0.4        
[139] future.apply_1.8.1        data.table_1.14.2         blob_1.2.2               
[142] labeling_0.4.2            splines_4.1.0             Cairo_1.5-12.2           
[145] ggupset_0.3.0             RCurl_1.98-1.5            broom_0.7.10             
[148] hms_1.1.1                 modelr_0.1.8              colorspace_2.0-2         
[151] ggbeeswarm_0.6.0          shape_1.4.6               aplot_0.1.1              
[154] sass_0.4.0                Rcpp_1.0.7                RANN_2.6.1               
[157] mvtnorm_1.1-3             circlize_0.4.13           fansi_0.5.0              
[160] tzdb_0.2.0                parallelly_1.28.1         R6_2.5.1                 
[163] ggridges_0.5.3            lifecycle_1.0.1           bluster_1.2.1            
[166] ggsignif_0.6.3            curl_4.3.2                minqa_1.2.4              
[169] leiden_0.3.9              jquerylib_0.1.4           DO.db_2.9                
[172] Matrix_1.3-4              qvalue_2.24.0             TH.data_1.1-0            
[175] RcppAnnoy_0.0.19          iterators_1.0.13          TMB_1.7.22               
[178] htmlwidgets_1.5.4         beachmat_2.8.1            polyclip_1.10-0          
[181] shadowtext_0.0.9          gridGraphics_0.5-1        rvest_1.0.2              
[184] ComplexHeatmap_2.8.0      mgcv_1.8-38               globals_0.14.0           
[187] patchwork_1.1.1           codetools_0.2-18          lubridate_1.8.0          
[190] GO.db_3.13.0              metapod_1.0.0             gtools_3.9.2             
[193] prettyunits_1.1.1         dbplyr_2.1.1              gtable_0.3.0             
[196] DBI_1.1.1                 git2r_0.28.0              highr_0.9                
[199] ggfun_0.0.4               tensor_1.5                httr_1.4.2               
[202] KernSmooth_2.23-20        vroom_1.5.5               stringi_1.7.5            
[205] progress_1.2.2            reshape2_1.4.4            farver_2.1.0             
[208] annotate_1.70.0           ggtree_3.0.4              xml2_1.3.2               
[211] colorRamps_2.3            boot_1.3-28               BiocNeighbors_1.10.0     
[214] lme4_1.1-27.1             geneplotter_1.70.0        ggplotify_0.1.0          
[217] scattermore_0.7           DESeq2_1.32.0             bit_4.0.4                
[220] scatterpie_0.1.7          spatstat.data_2.1-0       ggraph_2.0.5             
[223] pkgconfig_2.0.3           babelgene_21.4            lmerTest_3.1-3           
[226] rstatix_0.7.0             knitr_1.36               </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
