<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mechthild Lütge" />

<meta name="date" content="2020-05-14" />

<title>organ specific signatures - BRCs only</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSLO_BRC_CXCL13</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">organ specific signatures - BRCs only</h1>
<h4 class="author">Mechthild Lütge</h4>
<h4 class="date">14 May 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSLO_BRC_CXCL13/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210519code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210519)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210519code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210519)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree1f27cea5638ec11795ed815300ef71869e6d3b0btargetblank1f27ceaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/1f27cea5638ec11795ed815300ef71869e6d3b0b" target="_blank">1f27cea</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13tree1f27cea5638ec11795ed815300ef71869e6d3b0btargetblank1f27ceaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/1f27cea5638ec11795ed815300ef71869e6d3b0b" target="_blank">1f27cea</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/destiny/
    Ignored:    data/processedData/

Untracked files:
    Untracked:  analysis/SubsetSpecificGeneSignatures.Rmd

Unstaged changes:
    Modified:   analysis/CharacterizePP_CXCL13seurat.Rmd
    Modified:   analysis/CharacterizeSpleen_CXCL13seurat.Rmd
    Modified:   analysis/IntegrateAcrossSLOs.Rmd
    Modified:   analysis/SLOspecific_signature.Rmd
    Modified:   analysis/visFolBRC_merged.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(tidyverse)
  library(Seurat)
  library(magrittr)
  library(dplyr)
  library(purrr)
  library(ggplot2)
  library(here)
  library(runSeurat3)
  library(SingleCellExperiment)
  library(RColorBrewer)
  library(viridis)
  library(ggsci)
  library(scater)
  library(scran)
  library(pheatmap)
  library(biomaRt)
  library(fgsea)
  library(grid)
  library(gridExtra)
  library(clusterProfiler)
  library(org.Mm.eg.db)
  library(DOSE)
  library(enrichplot)
  library(msigdbr)
  library(muscat)
})</code></pre>
</div>
<div id="plotting-funct" class="section level2">
<h2>plotting funct</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, colVecIdent, colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  ## format gene names (depends on how gene list is inputed)
  selGenes &lt;- selGenes %&gt;% 
      mutate(geneID = (str_split(gene, &#39;\\.&#39;, simplify = T)[,2]))
  #selGenes &lt;- selGenes$labelNam
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;%
    filter(geneID %in% selGenes$geneID)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes$geneID,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=&quot;row&quot; ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc,
         color = colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(50),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
</div>
<div id="set-dir-and-read-input-data" class="section level2">
<h2>set dir and read input data</h2>
<pre class="r"><code>basedir &lt;- here()

## read seurat objects with BRCs
seurat &lt;- readRDS(file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))
## seurat &lt;-rerunSeurat3(seurat)
## saveRDS(seurat, file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color Vectors</h2>
<pre class="r"><code>colPal &lt;- c(&quot;#87c5af&quot;, &quot;#94033b&quot;, &quot;#fc8f3b&quot;, &quot;#2580fe&quot;, &quot;#514e72&quot;)
names(colPal) &lt;- c(&quot;LZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;DZFDC&quot;,&quot;PRC2&quot;)


colorSLO &lt;- c(viridis(3)[c(1,2)], brewer.pal(n=9, name=&quot;YlOrRd&quot;)[c(4)])
colorCond &lt;- brewer.pal(n=3, name=&quot;Paired&quot;)[c(1,2)]
colorSLOCond &lt;- c(&quot;#440154FF&quot;,&quot;#807DBA&quot;, &quot;#FEB24C&quot;, &quot;#F16913&quot;, &quot;#21908CFF&quot;)
colorBatch &lt;- pal_igv()(n=length(unique(seurat$batch)))
colCl &lt;- c(rcartocolor::carto_pal(name=&quot;Safe&quot;),pal_aaas()(8))

names(colorSLO) &lt;- c(&quot;LN&quot;, &quot;PP&quot;, &quot;Spleen&quot;)
names(colorCond) &lt;- c(&quot;naive&quot;, &quot;immunized&quot;)
names(colorBatch) &lt;- unique(seurat$batch)
names(colorSLOCond) &lt;- c(&quot;LN_immunized&quot;,&quot;LN_naive&quot;, &quot;Spleen_naive&quot;,
                         &quot;Spleen_immunized&quot;, &quot;PP_immunized&quot;)</code></pre>
</div>
<div id="visualize-data" class="section level2 tabset">
<h2 class="tabset">visualize data</h2>
<div id="clustering" class="section level3">
<h3>clustering</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colPal, group.by = &quot;clusterLabel&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="slo" class="section level3">
<h3>SLO</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLO, group.by = &quot;SLO&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/SLO-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLO, group.by = &quot;SLO&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/SLO-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="batch" class="section level3">
<h3>Batch</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorBatch, group.by = &quot;batch&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/Batch-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cond" class="section level3">
<h3>Cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorCond, group.by = &quot;cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/Cond-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="slo-plus-cond" class="section level3">
<h3>SLO plus cond</h3>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;, cols=colorSLOCond,group.by=&quot;SLO_plus_cond&quot;)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/SLO%20plus%20cond-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seurat, reduction = &quot;umap&quot;,cols=colorSLOCond,group.by = &quot;SLO_plus_cond&quot;,
        pt.size=0.6)+
  theme_void()</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/SLO%20plus%20cond-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="subset-specific-marker" class="section level2">
<h2>subset specific marker</h2>
<pre class="r"><code>Idents(seurat) &lt;- seurat$clusterLabel
seurat$SLO_clusterLabel &lt;- paste0(seurat$SLO, &quot;_&quot;, Idents(seurat))

seurat$clusterLabel &lt;- factor(seurat$clusterLabel,
                              levels=c(&quot;LZFDC&quot;, &quot;DZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;PRC2&quot;))

clVec &lt;- levels(seurat$clusterLabel)
conservedMarker &lt;- lapply(clVec, function(cl){
  markers &lt;- FindConservedMarkers(seurat, ident.1 = cl, ident.2 = NULL,
                                  grouping.var = &quot;SLO&quot;, only.pos =T,
                                  assay = &quot;originalexp&quot;) %&gt;% 
    mutate(subset=cl)
})
names(conservedMarker) &lt;- clVec</code></pre>
<div id="sc-heatmap-slo-specific-marker" class="section level3">
<h3>sc heatmap SLO specific marker</h3>
<pre class="r"><code>conservedMarkerDat &lt;- do.call(&quot;rbind&quot;, conservedMarker) %&gt;% 
  rownames_to_column(., var=&quot;long&quot;) %&gt;% 
  mutate(ensID=gsub(&quot;^[^\\.]*\\.&quot;, &quot;&quot;, long)) %&gt;% 
  mutate(label=gsub(&quot;^.*\\.&quot;, &quot;&quot;, ensID)) %&gt;% 
  filter(PP_p_val_adj &lt; 0.01 &amp; Spleen_p_val_adj &lt; 0.01 &amp; LN_p_val_adj &lt; 0.01) %&gt;% 
  filter(!duplicated(ensID))


DoHeatmap(seurat, features = conservedMarkerDat$ensID, group.by = &quot;clusterLabel&quot;, 
          group.colors = colPal, slot = &#39;scale.data&#39;, label = T,
          disp.min = -1, disp.max = 1.5, draw.lines = F) +
  scale_fill_gradientn(colors=colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(30)) +
  scale_y_discrete(breaks=conservedMarkerDat$ensID, labels=conservedMarkerDat$label)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/sc%20heatmap%20slo%20specific-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="only-markers-that-diff-up-in-all-pairwise-comparisons" class="section level3">
<h3>only markers that diff up in all pairwise comparisons</h3>
<pre class="r"><code>clVec &lt;- levels(seurat$clusterLabel)
highlyConservedMarker &lt;- lapply(clVec, function(cl){
  otherCl &lt;- clVec[which(clVec!=cl)]
  consMarker &lt;- NULL
  for(i in 1:length(otherCl)){
    cl2 &lt;- otherCl[i]
    markers &lt;- FindMarkers(seurat, ident.1 = cl, ident.2 = cl2,
                           only.pos = T) %&gt;%
      filter(p_val_adj &lt;0.01) %&gt;% 
      rownames_to_column(., var = &quot;gene&quot;)
    if(is.null(consMarker)){
      consMarker &lt;- markers$gene
    }
    consMarker &lt;- consMarker[which(consMarker %in% markers$gene)]
  }
  consMarkerDat &lt;- data.frame(gene=consMarker) %&gt;% 
    mutate(subset=cl)
  return(consMarkerDat)
})
names(highlyConservedMarker) &lt;- clVec


hConservedMarkerDat &lt;- do.call(&quot;rbind&quot;, highlyConservedMarker) %&gt;% 
  mutate(label=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) 
seurat$label_plus_SLO &lt;- paste0(seurat$clusterLabel, &quot;_&quot;, seurat$SLO)
DoHeatmap(seurat, features = hConservedMarkerDat$gene, group.by = &quot;clusterLabel&quot;, 
          group.colors = colPal, slot = &#39;scale.data&#39;, label = T,
          disp.min = -1, disp.max = 1.5, draw.lines = F) +
  scale_fill_gradientn(colors=colorRampPalette(c(&quot;#45628f&quot;, &quot;#F7F7F7&quot;, &quot;#de425b&quot;))(30)) +
  scale_y_discrete(breaks=hConservedMarkerDat$gene, labels=hConservedMarkerDat$label)</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/highly%20conserved%20makers-1.png" width="1344" style="display: block; margin: auto;" /> ## GSEA clusterProfiler {.tabset} ### cons genes</p>
<pre class="r"><code>clVec &lt;- levels(seurat$clusterLabel)
GOcons &lt;- lapply(clVec, function(cl){
  conservedMarkerDatSub &lt;- conservedMarkerDat %&gt;% 
    filter(subset == cl) %&gt;% mutate(ENS=gsub(&quot;\\..*$&quot;, &quot;&quot;, ensID)) %&gt;% 
    slice_min(., max_pval, n=200)
  egoSS &lt;- enrichGO(gene      = unique(conservedMarkerDatSub$ENS),
                OrgDb         = org.Mm.eg.db,
                keyType       = &#39;ENSEMBL&#39;,
                ont           = &quot;BP&quot;,
                pAdjustMethod = &quot;BH&quot;,
                pvalueCutoff  = 0.05,
                qvalueCutoff  = 0.05)
  egoSS &lt;- setReadable(egoSS, OrgDb = org.Mm.eg.db)
  egoSSres &lt;- egoSS@result %&gt;% filter(p.adjust &lt; 0.05) %&gt;% 
    mutate(subset=cl)
})

names(GOcons) &lt;- clVec</code></pre>
</div>
<div id="vis-summary-all" class="section level3">
<h3>vis summary all</h3>
<pre class="r"><code>selGO &lt;- c(&quot;LZFDC.GO:0001818&quot;,&quot;LZFDC.GO:0007159&quot;,&quot;LZFDC.GO:0032653&quot;,
           &quot;LZFDC.GO:0002822&quot;,&quot;LZFDC.GO:0070661&quot;,
           &quot;DZFDC.GO:0002688&quot;, &quot;DZFDC.GO:0048259&quot;, &quot;DZFDC.GO:0030336&quot;,
           &quot;MRC.GO:0050673&quot;, &quot;MRC.GO:0007015&quot;,
           &quot;TBRC.GO:005092&quot;, &quot;TBRC.GO:0035455&quot;, &quot;TBRC.GO:0031295&quot;,
           &quot;TBRC.GO:0006959&quot;,&quot;TBRC.GO:0032612&quot;,
           &quot;PRC2.GO:0030198&quot;, &quot;PRC2.GO:0035265&quot;, &quot;PRC2.GO:0031589&quot;,
           &quot;PRC2.GO:0032755&quot;, &quot;PRC2.GO:0032640&quot;)


GOconsDat &lt;- do.call(&quot;rbind&quot;, GOcons) %&gt;% rownames_to_column(., var=&quot;IDlab&quot;) %&gt;% 
  filter(IDlab %in% selGO) %&gt;% 
  arrange(., Count) %&gt;% arrange(., subset) %&gt;%
  mutate(nom=as.numeric(gsub(&quot;\\/.*&quot;, &quot;&quot;, GeneRatio))) %&gt;% 
  mutate(denom=as.numeric(gsub(&quot;.*\\/&quot;, &quot;&quot;, GeneRatio))) %&gt;% 
  mutate(geneRatio=nom/denom) %&gt;% 
  mutate(one_minPadjust=(1-qvalue))


level_order &lt;- unique(GOconsDat$Description)
ggplot(GOconsDat, aes(x=factor(Description, level= level_order), y=geneRatio)) + 
  geom_point(aes(col=subset, size=Count, alpha=one_minPadjust)) + 
  scale_color_manual(values = colPal)+
  scale_alpha_continuous(range = c(0.5,1), name=&quot;qvalue&quot;,
                         labels=c(0.04, 0.03, 0.02, 0.01),
                         breaks = c(0.96, 0.97, 0.98, 0.99)) +
  scale_size_continuous(range = c(3,6), labels = c(2,4,6,8),
                        breaks = c(2,4,6,8)) +
  geom_segment(aes(x=Description, 
                   xend=Description, 
                   y=min(geneRatio), 
                   yend=max(geneRatio)), 
               linetype=&quot;dashed&quot;, 
               size=0.1) +   # Draw dashed lines
  coord_flip() +
  theme_minimal() +
  theme(panel.border = element_rect(fill = NA), 
        panel.grid = element_blank(), axis.title.y = element_blank())</code></pre>
<p><img src="figure/SubsetSpecificGeneSignatures.Rmd/vis%20summary-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] muscat_1.6.0                msigdbr_7.4.1               enrichplot_1.12.2          
 [4] DOSE_3.18.1                 org.Mm.eg.db_3.13.0         AnnotationDbi_1.54.1       
 [7] clusterProfiler_4.0.2       gridExtra_2.3               fgsea_1.18.0               
[10] biomaRt_2.48.2              pheatmap_1.0.12             scran_1.20.1               
[13] scater_1.20.1               scuttle_1.2.0               ggsci_2.9                  
[16] viridis_0.6.1               viridisLite_0.4.0           RColorBrewer_1.1-2         
[19] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0 Biobase_2.52.0             
[22] GenomicRanges_1.44.0        GenomeInfoDb_1.28.1         IRanges_2.26.0             
[25] S4Vectors_0.30.0            BiocGenerics_0.38.0         MatrixGenerics_1.4.0       
[28] matrixStats_0.60.0          runSeurat3_0.1.0            here_1.0.1                 
[31] magrittr_2.0.1              SeuratObject_4.0.2          Seurat_4.0.3               
[34] forcats_0.5.1               stringr_1.4.0               dplyr_1.0.7                
[37] purrr_0.3.4                 readr_2.0.0                 tidyr_1.1.3                
[40] tibble_3.1.3                ggplot2_3.3.5               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5                ica_1.0-2                 foreach_1.5.1            
  [4] lmtest_0.9-38             rprojroot_2.0.2           crayon_1.4.1             
  [7] rbibutils_2.2.1           spatstat.core_2.3-0       MASS_7.3-54              
 [10] nlme_3.1-152              backports_1.2.1           reprex_2.0.0             
 [13] GOSemSim_2.18.0           rlang_0.4.11              XVector_0.32.0           
 [16] ROCR_1.0-11               readxl_1.3.1              irlba_2.3.3              
 [19] nloptr_1.2.2.2            limma_3.48.1              filelock_1.0.2           
 [22] BiocParallel_1.26.1       rjson_0.2.20              bit64_4.0.5              
 [25] glue_1.4.2                sctransform_0.3.2         pbkrtest_0.5.1           
 [28] vipor_0.4.5               spatstat.sparse_2.0-0     spatstat.geom_2.2-2      
 [31] haven_2.4.1               tidyselect_1.1.1          fitdistrplus_1.1-5       
 [34] variancePartition_1.22.0  XML_3.99-0.6              zoo_1.8-9                
 [37] xtable_1.8-4              evaluate_0.14             Rdpack_2.1.2             
 [40] cli_3.0.1                 zlibbioc_1.38.0           sn_2.0.0                 
 [43] rstudioapi_0.13           miniUI_0.1.1.1            bslib_0.2.5.1            
 [46] rpart_4.1-15              mathjaxr_1.4-0            fastmatch_1.1-3          
 [49] treeio_1.16.1             shiny_1.6.0               BiocSingular_1.8.1       
 [52] xfun_0.24                 clue_0.3-59               multtest_2.48.0          
 [55] cluster_2.1.2             caTools_1.18.2            tidygraph_1.2.0          
 [58] KEGGREST_1.32.0           ggrepel_0.9.1             ape_5.5                  
 [61] listenv_0.8.0             Biostrings_2.60.1         png_0.1-7                
 [64] future_1.21.0             withr_2.4.2               bitops_1.0-7             
 [67] ggforce_0.3.3             plyr_1.8.6                cellranger_1.1.0         
 [70] coda_0.19-4               dqrng_0.3.0               pillar_1.6.1             
 [73] gplots_3.1.1              GlobalOptions_0.1.2       cachem_1.0.5             
 [76] multcomp_1.4-17           fs_1.5.0                  GetoptLong_1.0.5         
 [79] DelayedMatrixStats_1.14.0 vctrs_0.3.8               ellipsis_0.3.2           
 [82] generics_0.1.0            metap_1.4                 tools_4.1.0              
 [85] beeswarm_0.4.0            munsell_0.5.0             tweenr_1.0.2             
 [88] emmeans_1.6.2-1           DelayedArray_0.18.0       fastmap_1.1.0            
 [91] compiler_4.1.0            abind_1.4-5               httpuv_1.6.1             
 [94] rcartocolor_2.0.0         plotly_4.9.4.1            GenomeInfoDbData_1.2.6   
 [97] glmmTMB_1.1.2             workflowr_1.6.2           edgeR_3.34.0             
[100] lattice_0.20-44           deldir_0.2-10             mutoss_0.1-12            
[103] utf8_1.2.2                later_1.2.0               BiocFileCache_2.0.0      
[106] jsonlite_1.7.2            scales_1.1.1              ScaledMatrix_1.0.0       
[109] tidytree_0.3.4            pbapply_1.4-3             sparseMatrixStats_1.4.0  
[112] estimability_1.3          genefilter_1.74.0         lazyeval_0.2.2           
[115] promises_1.2.0.1          doParallel_1.0.16         goftest_1.2-2            
[118] spatstat.utils_2.2-0      reticulate_1.20           sandwich_3.0-1           
[121] rmarkdown_2.9             cowplot_1.1.1             blme_1.0-5               
[124] statmod_1.4.36            Rtsne_0.15                downloader_0.4           
[127] uwot_0.1.10               igraph_1.2.6              plotrix_3.8-1            
[130] survival_3.2-11           numDeriv_2016.8-1.1       yaml_2.2.1               
[133] htmltools_0.5.1.1         memoise_2.0.0             locfit_1.5-9.4           
[136] graphlayouts_0.7.1        digest_0.6.27             assertthat_0.2.1         
[139] mime_0.11                 rappdirs_0.3.3            RSQLite_2.2.7            
[142] future.apply_1.7.0        data.table_1.14.0         blob_1.2.2               
[145] labeling_0.4.2            splines_4.1.0             Cairo_1.5-12.2           
[148] RCurl_1.98-1.3            broom_0.7.8               hms_1.1.0                
[151] modelr_0.1.8              colorspace_2.0-2          mnormt_2.0.2             
[154] BiocManager_1.30.16       tmvnsim_1.0-2             ggbeeswarm_0.6.0         
[157] shape_1.4.6               aplot_0.0.6               sass_0.4.0               
[160] Rcpp_1.0.7                RANN_2.6.1                mvtnorm_1.1-2            
[163] circlize_0.4.13           fansi_0.5.0               tzdb_0.1.2               
[166] parallelly_1.27.0         R6_2.5.0                  ggridges_0.5.3           
[169] lifecycle_1.0.0           TFisher_0.2.0             bluster_1.2.1            
[172] curl_4.3.2                minqa_1.2.4               leiden_0.3.8             
[175] jquerylib_0.1.4           DO.db_2.9                 Matrix_1.3-4             
[178] qvalue_2.24.0             TH.data_1.0-10            RcppAnnoy_0.0.18         
[181] iterators_1.0.13          TMB_1.7.20                htmlwidgets_1.5.3        
[184] beachmat_2.8.0            polyclip_1.10-0           shadowtext_0.0.8         
[187] rvest_1.0.1               ComplexHeatmap_2.8.0      mgcv_1.8-36              
[190] globals_0.14.0            patchwork_1.1.1           codetools_0.2-18         
[193] lubridate_1.7.10          GO.db_3.13.0              metapod_1.0.0            
[196] gtools_3.9.2              prettyunits_1.1.1         dbplyr_2.1.1             
[199] gtable_0.3.0              DBI_1.1.1                 git2r_0.28.0             
[202] highr_0.9                 tensor_1.5                httr_1.4.2               
[205] KernSmooth_2.23-20        stringi_1.7.3             progress_1.2.2           
[208] reshape2_1.4.4            farver_2.1.0              annotate_1.70.0          
[211] ggtree_3.0.2              xml2_1.3.2                colorRamps_2.3           
[214] rvcheck_0.1.8             boot_1.3-28               BiocNeighbors_1.10.0     
[217] lme4_1.1-27.1             geneplotter_1.70.0        scattermore_0.7          
[220] DESeq2_1.32.0             bit_4.0.4                 scatterpie_0.1.6         
[223] spatstat.data_2.1-0       ggraph_2.0.5              pkgconfig_2.0.3          
[226] babelgene_21.4            lmerTest_3.1-3            knitr_1.33               </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Jul 28 14:02:20 2021&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[10] base     

other attached packages:
 [1] muscat_1.6.0                msigdbr_7.4.1               enrichplot_1.12.2          
 [4] DOSE_3.18.1                 org.Mm.eg.db_3.13.0         AnnotationDbi_1.54.1       
 [7] clusterProfiler_4.0.2       gridExtra_2.3               fgsea_1.18.0               
[10] biomaRt_2.48.2              pheatmap_1.0.12             scran_1.20.1               
[13] scater_1.20.1               scuttle_1.2.0               ggsci_2.9                  
[16] viridis_0.6.1               viridisLite_0.4.0           RColorBrewer_1.1-2         
[19] SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0 Biobase_2.52.0             
[22] GenomicRanges_1.44.0        GenomeInfoDb_1.28.1         IRanges_2.26.0             
[25] S4Vectors_0.30.0            BiocGenerics_0.38.0         MatrixGenerics_1.4.0       
[28] matrixStats_0.60.0          runSeurat3_0.1.0            here_1.0.1                 
[31] magrittr_2.0.1              SeuratObject_4.0.2          Seurat_4.0.3               
[34] forcats_0.5.1               stringr_1.4.0               dplyr_1.0.7                
[37] purrr_0.3.4                 readr_2.0.0                 tidyr_1.1.3                
[40] tibble_3.1.3                ggplot2_3.3.5               tidyverse_1.3.1            

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5                ica_1.0-2                 foreach_1.5.1            
  [4] lmtest_0.9-38             rprojroot_2.0.2           crayon_1.4.1             
  [7] rbibutils_2.2.1           spatstat.core_2.3-0       MASS_7.3-54              
 [10] nlme_3.1-152              backports_1.2.1           reprex_2.0.0             
 [13] GOSemSim_2.18.0           rlang_0.4.11              XVector_0.32.0           
 [16] ROCR_1.0-11               readxl_1.3.1              irlba_2.3.3              
 [19] nloptr_1.2.2.2            limma_3.48.1              filelock_1.0.2           
 [22] BiocParallel_1.26.1       rjson_0.2.20              bit64_4.0.5              
 [25] glue_1.4.2                sctransform_0.3.2         pbkrtest_0.5.1           
 [28] vipor_0.4.5               spatstat.sparse_2.0-0     spatstat.geom_2.2-2      
 [31] haven_2.4.1               tidyselect_1.1.1          fitdistrplus_1.1-5       
 [34] variancePartition_1.22.0  XML_3.99-0.6              zoo_1.8-9                
 [37] xtable_1.8-4              evaluate_0.14             Rdpack_2.1.2             
 [40] cli_3.0.1                 zlibbioc_1.38.0           sn_2.0.0                 
 [43] rstudioapi_0.13           miniUI_0.1.1.1            bslib_0.2.5.1            
 [46] rpart_4.1-15              mathjaxr_1.4-0            fastmatch_1.1-3          
 [49] treeio_1.16.1             shiny_1.6.0               BiocSingular_1.8.1       
 [52] xfun_0.24                 clue_0.3-59               multtest_2.48.0          
 [55] cluster_2.1.2             caTools_1.18.2            tidygraph_1.2.0          
 [58] KEGGREST_1.32.0           ggrepel_0.9.1             ape_5.5                  
 [61] listenv_0.8.0             Biostrings_2.60.1         png_0.1-7                
 [64] future_1.21.0             withr_2.4.2               bitops_1.0-7             
 [67] ggforce_0.3.3             plyr_1.8.6                cellranger_1.1.0         
 [70] coda_0.19-4               dqrng_0.3.0               pillar_1.6.1             
 [73] gplots_3.1.1              GlobalOptions_0.1.2       cachem_1.0.5             
 [76] multcomp_1.4-17           fs_1.5.0                  GetoptLong_1.0.5         
 [79] DelayedMatrixStats_1.14.0 vctrs_0.3.8               ellipsis_0.3.2           
 [82] generics_0.1.0            metap_1.4                 tools_4.1.0              
 [85] beeswarm_0.4.0            munsell_0.5.0             tweenr_1.0.2             
 [88] emmeans_1.6.2-1           DelayedArray_0.18.0       fastmap_1.1.0            
 [91] compiler_4.1.0            abind_1.4-5               httpuv_1.6.1             
 [94] rcartocolor_2.0.0         plotly_4.9.4.1            GenomeInfoDbData_1.2.6   
 [97] glmmTMB_1.1.2             workflowr_1.6.2           edgeR_3.34.0             
[100] lattice_0.20-44           deldir_0.2-10             mutoss_0.1-12            
[103] utf8_1.2.2                later_1.2.0               BiocFileCache_2.0.0      
[106] jsonlite_1.7.2            scales_1.1.1              ScaledMatrix_1.0.0       
[109] tidytree_0.3.4            pbapply_1.4-3             sparseMatrixStats_1.4.0  
[112] estimability_1.3          genefilter_1.74.0         lazyeval_0.2.2           
[115] promises_1.2.0.1          doParallel_1.0.16         goftest_1.2-2            
[118] spatstat.utils_2.2-0      reticulate_1.20           sandwich_3.0-1           
[121] rmarkdown_2.9             cowplot_1.1.1             blme_1.0-5               
[124] statmod_1.4.36            Rtsne_0.15                downloader_0.4           
[127] uwot_0.1.10               igraph_1.2.6              plotrix_3.8-1            
[130] survival_3.2-11           numDeriv_2016.8-1.1       yaml_2.2.1               
[133] htmltools_0.5.1.1         memoise_2.0.0             locfit_1.5-9.4           
[136] graphlayouts_0.7.1        digest_0.6.27             assertthat_0.2.1         
[139] mime_0.11                 rappdirs_0.3.3            RSQLite_2.2.7            
[142] future.apply_1.7.0        data.table_1.14.0         blob_1.2.2               
[145] labeling_0.4.2            splines_4.1.0             Cairo_1.5-12.2           
[148] RCurl_1.98-1.3            broom_0.7.8               hms_1.1.0                
[151] modelr_0.1.8              colorspace_2.0-2          mnormt_2.0.2             
[154] BiocManager_1.30.16       tmvnsim_1.0-2             ggbeeswarm_0.6.0         
[157] shape_1.4.6               aplot_0.0.6               sass_0.4.0               
[160] Rcpp_1.0.7                RANN_2.6.1                mvtnorm_1.1-2            
[163] circlize_0.4.13           fansi_0.5.0               tzdb_0.1.2               
[166] parallelly_1.27.0         R6_2.5.0                  ggridges_0.5.3           
[169] lifecycle_1.0.0           TFisher_0.2.0             bluster_1.2.1            
[172] curl_4.3.2                minqa_1.2.4               leiden_0.3.8             
[175] jquerylib_0.1.4           DO.db_2.9                 Matrix_1.3-4             
[178] qvalue_2.24.0             TH.data_1.0-10            RcppAnnoy_0.0.18         
[181] iterators_1.0.13          TMB_1.7.20                htmlwidgets_1.5.3        
[184] beachmat_2.8.0            polyclip_1.10-0           shadowtext_0.0.8         
[187] rvest_1.0.1               ComplexHeatmap_2.8.0      mgcv_1.8-36              
[190] globals_0.14.0            patchwork_1.1.1           codetools_0.2-18         
[193] lubridate_1.7.10          GO.db_3.13.0              metapod_1.0.0            
[196] gtools_3.9.2              prettyunits_1.1.1         dbplyr_2.1.1             
[199] gtable_0.3.0              DBI_1.1.1                 git2r_0.28.0             
[202] highr_0.9                 tensor_1.5                httr_1.4.2               
[205] KernSmooth_2.23-20        stringi_1.7.3             progress_1.2.2           
[208] reshape2_1.4.4            farver_2.1.0              annotate_1.70.0          
[211] ggtree_3.0.2              xml2_1.3.2                colorRamps_2.3           
[214] rvcheck_0.1.8             boot_1.3-28               BiocNeighbors_1.10.0     
[217] lme4_1.1-27.1             geneplotter_1.70.0        scattermore_0.7          
[220] DESeq2_1.32.0             bit_4.0.4                 scatterpie_0.1.6         
[223] spatstat.data_2.1-0       ggraph_2.0.5              pkgconfig_2.0.3          
[226] babelgene_21.4            lmerTest_3.1-3            knitr_1.33               </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
