<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>create input files for cellphonedb</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CrossSLO_BRC_CXCL13</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">create input files for cellphonedb</h1>
<h4 class="author"></h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>CrossSLO_BRC_CXCL13/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210519code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210519)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210519code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210519)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13treeb2e5a81907d4d650867b04e8d061cb72989e04bftargetblankb2e5a81a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/b2e5a81907d4d650867b04e8d061cb72989e04bf" target="_blank">b2e5a81</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommluetgeCrossSLOBRCCXCL13treeb2e5a81907d4d650867b04e8d061cb72989e04bftargetblankb2e5a81a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/tree/b2e5a81907d4d650867b04e8d061cb72989e04bf" target="_blank">b2e5a81</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/GSEA/
    Ignored:    data/cellphonedb/
    Ignored:    data/destiny/
    Ignored:    data/lympho/
    Ignored:    data/processedData/
    Ignored:    data/processedDataLymphocytes/
    Ignored:    data/vision/
    Ignored:    figure/.DS_Store
    Ignored:    figure/plotSubsetSpecificSignatures.Rmd/
    Ignored:    figure/plotSubsetSpecificSignatures2.Rmd/.DS_Store

Unstaged changes:
    Modified:   analysis/DownstreamSignalingSignAverageExpression.Rmd
    Modified:   analysis/visFolBRC_merged.Rmd
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-6.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-6.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-6.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-10.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-11.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-12.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-6.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-7.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-8.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-9.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-1.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-10.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-11.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-12.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-2.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-3.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-4.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-5.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-6.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-7.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-8.pdf
    Modified:   figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-9.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/visSelIntHeatmapSignaling.Rmd</code>) and HTML (<code>docs/visSelIntHeatmapSignaling.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/visSelIntHeatmapSignaling.html" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
<td>
update plots
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/analysis/visSelIntHeatmapSignaling.Rmd" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
<td>
update interaction analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/visSelIntHeatmapSignaling.html" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
<td>
update interaction analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/analysis/visSelIntHeatmapSignaling.Rmd" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
<td>
add signaling analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mluetge/CrossSLO_BRC_CXCL13/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/visSelIntHeatmapSignaling.html" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
<td>
add signaling analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>suppressPackageStartupMessages({
  library(scater)
  library(dplyr)
  library(reshape2)
  library(Matrix)
  library(purrr)
  library(scran)
  library(Seurat)
  library(mclust)
  library(tidyverse)
  library(here)
  library(viridis)
  library(RColorBrewer)
  library(pheatmap)
})</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color Vectors</h2>
<pre class="r"><code>colBRC &lt;- c(&quot;#87c5af&quot;, &quot;#94033b&quot;, &quot;#fc8f3b&quot;, &quot;#2580fe&quot;, &quot;#514e72&quot;)
names(colBRC) &lt;- c(&quot;LZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;,&quot;DZFDC&quot;,&quot;PRC2&quot;)


colorSLO &lt;- c(viridis(3)[c(1,2)], brewer.pal(n=9, name=&quot;YlOrRd&quot;)[c(4)])
colorCond &lt;- brewer.pal(n=3, name=&quot;Paired&quot;)[c(1,2)]
colorSLOCond &lt;- c(&quot;#440154FF&quot;,&quot;#807DBA&quot;, &quot;#FEB24C&quot;, &quot;#F16913&quot;, &quot;#21908CFF&quot;)

names(colorSLO) &lt;- c(&quot;LN&quot;, &quot;PP&quot;, &quot;Spleen&quot;)
names(colorCond) &lt;- c(&quot;naive&quot;, &quot;immunized&quot;)
names(colorSLOCond) &lt;- c(&quot;LN_immunized&quot;,&quot;LN_naive&quot;, &quot;Spleen_naive&quot;,
                         &quot;Spleen_immunized&quot;, &quot;PP_immunized&quot;)


colLYM &lt;- c(&quot;#a0b9c4&quot;, &quot;#78abb7&quot;, &quot;#567a83&quot;, &quot;#3c454a&quot;, &quot;#0C3C52&quot;,
            &quot;#924440&quot;, &quot;#eab562&quot;,
            &quot;#f1d75f&quot;, &quot;#b5a147&quot;, &quot;#786c30&quot;)


names(colLYM) &lt;- c(&quot;naiveBcells&quot;, &quot;LZBcells&quot;, &quot;DZshmBcells&quot;, &quot;proDZ&quot;,
                   &quot;plasmaCells&quot;, 
                   &quot;Tfh&quot;, &quot;DCs&quot;,
                   &quot;mdMacrophages&quot;, &quot;trMacrophages&quot;, &quot;infMacrophages&quot;)

colLYM2 &lt;- c(&quot;#a0b9c4&quot;, &quot;#567a83&quot;, &quot;#0C3C52&quot;,
            &quot;#924440&quot;, &quot;#eab562&quot;,
            &quot;#b5a147&quot;)


names(colLYM2) &lt;- c(&quot;naiveBcells&quot;, &quot;GCBcells&quot;,
                   &quot;plasmaCells&quot;, 
                   &quot;Tfh&quot;, &quot;DCs&quot;,
                   &quot;Myeloids&quot;)</code></pre>
</div>
<div id="load-input-data" class="section level2">
<h2>load input data</h2>
<pre class="r"><code>basedir &lt;- here()
seuratLY &lt;- readRDS(file = paste0(basedir, 
                              &quot;/data/lympho/allSamples_mergedLab_seurat.rds&quot;))
seuratLY &lt;- subset(seuratLY, clusterLabel %in% names(colLYM))

GCBs &lt;- c(&quot;LZBcells&quot;, &quot;DZshmBcells&quot;, &quot;proDZ&quot;)
Myel &lt;- c(&quot;mdMacrophages&quot;, &quot;trMacrophages&quot;, &quot;infMacrophages&quot;)
seuratLY$clusterLabel2 &lt;- seuratLY$clusterLabel
seuratLY$clusterLabel2[which(seuratLY$clusterLabel %in% GCBs)] &lt;- &quot;GCBcells&quot;
seuratLY$clusterLabel2[which(seuratLY$clusterLabel %in% Myel)] &lt;- &quot;Myeloids&quot;

seuratBRC &lt;- readRDS(file=paste0(basedir, &quot;/data/folBRC_allSLO_seurat.rds&quot;))

## Each SLO individually
seuratLY$label_plus_SLO &lt;- paste0(seuratLY$clusterLabel2, &quot;_&quot;, seuratLY$SLO)
seuratBRC$label_plus_SLO &lt;- paste0(seuratBRC$clusterLabel, &quot;_&quot;, seuratBRC$SLO)


## plot input data
Idents(seuratLY) &lt;- seuratLY$clusterLabel2
DimPlot(seuratLY, reduction = &quot;umap&quot;, cols = colLYM2)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/input%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-input-data-1">
Past versions of “input data-1.png”
</button>
</p>
<div id="fig-no-spaces-input-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/input data-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/input data-1.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratBRC) &lt;- seuratBRC$clusterLabel
DimPlot(seuratBRC, reduction = &quot;umap&quot;, cols = colBRC)+
  theme_bw() +
  theme(axis.text = element_blank(), axis.ticks = element_blank(), 
        panel.grid.minor = element_blank()) +
  xlab(&quot;UMAP1&quot;) +
  ylab(&quot;UMAP2&quot;)</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/input%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-input-data-2">
Past versions of “input data-2.png”
</button>
</p>
<div id="fig-no-spaces-input-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/input data-2.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="assign-groups" class="section level2">
<h2>assign groups</h2>
<pre class="r"><code>seuratBRC$group &lt;- &quot;BRC&quot;
seuratLY$group &lt;- &quot;LYMPHO&quot;

Idents(seuratLY) &lt;- seuratLY$label_plus_SLO
Idents(seuratBRC) &lt;- seuratBRC$label_plus_SLO</code></pre>
</div>
<div id="read-sel-genes" class="section level2">
<h2>read sel genes</h2>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratBRC)) %&gt;% 
  mutate(geneID=gsub(&quot;.*\\.&quot;, &quot;&quot;, gene))
selGenes &lt;- read_tsv(file=paste0(basedir, 
                                 &quot;/data/GSEA/subsetSpecificFunctGenes&quot;, 
                                 &quot;InteractionSignaling.txt&quot;)) %&gt;% 
  left_join(., genes, by=&quot;geneID&quot;)

genesBRC &lt;- selGenes %&gt;% filter(group==&quot;BRC&quot;)
genesLym &lt;- selGenes %&gt;% filter(group==&quot;lympho&quot;)</code></pre>
</div>
<div id="plot-heatmap-sel-genes" class="section level2 tabset">
<h2 class="tabset">plot heatmap sel genes</h2>
<pre class="r"><code>avgHeatmap &lt;- function(seurat, selGenes, scaleDim = &quot;row&quot;, colVecIdent, 
                       colVecCond=NULL,
                       ordVec=NULL, gapVecR=NULL, gapVecC=NULL,cc=FALSE,
                       cr=FALSE, condCol=FALSE){
  
  ## format gene names (depends on how gene list is inputed)
  selGenes &lt;- selGenes %&gt;% 
      mutate(geneID = (str_split(gene, &#39;\\.&#39;, simplify = T)[,2]))
  #selGenes &lt;- selGenes$labelNam
  
  ## assay data
  clusterAssigned &lt;- as.data.frame(Idents(seurat)) %&gt;%
  dplyr::mutate(cell=rownames(.))
  colnames(clusterAssigned)[1] &lt;- &quot;ident&quot;
  seuratDat &lt;- GetAssayData(seurat)
  
  ## genes of interest
  genes &lt;- data.frame(gene=rownames(seurat)) %&gt;% 
    mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene)) %&gt;%
    filter(geneID %in% selGenes$geneID)

  ## matrix with averaged cnts per ident
  logNormExpres &lt;- as.data.frame(t(as.matrix(
    seuratDat[which(rownames(seuratDat) %in% genes$gene),])))
  logNormExpres &lt;- logNormExpres %&gt;% dplyr::mutate(cell=rownames(.)) %&gt;%
    dplyr::left_join(.,clusterAssigned, by=c(&quot;cell&quot;)) %&gt;%
    dplyr::select(-cell) %&gt;% dplyr::group_by(ident) %&gt;%
    dplyr::summarise_all(mean)
  logNormExpresMa &lt;- logNormExpres %&gt;% dplyr::select(-ident) %&gt;% as.matrix()
  rownames(logNormExpresMa) &lt;- logNormExpres$ident
  logNormExpresMa &lt;- t(logNormExpresMa)
  rownames(logNormExpresMa) &lt;- gsub(&quot;^.*?\\.&quot;,&quot;&quot;,rownames(logNormExpresMa))
  
  ## remove genes if they are all the same in all groups
  ind &lt;- apply(logNormExpresMa, 1, sd) == 0
  logNormExpresMa &lt;- logNormExpresMa[!ind,]
  genes &lt;- genes[!ind,]

  ## color columns according to cluster
  annotation_col &lt;- as.data.frame(gsub(&quot;(^.*?_)&quot;,&quot;&quot;,
                                       colnames(logNormExpresMa)))%&gt;%
    dplyr::mutate(celltype=gsub(&quot;(_.*$)&quot;,&quot;&quot;,colnames(logNormExpresMa)))
  colnames(annotation_col)[1] &lt;- &quot;col1&quot;
  annotation_col &lt;- annotation_col %&gt;%
    dplyr::mutate(cond = gsub(&quot;(^[0-9]_?)&quot;,&quot;&quot;,col1)) %&gt;%
    dplyr::select(cond, celltype)
  rownames(annotation_col) &lt;- colnames(logNormExpresMa) 

  ann_colors = list(
      cond = colVecCond,
      celltype=colVecIdent)
  if(is.null(ann_colors$cond)){
    annotation_col$cond &lt;- NULL
  }
  
  ## adjust order
  logNormExpresMa &lt;- logNormExpresMa[selGenes$geneID,]
  if(is.null(ordVec)){
    ordVec &lt;- levels(seurat)
  }
  logNormExpresMa &lt;- logNormExpresMa[,ordVec]

  ## scaled row-wise
  pheatmap(logNormExpresMa, scale=scaleDim ,treeheight_row = 0, cluster_rows = cr, 
         cluster_cols = cc, breaks = seq(from=-2, to=2,by=0.1),
         color = colorRampPalette(c(&quot;#111320&quot;, &quot;#4d5374&quot;, &quot;#899bff&quot;))(40),
         annotation_col = annotation_col, cellwidth=15, cellheight=10,
         annotation_colors = ann_colors, gaps_row = gapVecR, gaps_col = gapVecC)
}</code></pre>
<div id="brcs" class="section level3">
<h3>BRCs</h3>
<pre class="r"><code>gapC &lt;- seq(from = 3, to = length(levels(seuratBRC)), by = 3)
grpCnt &lt;- genesBRC %&gt;% group_by(functGrp) %&gt;% summarise(cnt=n())
gapR &lt;- data.frame(functGrp=unique(genesBRC$functGrp)) %&gt;% 
  left_join(.,grpCnt, by=&quot;functGrp&quot;) %&gt;% mutate(cumSum=cumsum(cnt)) 


ordBRC &lt;- c(&quot;LZFDC&quot;, &quot;DZFDC&quot;, &quot;MRC&quot;, &quot;TBRC&quot;, &quot;PRC2&quot;)
ordSLO &lt;- c(&quot;LN&quot;, &quot;Spleen&quot;, &quot;PP&quot;)
ordDat &lt;- expand.grid(SLO = ordSLO, BRC = ordBRC) %&gt;% 
  mutate(ordVec=paste0(BRC, &quot;_&quot;, SLO))

# pHeat &lt;- lapply(unique(genesBRC$functGrp), function(grp){
#   selGenesFunct &lt;- genesBRC %&gt;% filter(functGrp == grp)
#   pOut &lt;- avgHeatmap(seurat = seuratBRC, selGenes = selGenesFunct,
#                   colVecIdent = colBRC, 
#                   ordVec=ordDat$ordVec,
#                   gapVecC=gapC,cc=F, cr=F,
#                   condCol=T, colVecCond = colorSLO)
# })


pOut &lt;- avgHeatmap(seurat = seuratBRC, selGenes = genesBRC,
                  colVecIdent = colBRC, 
                  ordVec=ordDat$ordVec,
                  gapVecC=gapC,cc=F,
                  gapVecR=gapR$cumSum, cr=F,
                  condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/heatmap%20BRCs-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-BRCs-1">
Past versions of “heatmap BRCs-1.png”
</button>
</p>
<div id="fig-no-spaces-heatmap-BRCs-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap BRCs-1.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap BRCs-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap BRCs-1.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratBRC) &lt;- seuratBRC$clusterLabel
pOut &lt;- avgHeatmap(seurat = seuratBRC, selGenes = genesBRC,
                  colVecIdent = colBRC, 
                  ordVec=ordBRC,
                  gapVecC=NULL,cc=F,
                  gapVecR=gapR$cumSum, cr=F,
                  condCol=F)</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/heatmap%20BRCs-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-BRCs-2">
Past versions of “heatmap BRCs-2.png”
</button>
</p>
<div id="fig-no-spaces-heatmap-BRCs-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap BRCs-2.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap BRCs-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="immune-cells" class="section level3">
<h3>immune cells</h3>
<pre class="r"><code>gapC &lt;- seq(from = 3, to = length(levels(seuratLY)), by = 3)
grpCnt &lt;- genesLym %&gt;% group_by(functGrp) %&gt;% summarise(cnt=n())
gapR &lt;- data.frame(functGrp=unique(genesLym$functGrp)) %&gt;% 
  left_join(.,grpCnt, by=&quot;functGrp&quot;) %&gt;% mutate(cumSum=cumsum(cnt)) 


ordLYM &lt;- c(&quot;naiveBcells&quot;, &quot;GCBcells&quot;, &quot;plasmaCells&quot;, 
                   &quot;Tfh&quot;, &quot;Myeloids&quot;, &quot;DCs&quot;)
ordSLO &lt;- c(&quot;LN&quot;, &quot;Spleen&quot;, &quot;PP&quot;)
ordDat &lt;- expand.grid(SLO = ordSLO, LYM = ordLYM) %&gt;% 
  mutate(ordVec=paste0(LYM, &quot;_&quot;, SLO))

# pHeat &lt;- lapply(unique(genesLym$functGrp), function(grp){
#   selGenesFunct &lt;- genesLym %&gt;% filter(functGrp == grp)
#   pOut &lt;- avgHeatmap(seurat = seuratLY, selGenes = selGenesFunct,
#                   colVecIdent = colLYM2, 
#                   ordVec=ordDat$ordVec,
#                   gapVecC=gapC,cc=F, cr=F,
#                   condCol=T, colVecCond = colorSLO)
# })


pOut &lt;- avgHeatmap(seurat = seuratLY, selGenes = genesLym,
                  colVecIdent = colLYM2, 
                  ordVec=ordDat$ordVec,
                  gapVecC=gapC,cc=F,
                  gapVecR=gapR$cumSum, cr=F,
                  condCol=T, colVecCond = colorSLO)</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/heatmap%20ImmuneCells-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-ImmuneCells-1">
Past versions of “heatmap ImmuneCells-1.png”
</button>
</p>
<div id="fig-no-spaces-heatmap-ImmuneCells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap ImmuneCells-1.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap ImmuneCells-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap ImmuneCells-1.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Idents(seuratLY) &lt;- seuratLY$clusterLabel2
pOut &lt;- avgHeatmap(seurat = seuratLY, selGenes = genesLym,
                  colVecIdent = colLYM2, 
                  ordVec=ordLYM,
                  gapVecC=NULL,cc=F,
                  gapVecR=gapR$cumSum, cr=F,
                  condCol=F)</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/heatmap%20ImmuneCells-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-heatmap-ImmuneCells-2">
Past versions of “heatmap ImmuneCells-2.png”
</button>
</p>
<div id="fig-no-spaces-heatmap-ImmuneCells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap ImmuneCells-2.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/heatmap ImmuneCells-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="violin-plot-sel-genes" class="section level2 tabset">
<h2 class="tabset">violin plot sel genes</h2>
<div id="brc" class="section level3">
<h3>BRC</h3>
<pre class="r"><code>seuratBRC$clusterLabel &lt;- factor(seuratBRC$clusterLabel, levels = ordBRC)
pList &lt;- sapply(genesBRC$gene, function(x){
  p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &quot;clusterLabel&quot;,
               cols = colBRC, pt.size = 0
               )
  plot(p)
})</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-1">
Past versions of “violin sel genes BRC-1.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-1.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-2">
Past versions of “violin sel genes BRC-2.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-2.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-3">
Past versions of “violin sel genes BRC-3.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-3.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-4">
Past versions of “violin sel genes BRC-4.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-4.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-5">
Past versions of “violin sel genes BRC-5.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-5.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-5.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-6">
Past versions of “violin sel genes BRC-6.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-6.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-6.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-7">
Past versions of “violin sel genes BRC-7.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-7.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-7.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-7.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-8">
Past versions of “violin sel genes BRC-8.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-8.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-8.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-8.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-9">
Past versions of “violin sel genes BRC-9.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-9.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-9.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-9.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-10">
Past versions of “violin sel genes BRC-10.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-10.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-10.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-10.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-11">
Past versions of “violin sel genes BRC-11.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-11.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-11.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-11.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-12">
Past versions of “violin sel genes BRC-12.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-12.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-12.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-12.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-13">
Past versions of “violin sel genes BRC-13.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-13.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-13.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20BRC-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-BRC-14">
Past versions of “violin sel genes BRC-14.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-BRC-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-14.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes BRC-14.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="immune-cells-1" class="section level3">
<h3>Immune cells</h3>
<pre class="r"><code>seuratLY$clusterLabel2 &lt;- factor(seuratLY$clusterLabel2, levels = ordLYM)
pList &lt;- sapply(genesLym$gene, function(x){
  p &lt;- VlnPlot(object = seuratLY, features = x,
               group.by = &quot;clusterLabel2&quot;,
               cols = colLYM2, pt.size = 0
               )
  plot(p)
})</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-1">
Past versions of “violin sel genes ImmCells-1.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-1.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-2">
Past versions of “violin sel genes ImmCells-2.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-2.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-3">
Past versions of “violin sel genes ImmCells-3.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-3.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-4">
Past versions of “violin sel genes ImmCells-4.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-4.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-5">
Past versions of “violin sel genes ImmCells-5.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-5.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-6">
Past versions of “violin sel genes ImmCells-6.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-6.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-7">
Past versions of “violin sel genes ImmCells-7.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-7.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-7.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-7.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-8">
Past versions of “violin sel genes ImmCells-8.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-8.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-8.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-8.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-9">
Past versions of “violin sel genes ImmCells-9.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-9.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-9.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-9.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-10">
Past versions of “violin sel genes ImmCells-10.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-10.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-10.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-10.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-11">
Past versions of “violin sel genes ImmCells-11.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-11.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-11.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-11.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-12">
Past versions of “violin sel genes ImmCells-12.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-12.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-12.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-12.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-13">
Past versions of “violin sel genes ImmCells-13.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-13.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-13.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-13.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-14">
Past versions of “violin sel genes ImmCells-14.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-14.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-14.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-14.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-15">
Past versions of “violin sel genes ImmCells-15.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-15.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-15.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-15.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/violin%20sel%20genes%20ImmCells-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-violin-sel-genes-ImmCells-16">
Past versions of “violin sel genes ImmCells-16.png”
</button>
</p>
<div id="fig-no-spaces-violin-sel-genes-ImmCells-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/fe2d3c36c405b261fc34e79f2dcf0a3463c12a25/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-16.png" target="_blank">fe2d3c3</a>
</td>
<td>
mluetge
</td>
<td>
2022-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-16.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/d202c26a3d47b21ce7a146a8a175c19cd9d5c2e8/docs/figure/visSelIntHeatmapSignaling.Rmd/violin sel genes ImmCells-16.png" target="_blank">d202c26</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-01
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="downstream-signaling" class="section level2">
<h2>downstream signaling</h2>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratBRC)) %&gt;% 
  mutate(geneID=gsub(&quot;.*\\.&quot;, &quot;&quot;, gene))
selGenes &lt;- read_tsv(file=paste0(basedir, 
                            &quot;/data/cellphonedb/downstreamSignalingPW.txt&quot;)) %&gt;% 
  left_join(., genes, by=&quot;geneID&quot;) </code></pre>
<div id="vlnplots" class="section level3 tabset">
<h3 class="tabset">vlnPlots</h3>
<pre class="r"><code>allSign &lt;- unique(selGenes$signaling)
template_vp &lt;- c(
    &quot;#### {{sig}}\n&quot;,
    &quot;```{r down vp {{sig}}}\n&quot;,
    &quot;sign &lt;- as.character(&#39;{{sig}}&#39;)&quot;,
    &quot;signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)&quot;,
    &quot;pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })&quot;,
    &quot;```\n&quot;,
    &quot;\n&quot;
)
    
plots_vp &lt;- lapply(allSign, 
  function(sig) knitr::knit_expand(text = template_vp)
)</code></pre>
<div id="tgfbdown" class="section level4">
<h4>tgfbDown</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;tgfbDown&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-1">
Past versions of “down vp tgfbDown-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-2">
Past versions of “down vp tgfbDown-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-3">
Past versions of “down vp tgfbDown-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-4">
Past versions of “down vp tgfbDown-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-5">
Past versions of “down vp tgfbDown-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-6">
Past versions of “down vp tgfbDown-6.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-7">
Past versions of “down vp tgfbDown-7.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-7.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-8">
Past versions of “down vp tgfbDown-8.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-8.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-9">
Past versions of “down vp tgfbDown-9.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-9.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-10">
Past versions of “down vp tgfbDown-10.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-10.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-11">
Past versions of “down vp tgfbDown-11.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-11.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfbDown-12">
Past versions of “down vp tgfbDown-12.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfbDown-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfbDown-12.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tgfb" class="section level4">
<h4>tgfb</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;tgfb&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-1">
Past versions of “down vp tgfb-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-2">
Past versions of “down vp tgfb-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-3">
Past versions of “down vp tgfb-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-4">
Past versions of “down vp tgfb-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-5">
Past versions of “down vp tgfb-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-6">
Past versions of “down vp tgfb-6.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-7">
Past versions of “down vp tgfb-7.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-7.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-8">
Past versions of “down vp tgfb-8.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-8.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-9">
Past versions of “down vp tgfb-9.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-9.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-10">
Past versions of “down vp tgfb-10.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-10.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-11">
Past versions of “down vp tgfb-11.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-11.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-tgfb-12">
Past versions of “down vp tgfb-12.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-tgfb-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp tgfb-12.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nctgfbpi3k" class="section level4">
<h4>ncTgfbPI3K</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;ncTgfbPI3K&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbPI3K-1">
Past versions of “down vp ncTgfbPI3K-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbPI3K-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbPI3K-2">
Past versions of “down vp ncTgfbPI3K-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbPI3K-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbPI3K-3">
Past versions of “down vp ncTgfbPI3K-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbPI3K-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbPI3K-4">
Past versions of “down vp ncTgfbPI3K-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbPI3K-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbPI3K-5">
Past versions of “down vp ncTgfbPI3K-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbPI3K-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbPI3K-6">
Past versions of “down vp ncTgfbPI3K-6.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbPI3K-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbPI3K-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nctgfbnfkb" class="section level4">
<h4>ncTgfbNFKB</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;ncTgfbNFKB&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbNFKB-1">
Past versions of “down vp ncTgfbNFKB-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbNFKB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbNFKB-2">
Past versions of “down vp ncTgfbNFKB-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbNFKB-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbNFKB-3">
Past versions of “down vp ncTgfbNFKB-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbNFKB-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbNFKB-4">
Past versions of “down vp ncTgfbNFKB-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbNFKB-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbNFKB-5">
Past versions of “down vp ncTgfbNFKB-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbNFKB-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbNFKB-6">
Past versions of “down vp ncTgfbNFKB-6.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbNFKB-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbNFKB-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nctgfbras" class="section level4">
<h4>ncTgfbRas</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;ncTgfbRas&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRas-1">
Past versions of “down vp ncTgfbRas-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRas-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRas-2">
Past versions of “down vp ncTgfbRas-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRas-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRas-3">
Past versions of “down vp ncTgfbRas-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRas-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRas-4">
Past versions of “down vp ncTgfbRas-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRas-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRas-5">
Past versions of “down vp ncTgfbRas-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRas-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRas-6">
Past versions of “down vp ncTgfbRas-6.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRas-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRas-6.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nctgfbmapk" class="section level4">
<h4>ncTgfbMAPK</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;ncTgfbMAPK&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbMAPK-1">
Past versions of “down vp ncTgfbMAPK-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbMAPK-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbMAPK-2">
Past versions of “down vp ncTgfbMAPK-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbMAPK-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbMAPK-3">
Past versions of “down vp ncTgfbMAPK-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbMAPK-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbMAPK-4">
Past versions of “down vp ncTgfbMAPK-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbMAPK-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbMAPK-5">
Past versions of “down vp ncTgfbMAPK-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbMAPK-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbMAPK-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nctgfbrhoa" class="section level4">
<h4>ncTgfbRhoa</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;ncTgfbRhoa&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRhoa-1">
Past versions of “down vp ncTgfbRhoa-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRhoa-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRhoa-2">
Past versions of “down vp ncTgfbRhoa-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRhoa-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-ncTgfbRhoa-3">
Past versions of “down vp ncTgfbRhoa-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-ncTgfbRhoa-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp ncTgfbRhoa-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="il4" class="section level4">
<h4>Il4</h4>
<pre class="r"><code>sign &lt;- as.character(&#39;Il4&#39;)
signGenes &lt;- selGenes %&gt;% dplyr::filter(signaling == sign)
pList &lt;- sapply(signGenes$gene, function(x){
        p &lt;- VlnPlot(object = seuratBRC, features = x,
               group.by = &#39;clusterLabel&#39;,
               cols = colBRC, pt.size = 0
               )
        plot(p)
        })</code></pre>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-Il4-1">
Past versions of “down vp Il4-1.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-Il4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-1.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-Il4-2">
Past versions of “down vp Il4-2.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-Il4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-2.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-Il4-3">
Past versions of “down vp Il4-3.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-Il4-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-3.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-Il4-4">
Past versions of “down vp Il4-4.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-Il4-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-4.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-down-vp-Il4-5">
Past versions of “down vp Il4-5.png”
</button>
</p>
<div id="fig-no-spaces-down-vp-Il4-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mluetge/CrossSLO_BRC_CXCL13/blob/18aa3552ba57cf0dd22dc96daa0260d008c7107f/figure/visSelIntHeatmapSignaling.Rmd/down vp Il4-5.png" target="_blank">18aa355</a>
</td>
<td>
mluetge
</td>
<td>
2021-12-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pheatmap_1.0.12             RColorBrewer_1.1-3          viridis_0.6.2              
 [4] viridisLite_0.4.0           here_1.0.1                  forcats_0.5.1              
 [7] stringr_1.4.0               readr_2.1.2                 tidyr_1.2.0                
[10] tibble_3.1.6                tidyverse_1.3.1             mclust_5.4.9               
[13] SeuratObject_4.0.4          Seurat_4.1.0                scran_1.20.1               
[16] purrr_0.3.4                 Matrix_1.4-1                reshape2_1.4.4             
[19] dplyr_1.0.8                 scater_1.20.1               ggplot2_3.3.5              
[22] scuttle_1.2.1               SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
[25] Biobase_2.52.0              GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[28] IRanges_2.26.0              S4Vectors_0.30.2            BiocGenerics_0.38.0        
[31] MatrixGenerics_1.4.3        matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                reticulate_1.24           tidyselect_1.1.2         
  [4] htmlwidgets_1.5.4         grid_4.1.0                BiocParallel_1.26.2      
  [7] Rtsne_0.15                munsell_0.5.0             ScaledMatrix_1.0.0       
 [10] codetools_0.2-18          ica_1.0-2                 statmod_1.4.36           
 [13] future_1.24.0             miniUI_0.1.1.1            withr_2.5.0              
 [16] spatstat.random_2.2-0     colorspace_2.0-3          highr_0.9                
 [19] knitr_1.38                rstudioapi_0.13           ROCR_1.0-11              
 [22] tensor_1.5                listenv_0.8.0             labeling_0.4.2           
 [25] git2r_0.30.1              GenomeInfoDbData_1.2.6    polyclip_1.10-0          
 [28] bit64_4.0.5               farver_2.1.0              rprojroot_2.0.3          
 [31] parallelly_1.31.0         vctrs_0.4.1               generics_0.1.2           
 [34] xfun_0.30                 R6_2.5.1                  ggbeeswarm_0.6.0         
 [37] rsvd_1.0.5                locfit_1.5-9.5            bitops_1.0-7             
 [40] spatstat.utils_2.3-0      DelayedArray_0.18.0       assertthat_0.2.1         
 [43] vroom_1.5.7               promises_1.2.0.1          scales_1.2.0             
 [46] beeswarm_0.4.0            gtable_0.3.0              beachmat_2.8.1           
 [49] globals_0.14.0            goftest_1.2-3             workflowr_1.7.0          
 [52] rlang_1.0.2               splines_4.1.0             lazyeval_0.2.2           
 [55] spatstat.geom_2.4-0       broom_0.8.0               yaml_2.3.5               
 [58] abind_1.4-5               modelr_0.1.8              backports_1.4.1          
 [61] httpuv_1.6.5              tools_4.1.0               ellipsis_0.3.2           
 [64] spatstat.core_2.4-2       jquerylib_0.1.4           ggridges_0.5.3           
 [67] Rcpp_1.0.8.3              plyr_1.8.7                sparseMatrixStats_1.4.2  
 [70] zlibbioc_1.38.0           RCurl_1.98-1.6            rpart_4.1.16             
 [73] deldir_1.0-6              pbapply_1.5-0             cowplot_1.1.1            
 [76] zoo_1.8-9                 haven_2.4.3               ggrepel_0.9.1            
 [79] cluster_2.1.3             fs_1.5.2                  magrittr_2.0.3           
 [82] data.table_1.14.2         scattermore_0.8           reprex_2.0.1             
 [85] lmtest_0.9-40             RANN_2.6.1                whisker_0.4              
 [88] fitdistrplus_1.1-8        hms_1.1.1                 patchwork_1.1.1          
 [91] mime_0.12                 evaluate_0.15             xtable_1.8-4             
 [94] readxl_1.4.0              gridExtra_2.3             compiler_4.1.0           
 [97] KernSmooth_2.23-20        crayon_1.5.1              htmltools_0.5.2          
[100] tzdb_0.3.0                mgcv_1.8-40               later_1.3.0              
[103] lubridate_1.8.0           DBI_1.1.2                 dbplyr_2.1.1             
[106] MASS_7.3-56               cli_3.2.0                 metapod_1.0.0            
[109] igraph_1.3.0              pkgconfig_2.0.3           plotly_4.10.0            
[112] spatstat.sparse_2.1-0     xml2_1.3.3                vipor_0.4.5              
[115] bslib_0.3.1               dqrng_0.3.0               XVector_0.32.0           
[118] rvest_1.0.2               digest_0.6.29             sctransform_0.3.3        
[121] RcppAnnoy_0.0.19          spatstat.data_2.1-4       cellranger_1.1.0         
[124] rmarkdown_2.13            leiden_0.3.9              uwot_0.1.11              
[127] edgeR_3.34.1              DelayedMatrixStats_1.14.3 shiny_1.7.1              
[130] lifecycle_1.0.1           nlme_3.1-157              jsonlite_1.8.0           
[133] BiocNeighbors_1.10.0      limma_3.48.3              fansi_1.0.3              
[136] pillar_1.7.0              lattice_0.20-45           fastmap_1.1.0            
[139] httr_1.4.2                survival_3.3-1            glue_1.6.2               
[142] png_0.1-7                 bit_4.0.4                 bluster_1.2.1            
[145] stringi_1.7.6             sass_0.4.1                BiocSingular_1.8.1       
[148] irlba_2.3.5               future.apply_1.8.1       </code></pre>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Wed Apr 27 09:33:15 2022&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] pheatmap_1.0.12             RColorBrewer_1.1-3          viridis_0.6.2              
 [4] viridisLite_0.4.0           here_1.0.1                  forcats_0.5.1              
 [7] stringr_1.4.0               readr_2.1.2                 tidyr_1.2.0                
[10] tibble_3.1.6                tidyverse_1.3.1             mclust_5.4.9               
[13] SeuratObject_4.0.4          Seurat_4.1.0                scran_1.20.1               
[16] purrr_0.3.4                 Matrix_1.4-1                reshape2_1.4.4             
[19] dplyr_1.0.8                 scater_1.20.1               ggplot2_3.3.5              
[22] scuttle_1.2.1               SingleCellExperiment_1.14.1 SummarizedExperiment_1.22.0
[25] Biobase_2.52.0              GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
[28] IRanges_2.26.0              S4Vectors_0.30.2            BiocGenerics_0.38.0        
[31] MatrixGenerics_1.4.3        matrixStats_0.61.0         

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                reticulate_1.24           tidyselect_1.1.2         
  [4] htmlwidgets_1.5.4         grid_4.1.0                BiocParallel_1.26.2      
  [7] Rtsne_0.15                munsell_0.5.0             ScaledMatrix_1.0.0       
 [10] codetools_0.2-18          ica_1.0-2                 statmod_1.4.36           
 [13] future_1.24.0             miniUI_0.1.1.1            withr_2.5.0              
 [16] spatstat.random_2.2-0     colorspace_2.0-3          highr_0.9                
 [19] knitr_1.38                rstudioapi_0.13           ROCR_1.0-11              
 [22] tensor_1.5                listenv_0.8.0             labeling_0.4.2           
 [25] git2r_0.30.1              GenomeInfoDbData_1.2.6    polyclip_1.10-0          
 [28] bit64_4.0.5               farver_2.1.0              rprojroot_2.0.3          
 [31] parallelly_1.31.0         vctrs_0.4.1               generics_0.1.2           
 [34] xfun_0.30                 R6_2.5.1                  ggbeeswarm_0.6.0         
 [37] rsvd_1.0.5                locfit_1.5-9.5            bitops_1.0-7             
 [40] spatstat.utils_2.3-0      DelayedArray_0.18.0       assertthat_0.2.1         
 [43] vroom_1.5.7               promises_1.2.0.1          scales_1.2.0             
 [46] beeswarm_0.4.0            gtable_0.3.0              beachmat_2.8.1           
 [49] globals_0.14.0            goftest_1.2-3             workflowr_1.7.0          
 [52] rlang_1.0.2               splines_4.1.0             lazyeval_0.2.2           
 [55] spatstat.geom_2.4-0       broom_0.8.0               yaml_2.3.5               
 [58] abind_1.4-5               modelr_0.1.8              backports_1.4.1          
 [61] httpuv_1.6.5              tools_4.1.0               ellipsis_0.3.2           
 [64] spatstat.core_2.4-2       jquerylib_0.1.4           ggridges_0.5.3           
 [67] Rcpp_1.0.8.3              plyr_1.8.7                sparseMatrixStats_1.4.2  
 [70] zlibbioc_1.38.0           RCurl_1.98-1.6            rpart_4.1.16             
 [73] deldir_1.0-6              pbapply_1.5-0             cowplot_1.1.1            
 [76] zoo_1.8-9                 haven_2.4.3               ggrepel_0.9.1            
 [79] cluster_2.1.3             fs_1.5.2                  magrittr_2.0.3           
 [82] data.table_1.14.2         scattermore_0.8           reprex_2.0.1             
 [85] lmtest_0.9-40             RANN_2.6.1                whisker_0.4              
 [88] fitdistrplus_1.1-8        hms_1.1.1                 patchwork_1.1.1          
 [91] mime_0.12                 evaluate_0.15             xtable_1.8-4             
 [94] readxl_1.4.0              gridExtra_2.3             compiler_4.1.0           
 [97] KernSmooth_2.23-20        crayon_1.5.1              htmltools_0.5.2          
[100] tzdb_0.3.0                mgcv_1.8-40               later_1.3.0              
[103] lubridate_1.8.0           DBI_1.1.2                 dbplyr_2.1.1             
[106] MASS_7.3-56               cli_3.2.0                 metapod_1.0.0            
[109] igraph_1.3.0              pkgconfig_2.0.3           plotly_4.10.0            
[112] spatstat.sparse_2.1-0     xml2_1.3.3                vipor_0.4.5              
[115] bslib_0.3.1               dqrng_0.3.0               XVector_0.32.0           
[118] rvest_1.0.2               digest_0.6.29             sctransform_0.3.3        
[121] RcppAnnoy_0.0.19          spatstat.data_2.1-4       cellranger_1.1.0         
[124] rmarkdown_2.13            leiden_0.3.9              uwot_0.1.11              
[127] edgeR_3.34.1              DelayedMatrixStats_1.14.3 shiny_1.7.1              
[130] lifecycle_1.0.1           nlme_3.1-157              jsonlite_1.8.0           
[133] BiocNeighbors_1.10.0      limma_3.48.3              fansi_1.0.3              
[136] pillar_1.7.0              lattice_0.20-45           fastmap_1.1.0            
[139] httr_1.4.2                survival_3.3-1            glue_1.6.2               
[142] png_0.1-7                 bit_4.0.4                 bluster_1.2.1            
[145] stringi_1.7.6             sass_0.4.1                BiocSingular_1.8.1       
[148] irlba_2.3.5               future.apply_1.8.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
